From 0926530b2b240e62a15362d7295584070bf768d0 Mon Sep 17 00:00:00 2001
From: Ishimoto Shinobu <nagakamira@gmail.com>
Date: Thu, 16 Jul 2020 17:06:16 +0300
Subject: [PATCH] libnsl: no gettext

Signed-off-by: Ishimoto Shinobu <nagakamira@gmail.com>
---
 Makefile.am  | 2 +-
 configure.ac | 2 +-
 2 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/Makefile.am b/Makefile.am
index 8304907..e515f66 100644
--- a/Makefile.am
+++ b/Makefile.am
@@ -7,7 +7,7 @@ AUTOMAKE_OPTIONS = 1.9 foreign dist-xz check-news
 ACLOCAL_AMFLAGS=-I m4
 CLEANFILES = *~
 
-SUBDIRS = po src
+SUBDIRS = src
 
 pkgconfigdir=$(libdir)/pkgconfig
 pkgconfig_DATA = libnsl.pc
diff --git a/configure.ac b/configure.ac
index 90c9423..31fc241 100644
--- a/configure.ac
+++ b/configure.ac
@@ -37,4 +37,4 @@ AM_GNU_GETTEXT_REQUIRE_VERSION([0.19.2])
 AM_GNU_GETTEXT_VERSION([0.20.2])
 AM_GNU_GETTEXT([external])
 
-AC_OUTPUT([Makefile src/Makefile libnsl.pc po/Makefile.in])
+AC_OUTPUT([Makefile src/Makefile libnsl.pc])
-- 
2.28.0.rc0

