From 7d6730dc8d5291fe8a3383a12ad6b55b83774b64 Mon Sep 17 00:00:00 2001
From: Ishimoto Shinobu <nagakamira@gmail.com>
Date: Mon, 22 Jun 2020 04:46:07 +0900
Subject: [PATCH] libressl

Signed-off-by: Ishimoto Shinobu <nagakamira@gmail.com>
---
 src/idevice.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/idevice.c b/src/idevice.c
index fea1e03..47c9ab4 100644
--- a/src/idevice.c
+++ b/src/idevice.c
@@ -1015,7 +1015,7 @@ LIBIMOBILEDEVICE_API idevice_error_t idevice_connection_enable_ssl(idevice_conne
 		return ret;
 	}
 
-#if OPENSSL_VERSION_NUMBER >= 0x10100000L
+#if OPENSSL_VERSION_NUMBER >= 0x10100000L && !defined(LIBRESSL_VERSION_NUMBER)
 	SSL_CTX_set_security_level(ssl_ctx, 0);
 #endif
 
-- 
2.27.0

