From af26552e612d5f6875747ea37f85961f94c171ad Mon Sep 17 00:00:00 2001
From: Ishimoto Shinobu <nagakamira@gmail.com>
Date: Wed, 8 Jan 2020 20:49:20 +0900
Subject: [PATCH] use busybox provided udhcpc script

Signed-off-by: Ishimoto Shinobu <nagakamira@gmail.com>
---
 net/udhcpc.sh.Linux.in | 1 -
 1 file changed, 1 deletion(-)

diff --git a/net/udhcpc.sh.Linux.in b/net/udhcpc.sh.Linux.in
index 53fe5eb..c58c25d 100644
--- a/net/udhcpc.sh.Linux.in
+++ b/net/udhcpc.sh.Linux.in
@@ -74,7 +74,6 @@ udhcpc_start()
 	esac
 
 	eval "${x}" "${args}" --interface="${IFACE}" --now \
-		--script="@LIBEXECDIR@/sh/udhcpc-hook.sh" \
 		--pidfile="${pidfile}" >/dev/null
 	eend $? || return 1
 
-- 
2.24.0

