From f9195d1537f8f72678be2c3d75dde4a7b7947eb5 Mon Sep 17 00:00:00 2001
From: Ishimoto Shinobu <nagakamira@gmail.com>
Date: Tue, 23 Feb 2021 00:09:56 +0900
Subject: [PATCH 2/5] don't build images

Signed-off-by: Ishimoto Shinobu <nagakamira@gmail.com>
---
 Makefile | 1 -
 1 file changed, 1 deletion(-)

diff --git a/Makefile b/Makefile
index be9df04..5074ae6 100644
--- a/Makefile
+++ b/Makefile
@@ -12,7 +12,6 @@
 #
 all: test
 	make -C src all
-	make -C images all
 
 #
 #  make help
-- 
2.29.2

