From fb7a1a130ec65804c50e9ac4db2db922e990b511 Mon Sep 17 00:00:00 2001
From: Ishimoto Shinobu <nagakamira@gmail.com>
Date: Tue, 23 Feb 2021 00:11:35 +0900
Subject: [PATCH 3/5] remove installing images

Signed-off-by: Ishimoto Shinobu <nagakamira@gmail.com>
---
 Makefile | 5 +----
 1 file changed, 1 insertion(+), 4 deletions(-)

diff --git a/Makefile b/Makefile
index 5074ae6..713a73c 100644
--- a/Makefile
+++ b/Makefile
@@ -44,7 +44,7 @@ help:
 #
 alles: test
 	$(MAKE) -C src alles
-	$(MAKE) -C images all
+	$(MAKE) -C all
 
 #
 # documentation files
@@ -102,7 +102,6 @@ ins:
 #
 install:  all
 	$(MAKE) -C src install
-	$(MAKE) -C images install
 	$(MAKE) -C hooks install
 	$(MAKE) -C sample install
 	$(MAKE) -C scripts install
@@ -117,7 +116,6 @@ tidy:
 clean: tidy
 	rm -f test.img
 	$(MAKE) -C src clean
-	$(MAKE) -C images clean
 	$(MAKE) -C diagnose clean
 	$(MAKE) -C dos clean
 
@@ -129,7 +127,6 @@ distclean: clean
 
 uninstall:
 	$(MAKE) -C src uninstall
-	$(MAKE) -C images uninstall
 	$(MAKE) -C hooks uninstall
 	$(MAKE) -C sample uninstall
 	$(MAKE) -C scripts uninstall
-- 
2.29.2

