From 85938d3aee6d8e336b16420f8ce04a86afd7a07d Mon Sep 17 00:00:00 2001
From: Ishimoto Shinobu <nagakamira@gmail.com>
Date: Tue, 23 Feb 2021 00:14:02 +0900
Subject: [PATCH 4/5] use above 4G of memory

Signed-off-by: Ishimoto Shinobu <nagakamira@gmail.com>
---
 src/second.S | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/src/second.S b/src/second.S
index 37e9f1b..22e90e2 100644
--- a/src/second.S
+++ b/src/second.S
@@ -2975,6 +2975,8 @@ e8no1:	shrd	eax,ecx,#10	; convert to 1k blocks
 	shrd	memmap+8,eax,#10 ; convert to 1k
 	cmp	dword memmap,#1024	; below 1M
 	jb	e8go2		; below 1M, no interest
+	cmp	dword memmap,#4*1024*1024	; above 4G
+	jae	e8go2		; above 4G, no interest
 	cmp	esi,memmap+8	; check size
 	ja	e8go2		; want largest
 	mov	edx,memmap	; start (in 1k)
-- 
2.29.2

