From 4ec24eaba4e30db69fe583a22508f6ce956db5b4 Mon Sep 17 00:00:00 2001
From: Ishimoto Shinobu <nagakamira@gmail.com>
Date: Wed, 29 Jan 2020 12:05:47 -0600
Subject: [PATCH] definitions

Signed-off-by: Ishimoto Shinobu <nagakamira@gmail.com>
---
 arch/x86/mm/fault.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/arch/x86/mm/fault.c b/arch/x86/mm/fault.c
index 1375c406b..230a4e187 100644
--- a/arch/x86/mm/fault.c
+++ b/arch/x86/mm/fault.c
@@ -33,6 +33,10 @@
 #define CREATE_TRACE_POINTS
 #include <asm/trace/exceptions.h>
 
+#define PF_INSTR (1<<4)
+#define PF_PROT	 (1<<0)
+#define PF_WRITE (1<<0)
+
 /*
  * Returns 0 if mmiotrace is disabled, or if the fault is not
  * handled by mmiotrace:
-- 
2.18.2

