From 307cdd2dceda036f588ebd302c7a0c51cb4e3337 Mon Sep 17 00:00:00 2001
From: Con Kolivas <kernel@kolivas.org>
Date: Sat, 12 Aug 2017 12:02:04 +1000
Subject: [PATCH 14/16] Swap sucks.

---
 mm/vmscan.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/mm/vmscan.c b/mm/vmscan.c
index 572fb17c6273..0e42b5dc689e 100644
--- a/mm/vmscan.c
+++ b/mm/vmscan.c
@@ -177,7 +177,7 @@ struct scan_control {
 /*
  * From 0 .. 100.  Higher means more swappy.
  */
-int vm_swappiness = 60;
+int vm_swappiness = 33;
 /*
  * The total number of pages which are beyond the high watermark within all
  * zones.
-- 
2.20.1

