From 87e052a82c27381b8f063cbb10d066e498d14a3d Mon Sep 17 00:00:00 2001
From: Ishimoto Shinobu <nagakamira@gmail.com>
Date: Mon, 24 Aug 2020 18:40:37 +0500
Subject: [PATCH] disable gettext

Signed-off-by: Ishimoto Shinobu <nagakamira@gmail.com>
---
 configure.ac | 10 +---------
 1 file changed, 1 insertion(+), 9 deletions(-)

diff --git a/configure.ac b/configure.ac
index fbfc57e..9d6f070 100644
--- a/configure.ac
+++ b/configure.ac
@@ -9,15 +9,7 @@ AC_DISABLE_STATIC
 AM_INIT_AUTOMAKE([foreign -Wall -Werror -Wno-syntax -Wno-portability])
 AC_CONFIG_MACRO_DIR([m4])
 
-# Check for the gettext programs
-AC_PATH_PROG([XGETTEXT], [xgettext])
-AC_PATH_PROG([MSGFMT], [msgfmt])
-AC_PATH_PROG([MSGMERGE], [msgmerge])
-AS_IF([test -z "$XGETTEXT" -o -z "$MSGFMT" -o -z "$MSGMERGE"],
-      [AC_MSG_FAILURE([gettext not found])])
-
-# Define this so gettext.h works without requiring the whole gettext macro
-CFLAGS="${CFLAGS} -DENABLE_NLS"
+CFLAGS="${CFLAGS}"
 
 AM_PATH_PYTHON
 
-- 
2.18.4

