--- a/lib/Driver/ToolChains/Linux.cpp
+++ b/lib/Driver/ToolChains/Linux.cpp
@@ -816,8 +816,7 @@ void Linux::AddIAMCUIncludeArgs(const ArgList &DriverArgs,
 }

 bool Linux::isPIEDefault() const {
-  return (getTriple().isAndroid() && !getTriple().isAndroidVersionLT(16)) ||
-          getTriple().isMusl() || getSanitizerArgs().requiresPIE();
+  return true;
 }

 unsigned Linux::GetDefaultStackProtectorLevel(bool KernelOrKext) const {
