diff --git a/frontends/gtk/res/options.gtk2.ui b/frontends/gtk/res/options.gtk2.ui
index d5542ba..0044395 100644
--- a/frontends/gtk/res/options.gtk2.ui
+++ b/frontends/gtk/res/options.gtk2.ui
@@ -2717,67 +2717,16 @@
     </columns>
     <data>
       <row>
-        <col id="0">Google</col>
-      </row>
-      <row>
-        <col id="0">Yahoo!</col>
-      </row>
-      <row>
-        <col id="0">Microsoft Live</col>
-      </row>
-      <row>
-        <col id="0">Buisiness.com</col>
-      </row>
-      <row>
-        <col id="0">Omgili</col>
-      </row>
-      <row>
-        <col id="0">BBC News</col>
-      </row>
-      <row>
-        <col id="0">Ubuntu Packages</col>
-      </row>
-      <row>
         <col id="0">Creative Commons</col>
       </row>
       <row>
-        <col id="0">Ask</col>
-      </row>
-      <row>
-        <col id="0">Answers</col>
-      </row>
-      <row>
         <col id="0">Dictionary.com</col>
       </row>
       <row>
-        <col id="0">YouTube</col>
-      </row>
-      <row>
-        <col id="0">AeroMP3</col>
-      </row>
-      <row>
-        <col id="0">AOL</col>
-      </row>
-      <row>
-        <col id="0">Baidu</col>
-      </row>
-      <row>
-        <col id="0">Amazon</col>
-      </row>
-      <row>
-        <col id="0">Ebay</col>
-      </row>
-      <row>
-        <col id="0">IMBD</col>
-      </row>
-      <row>
-        <col id="0">ESPN</col>
-      </row>
-      <row>
         <col id="0">Wikipedia</col>
       </row>
       <row>
-        <col id="0">DuckDuckGo</col>
+        <col id="0">DuckDuckGo HTML</col>
       </row>
     </data>
   </object>
diff --git a/frontends/gtk/res/options.gtk3.ui b/frontends/gtk/res/options.gtk3.ui
index 2a3516f..8ff684f 100644
--- a/frontends/gtk/res/options.gtk3.ui
+++ b/frontends/gtk/res/options.gtk3.ui
@@ -2950,67 +2950,16 @@
     </columns>
     <data>
       <row>
-        <col id="0" translatable="no">Google</col>
-      </row>
-      <row>
-        <col id="0" translatable="no">Yahoo!</col>
-      </row>
-      <row>
-        <col id="0" translatable="no">Microsoft Live</col>
-      </row>
-      <row>
-        <col id="0" translatable="no">Buisiness.com</col>
-      </row>
-      <row>
-        <col id="0" translatable="no">Omgili</col>
-      </row>
-      <row>
-        <col id="0" translatable="no">BBC News</col>
-      </row>
-      <row>
-        <col id="0" translatable="no">Ubuntu Packages</col>
-      </row>
-      <row>
         <col id="0" translatable="no">Creative Commons</col>
       </row>
       <row>
-        <col id="0" translatable="no">Ask</col>
-      </row>
-      <row>
-        <col id="0" translatable="no">Answers</col>
-      </row>
-      <row>
         <col id="0" translatable="no">Dictionary.com</col>
       </row>
       <row>
-        <col id="0" translatable="no">YouTube</col>
-      </row>
-      <row>
-        <col id="0" translatable="no">AeroMP3</col>
-      </row>
-      <row>
-        <col id="0" translatable="no">AOL</col>
-      </row>
-      <row>
-        <col id="0" translatable="no">Baidu</col>
-      </row>
-      <row>
-        <col id="0" translatable="no">Amazon</col>
-      </row>
-      <row>
-        <col id="0" translatable="no">Ebay</col>
-      </row>
-      <row>
-        <col id="0" translatable="no">IMBD</col>
-      </row>
-      <row>
-        <col id="0" translatable="no">ESPN</col>
-      </row>
-      <row>
         <col id="0" translatable="no">Wikipedia</col>
       </row>
       <row>
-        <col id="0" translatable="no">DuckDuckGo</col>
+        <col id="0" translatable="no">DuckDuckGo HTML</col>
       </row>
     </data>
   </object>
