From 1b6cb991893b5bb5a006b58fc5aa0986d9b770b6 Mon Sep 17 00:00:00 2001
From: Ishimoto Shinobu <nagakamira@gmail.com>
Date: Mon, 2 Dec 2019 05:33:40 +0900
Subject: [PATCH] restore consolekit support

Signed-off-by: Ishimoto Shinobu <nagakamira@gmail.com>
---
 js/misc/loginManager.js | 77 +++++++++++++++++++++++++++++++++++------
 1 file changed, 66 insertions(+), 11 deletions(-)

diff --git a/js/misc/loginManager.js b/js/misc/loginManager.js
index f1ab257..d733335 100644
--- a/js/misc/loginManager.js
+++ b/js/misc/loginManager.js
@@ -3,6 +3,9 @@
 
 const { GLib, Gio } = imports.gi;
 const Signals = imports.signals;
+const UPowerGlib = imports.gi.UPowerGlib;
+
+const Main = imports.ui.main;
 
 const { loadInterfaceXML } = imports.misc.fileUtils;
 
@@ -14,6 +17,32 @@ const SystemdLoginManager = Gio.DBusProxy.makeProxyWrapper(SystemdLoginManagerIf
 const SystemdLoginSession = Gio.DBusProxy.makeProxyWrapper(SystemdLoginSessionIface);
 const SystemdLoginUser = Gio.DBusProxy.makeProxyWrapper(SystemdLoginUserIface);
 
+const ConsoleKitManagerIface = '<node> \
+<interface name="org.freedesktop.ConsoleKit.Manager"> \
+<method name="CanRestart"> \
+    <arg type="b" direction="out"/> \
+</method> \
+<method name="CanStop"> \
+    <arg type="b" direction="out"/> \
+</method> \
+<method name="Restart" /> \
+<method name="Stop" /> \
+<method name="GetCurrentSession"> \
+    <arg type="o" direction="out" /> \
+</method> \
+</interface> \
+</node>';
+
+const ConsoleKitSessionIface = '<node> \
+<interface name="org.freedesktop.ConsoleKit.Session"> \
+<signal name="Lock" /> \
+<signal name="Unlock" /> \
+</interface> \
+</node>';
+
+const ConsoleKitManager = Gio.DBusProxy.makeProxyWrapper(ConsoleKitManagerIface);
+const ConsoleKitSession = Gio.DBusProxy.makeProxyWrapper(ConsoleKitSessionIface);
+
 function haveSystemd() {
     return GLib.access("/run/systemd/seats", 0) >= 0;
 }
@@ -43,7 +72,7 @@ function canLock() {
                                                -1, null);
 
         let version = result.deep_unpack()[0].deep_unpack();
-        return haveSystemd() && versionCompare('3.5.91', version);
+       return versionCompare('3.5.91', version);
     } catch (e) {
         return false;
     }
@@ -83,7 +112,7 @@ function getLoginManager() {
         if (haveSystemd())
             _loginManager = new LoginManagerSystemd();
         else
-            _loginManager = new LoginManagerDummy();
+            _loginManager = new LoginManagerConsoleKit();
     }
 
     return _loginManager;
@@ -101,6 +130,9 @@ var LoginManagerSystemd = class {
                                   this._prepareForSleep.bind(this));
     }
 
+    // Having this function is a bit of a hack since the Systemd and ConsoleKit
+    // session objects have different interfaces - but in both cases there are
+    // Lock/Unlock signals, and that's all we count upon at the moment.
     getCurrentSessionProxy(callback) {
         if (this._currentSession) {
             callback (this._currentSession);
@@ -199,15 +231,37 @@ var LoginManagerSystemd = class {
 };
 Signals.addSignalMethods(LoginManagerSystemd.prototype);
 
-var LoginManagerDummy = class {
-    getCurrentSessionProxy(_callback) {
-        // we could return a DummySession object that fakes whatever callers
-        // expect (at the time of writing: connect() and connectSignal()
-        // methods), but just never calling the callback should be safer
+var LoginManagerConsoleKit = class {
+    constructor () {
+        this._proxy = new ConsoleKitManager(Gio.DBus.system,
+                                            'org.freedesktop.ConsoleKit',
+                                            '/org/freedesktop/ConsoleKit/Manager');
+        this._upClient = new UPowerGlib.Client();
+    }
+
+    getCurrentSessionProxy(callback) {
+        if (this._currentSession) {
+            callback (this._currentSession);
+            return;
+        }
+
+        this._proxy.GetCurrentSessionRemote((result, error) => {
+                if (error) {
+                    logError(error, 'Could not get a proxy for the current session');
+                } else {
+                    this._currentSession = new ConsoleKitSession(Gio.DBus.system,
+                                                                 'org.freedesktop.ConsoleKit',
+                                                                 result[0]);
+                    callback(this._currentSession);
+                }
+            });
     }
 
     canSuspend(asyncCallback) {
-        asyncCallback(false, false);
+        this._proxy.CanSuspendRemote((result, error) => {
+            asyncCallback(this._upClient.get_can_suspend(), false);
+            return false;
+        });
     }
 
     listSessions(asyncCallback) {
@@ -215,12 +269,13 @@ var LoginManagerDummy = class {
     }
 
     suspend() {
-        this.emit('prepare-for-sleep', true);
-        this.emit('prepare-for-sleep', false);
+        Main.screenShield.lock(true);
+
+        this._upClient.suspend_sync(null)
     }
 
     inhibit(reason, callback) {
         callback(null);
     }
 };
-Signals.addSignalMethods(LoginManagerDummy.prototype);
+Signals.addSignalMethods(LoginManagerConsoleKit.prototype);
-- 
2.24.0

From 854ffb7222ed7f4cb41033d8720cec9a3a7b7f43 Mon Sep 17 00:00:00 2001
From: Ishimoto Shinobu <nagakamira@gmail.com>
Date: Mon, 2 Dec 2019 05:39:57 +0900
Subject: [PATCH] fixes

---
 js/misc/loginManager.js | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/js/misc/loginManager.js b/js/misc/loginManager.js
index d733335baa..e9bd75bdc0 100644
--- a/js/misc/loginManager.js
+++ b/js/misc/loginManager.js
@@ -258,10 +258,10 @@ var LoginManagerConsoleKit = class {
     }
 
     canSuspend(asyncCallback) {
-        this._proxy.CanSuspendRemote((result, error) => {
+        Mainloop.idle_add(Lang.bind(this, function() {
             asyncCallback(this._upClient.get_can_suspend(), false);
             return false;
-        });
+        }));
     }
 
     listSessions(asyncCallback) {

