From 0bfed9bbba9555dd1620549fcd350c7891f44bc6 Mon Sep 17 00:00:00 2001
From: Ishimoto Shinobu <nagakamira@gmail.com>
Date: Mon, 2 Dec 2019 20:39:11 +0900
Subject: [PATCH] restore consolekit support

Signed-off-by: Ishimoto Shinobu <nagakamira@gmail.com>
---
 js/misc/loginManager.js | 10 +++++++++-
 1 file changed, 9 insertions(+), 1 deletion(-)

diff --git a/js/misc/loginManager.js b/js/misc/loginManager.js
index deb2d8d..0fba021 100644
--- a/js/misc/loginManager.js
+++ b/js/misc/loginManager.js
@@ -258,7 +258,13 @@ var LoginManagerConsoleKit = class {
     }
 
     canSuspend(asyncCallback) {
-        asyncCallback(this._upClient.get_can_suspend(), false);
+        this._proxy.CanSuspendRemote((result, error) => {
+            if (error) {
+                asyncCallback(false, false);
+            } else {
+                asyncCallback(this._upClient.get_can_suspend(), false);
+            }
+        });
     }
 
     listSessions(asyncCallback) {
@@ -266,6 +272,8 @@ var LoginManagerConsoleKit = class {
     }
 
     suspend() {
+        this._proxy.SuspendRemote(true);
+
         Main.screenShield.lock(true);
 
         this._upClient.suspend_sync(null)
-- 
2.24.0

