diff -Naur rustc-1.27.0-src.orig/src/librustc_target/spec/linux_musl_base.rs rustc-1.27.0-src/src/librustc_target/spec/linux_musl_base.rs
--- rustc-1.27.0-src.orig/src/librustc_target/spec/linux_musl_base.rs	2018-06-27 09:52:22.322947757 -0700
+++ rustc-1.27.0-src/src/librustc_target/spec/linux_musl_base.rs	2018-06-27 10:08:26.989930481 -0700
@@ -59,8 +59,6 @@
     base.pre_link_objects_exe.push("crti.o".to_string());
     base.post_link_objects.push("crtn.o".to_string());
 
-    // These targets statically link libc by default
-    base.crt_static_default = true;
     // These targets allow the user to choose between static and dynamic linking.
     base.crt_static_respected = true;
 
