From 1e9d653ccb0da803b1f30ded65c2dd727c89100c Mon Sep 17 00:00:00 2001
From: Samuel Holland <samuel@sholland.org>
Date: Mon, 11 Sep 2017 11:21:56 -0500
Subject: [PATCH 7/8] Add openssl configuration for musl targets

---
 src/bootstrap/native.rs | 7 ++++++-
 1 file changed, 6 insertions(+), 1 deletion(-)

diff --git a/src/bootstrap/native.rs b/src/bootstrap/native.rs
index 3c91cf3ecc..198ac36789 100644
--- a/src/bootstrap/native.rs
+++ b/src/bootstrap/native.rs
@@ -457,8 +457,11 @@ impl Step for Openssl {
             "arm-linux-androideabi" => "android",
             "arm-unknown-linux-gnueabi" => "linux-armv4",
             "arm-unknown-linux-gnueabihf" => "linux-armv4",
-            "armv7-linux-androideabi" => "android-armv7",
+            "arm-unknown-linux-musleabi" => "linux-armv4",
+            "arm-unknown-linux-musleabihf" => "linux-armv4",
+            "armv7-linux-androideabi" => "android-armv4",
             "armv7-unknown-linux-gnueabihf" => "linux-armv4",
+            "armv7-unknown-linux-musleabihf" => "linux-armv4",
             "i586-unknown-linux-gnu" => "linux-elf",
             "i586-unknown-linux-musl" => "linux-elf",
             "i686-apple-darwin" => "darwin-i386-cc",
@@ -468,9 +471,11 @@ impl Step for Openssl {
             "i686-unknown-linux-musl" => "linux-elf",
             "i686-unknown-netbsd" => "BSD-x86-elf",
             "mips-unknown-linux-gnu" => "linux-mips32",
+            "mips-unknown-linux-musl" => "linux-mips32",
             "mips64-unknown-linux-gnuabi64" => "linux64-mips64",
             "mips64el-unknown-linux-gnuabi64" => "linux64-mips64",
             "mipsel-unknown-linux-gnu" => "linux-mips32",
+            "mipsel-unknown-linux-musl" => "linux-mips32",
             "powerpc-unknown-linux-gnu" => "linux-ppc",
             "powerpc64-unknown-linux-gnu" => "linux-ppc64",
             "powerpc64le-unknown-linux-gnu" => "linux-ppc64le",
-- 
2.16.1

