diff -Naur rustc-1.26.2-src.orig/src/bootstrap/builder.rs rustc-1.26.2-src/src/bootstrap/builder.rs
--- rustc-1.26.2-src.orig/src/bootstrap/builder.rs	2018-06-16 11:22:29.173391907 -0700
+++ rustc-1.26.2-src/src/bootstrap/builder.rs	2018-06-16 11:31:32.573382176 -0700
@@ -709,7 +709,7 @@
 
             let cflags = self.cflags(target).join(" ");
             cargo.env(format!("CFLAGS_{}", target), cflags.clone())
-                 .env("CFLAGS", cflags.clone());
+                 .env("CFLAGS", cflags);
 
             if let Some(ar) = self.ar(target) {
                 let ranlib = format!("{} s", ar.display());
@@ -722,9 +722,7 @@
             if let Ok(cxx) = self.cxx(target) {
                 let cxx = ccacheify(&cxx);
                 cargo.env(format!("CXX_{}", target), &cxx)
-                     .env("CXX", &cxx)
-                     .env(format!("CXXFLAGS_{}", target), cflags.clone())
-                     .env("CXXFLAGS", cflags);
+                     .env("CXX", &cxx);
             }
         }
 
