diff --git a/src/llvm/include/llvm/Analysis/TargetLibraryInfo.def b/src/llvm/include/llvm/Analysis/TargetLibraryInfo.def
index 579d23004d1..6cd9d2722fb 100644
--- a/src/llvm/include/llvm/Analysis/TargetLibraryInfo.def
+++ b/src/llvm/include/llvm/Analysis/TargetLibraryInfo.def
@@ -27,6 +27,15 @@
 #define TLI_DEFINE_STRING_INTERNAL(string_repr) string_repr,
 #endif
 
+// avoid name conflicts with musl-libc
+#undef fopen64
+#undef fseeko64
+#undef ftello64
+#undef fstat64
+#undef lstat64
+#undef stat64
+#undef tmpfile64
+
 /// void *new(unsigned int);
 TLI_DEFINE_ENUM_INTERNAL(msvc_new_int)
 TLI_DEFINE_STRING_INTERNAL("??2@YAPAXI@Z")
