From 8dbc49deefb50f4af05710d3e6ad1ca7ff6dbfac Mon Sep 17 00:00:00 2001
From: Ishimoto Shinobu <nagakamira@gmail.com>
Date: Sun, 21 Feb 2021 09:43:21 +0900
Subject: [PATCH] remove -latomic library

Signed-off-by: Ishimoto Shinobu <nagakamira@gmail.com>
---
 node.gyp | 3 ---
 1 file changed, 3 deletions(-)

diff --git a/node.gyp b/node.gyp
index 131461de..051d2816 100644
--- a/node.gyp
+++ b/node.gyp
@@ -343,9 +343,6 @@
           '-Wl,-bnoerrmsg',
         ],
       }],
-      ['OS == "linux" and llvm_version != "0.0"', {
-        'libraries': ['-latomic'],
-      }],
     ],
   },
 
-- 
2.29.2

