From 42f7096030746cb347f19297bef4f18af16620ae Mon Sep 17 00:00:00 2001
From: Ishimoto Shinobu <nagakamira@gmail.com>
Date: Sun, 23 Aug 2020 22:44:14 +0500
Subject: [PATCH] disable gtk2

Signed-off-by: Ishimoto Shinobu <nagakamira@gmail.com>
---
 configure.ac | 7 -------
 1 file changed, 7 deletions(-)

diff --git a/configure.ac b/configure.ac
index ace54d1..1ed10b9 100644
--- a/configure.ac
+++ b/configure.ac
@@ -66,13 +66,6 @@ AS_IF([test "x$with_gtk" = x3],
                                          atk
                                          glib-2.0 >= $GLIB_REQUIRED_VERSION,
                                          [have_gtk=yes]
-)
-        ],
-      [test "x$with_gtk" = x2],
-        [PKG_CHECK_MODULES(DBUSMENUGTK,  gtk+-2.0 >= $GTK_REQUIRED_VERSION
-                                         atk
-                                         glib-2.0 >= $GLIB_REQUIRED_VERSION,
-                                         [have_gtk=yes]
 )
         ],
       [AC_MSG_FAILURE([Value for --with-gtk was neither 2 nor 3])]
-- 
2.18.4

