From 9d1799385497f51a3cbad0d10717f1864ca283b0 Mon Sep 17 00:00:00 2001
From: Ishimoto Shinobu <nagakamira@gmail.com>
Date: Fri, 26 Feb 2021 06:12:51 +0900
Subject: [PATCH] link against libexecinfo

Signed-off-by: Ishimoto Shinobu <nagakamira@gmail.com>
---
 src/Makefile.am | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/src/Makefile.am b/src/Makefile.am
index a389ce6..eafa77d 100644
--- a/src/Makefile.am
+++ b/src/Makefile.am
@@ -19,7 +19,8 @@ lxdm_binary_LDADD = \
 	$(GLIB_LIBS) \
 	$(XCONN_LIBS) \
 	$(CONSOLEKIT_LIBS) \
-	$(NULL)
+	$(NULL) \
+	-lexecinfo
 
 lxdm_binary_SOURCES = \
 	lxdm.c lxdm.h ui.c lxcom.c lxcom.h xconn.c xconn.h auth.c auth.h lxcommon.h \
-- 
2.29.2

