--- lxdm-0.4.1.orig/data/lxdm.conf.in	2012-03-21 17:31:21.524659586 +0100
+++ lxdm-0.4.1/data/lxdm.conf.in	2012-03-21 17:34:31.601351294 +0100
@@ -23,9 +23,11 @@
 
 [server]
 ## arg used to start xserver, not fully function
-# arg=/usr/bin/X -background vt1
+arg=/usr/bin/X -background none vt1
+
 # uncomment this if you really want xserver listen to tcp
 # tcp_listen=1
+
 # uncoment this if you want reset the xserver after logou
 # reset=1
 
