From 1a848d704bd1069852581286d7d3ef97e8425902 Mon Sep 17 00:00:00 2001
From: Ishimoto Shinobu <nagakamira@gmail.com>
Date: Sat, 21 Dec 2019 02:34:14 +0900
Subject: [PATCH] older librsvg

---
 meson.build | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/meson.build b/meson.build
index 941970a02..b1efab909 100644
--- a/meson.build
+++ b/meson.build
@@ -195,7 +195,7 @@ config_h.set('HAVE_INTROSPECTION', enable_introspection)
 # RSVG (optional for scaling svg image)
 enable_librsvg = get_option('librsvg')
 if enable_librsvg
-  common_deps += dependency('librsvg-2.0', version: '>= 2.44.0')
+  common_deps += dependency('librsvg-2.0', version: '>= 2.36.2')
 endif
 config_h.set('HAVE_RSVG', enable_librsvg)
 
