diff -ruN LVM2.2.02.184-org/libdaemon/server/daemon-server.c LVM2.2.02.184/libdaemon/server/daemon-server.c
--- LVM2.2.02.184-org/libdaemon/server/daemon-server.c	2019-03-22 19:15:00.000000000 +0900
+++ LVM2.2.02.184/libdaemon/server/daemon-server.c	2019-05-10 00:27:23.539569912 +0900
@@ -18,6 +18,7 @@
 #include "daemon-server.h"
 #include "daemon-log.h"
 
+#include <fcntl.h>
 #include <dlfcn.h>
 #include <errno.h>
 #include <pthread.h>
