From a488f6aa236b936b913e55fb8205347637574c89 Mon Sep 17 00:00:00 2001
From: Ishimoto Shinobu <nagakamira@gmail.com>
Date: Sat, 21 Nov 2020 19:02:14 +0300
Subject: [PATCH] intl

Signed-off-by: Ishimoto Shinobu <nagakamira@gmail.com>
---
 Makefile.am | 3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/Makefile.am b/Makefile.am
index c97d57a..f5e3e86 100644
--- a/Makefile.am
+++ b/Makefile.am
@@ -2,8 +2,7 @@ SUBDIRS = pixmaps\
 		  icons\
 		  shared\
 		  ${PLUGINS_DIRS}\
-		  po\
-		  intl
+		  po
 
 bin_PROGRAMS = deadbeef
 
-- 
2.28.0

