diff -ruN asciidoc-8.6.10/a2x.py asciidoc-8.6.10_b/a2x.py
--- asciidoc-8.6.10/a2x.py	2017-09-28 18:10:02.000000000 -0700
+++ asciidoc-8.6.10_b/a2x.py	2019-01-12 06:20:00.673015532 -0800
@@ -498,7 +498,6 @@
             self.asciidoc_opts += ' --doctype %s' % self.doctype
         for attr in self.attributes:
             self.asciidoc_opts += ' --attribute "%s"' % attr
-#        self.xsltproc_opts += ' --nonet'
         if self.verbose:
             self.asciidoc_opts += ' --verbose'
             self.dblatex_opts += ' -V'
@@ -649,7 +648,7 @@
         shell('"%s" --backend docbook -a "a2x-format=%s" %s --out-file "%s" "%s"' %
              (self.asciidoc, self.format, self.asciidoc_opts, docbook_file, self.asciidoc_file))
         if not self.no_xmllint and XMLLINT:
-            shell('"%s" --nonet --noout --valid "%s"' % (XMLLINT, docbook_file))
+            shell('"%s" --noout --valid "%s"' % (XMLLINT, docbook_file))
 
     def to_xhtml(self):
         self.to_docbook()
