From f3b0db5026143b7984365e4160d15cd36d4bdc95 Mon Sep 17 00:00:00 2001
From: Ishimoto Shinobu <nagakamira@gmail.com>
Date: Sat, 20 Feb 2021 21:28:00 +0900
Subject: [PATCH 2/2] missing readline symbol

Signed-off-by: Ishimoto Shinobu <nagakamira@gmail.com>
---
 client/display.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/client/display.c b/client/display.c
index fc90672..4091b7d 100644
--- a/client/display.c
+++ b/client/display.c
@@ -592,7 +592,9 @@ void display_enable_cmd_prompt(void)
 	 * The following sequence of rl_* commands forces readline to properly
 	 * update its internal state and re-display the new prompt.
 	 */
+#ifdef rl_save_prompt
 	rl_save_prompt();
+#endif
 	rl_redisplay();
 #ifdef rl_restore_prompt
 	rl_restore_prompt();
-- 
2.29.2

