From 802accd4216b62eb850b1c58d83467d556340076 Mon Sep 17 00:00:00 2001
From: Ishimoto Shinobu <nagakamira@gmail.com>
Date: Sun, 13 Dec 2020 01:12:49 +0900
Subject: [PATCH 2/4] fix bashisms

Signed-off-by: Ishimoto Shinobu <nagakamira@gmail.com>
---
 scripts/fixfiles | 23 ++++++++++++++---------
 1 file changed, 14 insertions(+), 9 deletions(-)

diff --git a/scripts/fixfiles b/scripts/fixfiles
index 5d77703..d0ba0bb 100755
--- a/scripts/fixfiles
+++ b/scripts/fixfiles
@@ -1,4 +1,4 @@
-#!/bin/bash
+#!/bin/sh
 # fixfiles
 #
 # Script to restore labels on a SELinux box
@@ -27,7 +27,7 @@ set -o nounset
 # number if the current kernel version is greater than 2.6.30, a negative
 # number if the current is less than 2.6.30 and 0 if they are the same.
 #
-function useseclabel {
+useseclabel () {
 	VER=`uname -r`
 	SUP=2.6.30
 	expr '(' "$VER" : '\([^.]*\)' ')' '-' '(' "$SUP" : '\([^.]*\)' ')' '|' \
@@ -93,9 +93,10 @@ exclude_dirs_from_relabelling() {
 	  # skip not absolute path
 	  # skip not directory
 	  [ -z "${i}" ] && continue
-	  [[ "${i}" =~ ^[[:blank:]]*# ]] && continue
-	  [[ ! "${i}" =~ ^/.* ]] && continue
-	  [[ ! -d "${i}" ]] && continue
+	  echo "${i}" | egrep -q '^[[:space:]]*#' && continue
+	  echo "${i}" | egrep -v '^/.*' && continue
+	  [ ! -d "${i}" ] && continue
+
 	  exclude_from_relabelling="$exclude_from_relabelling -e $i"
 	done < /etc/selinux/fixfiles_exclude_dirs
     fi
@@ -138,7 +139,7 @@ fi
 # Log directories excluded from relabelling by configuration file
 #
 LogExcluded() {
-for i in ${EXCLUDEDIRS//-e / }; do
+for i in `echo ${EXCLUDEDIRS} | sed -e 's/-e / /g'`; do
     echo "skipping the directory $i"
 done
 }
@@ -201,8 +202,12 @@ fi
 }
 
 rpmlist() {
-rpm -q --qf '[%{FILESTATES} %{FILENAMES}\n]' "$1" | grep '^0 ' | cut -f2- -d ' '
-[ ${PIPESTATUS[0]} != 0 ] && echo "$1 not found" >/dev/stderr
+    if rpm -q --qf '[%{FILESTATES} %{FILENAMES}\n]' "$1" >/dev/null
+    then
+        rpm -q --qf '[%{FILESTATES} %{FILENAMES}\n]' "$1" | grep '^0 ' | cut -f2- -d ' '
+    else
+        echo "$1 not found" >/dev/stderr
+    fi
 }
 
 #
@@ -277,7 +282,7 @@ relabel() {
 	exit 1
     fi
 
-    if [ $fullFlag == 1  ]; then
+    if [ $fullFlag = 1  ]; then
 	fullrelabel
 	return
     fi
-- 
2.29.2

