From 2c540e4c009d99aa95c69023a4823ee4a31b6c56 Mon Sep 17 00:00:00 2001
From: Ishimoto Shinobu <nagakamira@gmail.com>
Date: Sun, 13 Dec 2020 01:13:19 +0900
Subject: [PATCH 3/4] disable native language support

Signed-off-by: Ishimoto Shinobu <nagakamira@gmail.com>
---
 Makefile | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Makefile b/Makefile
index b930b29..8c12510 100644
--- a/Makefile
+++ b/Makefile
@@ -1,4 +1,4 @@
-SUBDIRS = setfiles load_policy newrole run_init secon sestatus semodule setsebool scripts po man hll
+SUBDIRS = setfiles load_policy newrole run_init secon sestatus semodule setsebool scripts man hll
 
 all install relabel clean indent:
 	@for subdir in $(SUBDIRS); do \
-- 
2.29.2

