From 4313dc1114c0ff8027311ac22d3590e0b6ce3d7a Mon Sep 17 00:00:00 2001
From: Ishimoto Shinobu <nagakamira@gmail.com>
Date: Sun, 13 Dec 2020 01:13:36 +0900
Subject: [PATCH 4/4] fix ataraxia kernel name

Signed-off-by: Ishimoto Shinobu <nagakamira@gmail.com>
---
 scripts/fixfiles | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/scripts/fixfiles b/scripts/fixfiles
index d0ba0bb..09888a8 100755
--- a/scripts/fixfiles
+++ b/scripts/fixfiles
@@ -28,7 +28,7 @@ set -o nounset
 # number if the current is less than 2.6.30 and 0 if they are the same.
 #
 useseclabel () {
-	VER=`uname -r`
+	VER=`uname -r | sed 's/-ataraxia//g'`
 	SUP=2.6.30
 	expr '(' "$VER" : '\([^.]*\)' ')' '-' '(' "$SUP" : '\([^.]*\)' ')' '|' \
 	     '(' "$VER.0" : '[^.]*[.]\([^.]*\)' ')' '-' '(' "$SUP.0" : '[^.]*[.]\([^.]*\)' ')' '|' \
-- 
2.29.2

