From d179828f7311d6ce6e944532e4659adc3a103357 Mon Sep 17 00:00:00 2001
From: Ishimoto Shinobu <nagakamira@gmail.com>
Date: Fri, 7 Feb 2020 03:11:43 -0600
Subject: [PATCH] fix

Signed-off-by: Ishimoto Shinobu <nagakamira@gmail.com>
---
 src/neko/neko.in | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/neko/neko.in b/src/neko/neko.in
index cfc7aa3..7281d73 100644
--- a/src/neko/neko.in
+++ b/src/neko/neko.in
@@ -840,8 +840,8 @@ main() {
 		search|sr) shift; mode="npk_search" ;;
 		sync|sy) shift; mode="npk_sync" ;;
 		upgrade|up) shift; mode="npk_upgrade" ;;
-		version) version ;;
-		*|help) usage ;;
+		version) npk_version ;;
+		*|help) npk_usage ;;
 	esac
 
 	while getopts :fwndestCDEBPc:m:r: opts; do
-- 
2.18.2

