From 920f90b930686dfb67aa3cb00c8b00bc49bec85b Mon Sep 17 00:00:00 2001
From: Ishimoto Shinobu <nagakamira@gmail.com>
Date: Mon, 24 Feb 2020 20:36:31 +0900
Subject: [PATCH] fix build on recent musl

---
 src/libopenbsd/fts.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/libopenbsd/fts.c b/src/libopenbsd/fts.c
index bd1eef1..636eb63 100644
--- a/src/libopenbsd/fts.c
+++ b/src/libopenbsd/fts.c
@@ -39,6 +39,7 @@
 #include <limits.h>
 #include <stdlib.h>
 #include <string.h>
+#include <stdint.h>
 #include <unistd.h>
 
 #include <bsd.h>
