From 120591c82a215f5c1d86511362abf8c70696f370 Mon Sep 17 00:00:00 2001
From: Ishimoto Shinobu <nagakamira@gmail.com>
Date: Sun, 20 Dec 2020 22:53:45 +0900
Subject: [PATCH] no docs

Signed-off-by: Ishimoto Shinobu <nagakamira@gmail.com>
---
 Makefile.am  | 2 +-
 configure.ac | 8 --------
 2 files changed, 1 insertion(+), 9 deletions(-)

diff --git a/Makefile.am b/Makefile.am
index 6060263..25f14da 100644
--- a/Makefile.am
+++ b/Makefile.am
@@ -1,6 +1,6 @@
 ## Process this file with automake to produce Makefile.in
 
-SUBDIRS = actions data src docs po
+SUBDIRS = actions data src po
 
 if BUILD_TEST
 SUBDIRS += test
diff --git a/configure.ac b/configure.ac
index 4ac2219..8af957e 100644
--- a/configure.ac
+++ b/configure.ac
@@ -51,8 +51,6 @@ AC_PATH_PROG([XSLTPROC], [xsltproc])
 fi
 AM_CONDITIONAL(MAN_PAGES_ENABLED, test x$enable_man_pages = xyes)
 
-GTK_DOC_CHECK([1.3])
-
 #### gcc warning flags
 
 CC_CHECK_FLAGS_APPEND([WARN_CFLAGS], [CFLAGS], [\
@@ -543,11 +541,6 @@ src/polkitbackend/Makefile
 src/polkitagent/Makefile
 src/programs/Makefile
 src/examples/Makefile
-docs/version.xml
-docs/extensiondir.xml
-docs/Makefile
-docs/polkit/Makefile
-docs/man/Makefile
 po/Makefile.in
 test/Makefile
 test/polkit/Makefile
@@ -591,7 +584,6 @@ echo "
 fi
 echo "
         Maintainer mode:            ${USE_MAINTAINER_MODE}
-        Building api docs:          ${enable_gtk_doc}
         Building man pages:         ${enable_man_pages}
         Building examples:          ${enable_examples}
 
-- 
2.29.2

