From e9b4a99d85069da7b62fcf1addf6af17179068ca Mon Sep 17 00:00:00 2001
From: Ishimoto Shinobu <nagakamira@gmail.com>
Date: Mon, 4 Jan 2021 02:20:12 +0900
Subject: [PATCH] change default wallpaper

Signed-off-by: Ishimoto Shinobu <nagakamira@gmail.com>
---
 common/xfdesktop-common.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/common/xfdesktop-common.h b/common/xfdesktop-common.h
index ae9f5f8..e4999f4 100644
--- a/common/xfdesktop-common.h
+++ b/common/xfdesktop-common.h
@@ -35,7 +35,7 @@
 #include <stdarg.h>
 
 #define XFDESKTOP_CHANNEL        "xfce4-desktop"
-#define DEFAULT_BACKDROP         DATADIR "/backgrounds/xfce/xfce-verticals.png"
+#define DEFAULT_BACKDROP         DATADIR "/backgrounds/xfce/default.jpg"
 #define DEFAULT_ICON_FONT_SIZE   12
 #define DEFAULT_ICON_SIZE        48
 #define ITHEME_FLAGS             (GTK_ICON_LOOKUP_USE_BUILTIN)
-- 
2.29.2

