From 98c2dfa60dea0d926176f5154b5650e24d0a33a7 Mon Sep 17 00:00:00 2001
From: Ishimoto Shinobu <nagakamira@gmail.com>
Date: Fri, 13 Nov 2020 23:09:53 +0300
Subject: [PATCH 4/7] better firstboot

Signed-off-by: Ishimoto Shinobu <nagakamira@gmail.com>
---
 units/systemd-firstboot.service | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/units/systemd-firstboot.service b/units/systemd-firstboot.service
index 9f5c710..ac13765 100644
--- a/units/systemd-firstboot.service
+++ b/units/systemd-firstboot.service
@@ -20,7 +20,10 @@ ConditionFirstBoot=yes
 [Service]
 Type=oneshot
 RemainAfterExit=yes
-ExecStart=systemd-firstboot --prompt-locale --prompt-timezone --prompt-root-password
+ExecStart=systemd-firstboot --prompt-keymap --prompt-timezone --prompt-hostname --prompt-root-password
 StandardOutput=tty
 StandardInput=tty
 StandardError=tty
+
+[Install]
+WantedBy=sysinit.target
-- 
2.28.0

