From 8632490c2546a9ab0b36f93e74bd2a34e398030f Mon Sep 17 00:00:00 2001
From: Ishimoto Shinobu <nagakamira@gmail.com>
Date: Thu, 3 Dec 2020 01:45:04 +0900
Subject: [PATCH 6/7] tune etc.conf

Signed-off-by: Ishimoto Shinobu <nagakamira@gmail.com>
---
 tmpfiles.d/etc.conf.m4 | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/tmpfiles.d/etc.conf.m4 b/tmpfiles.d/etc.conf.m4
index f82e0b8..a656e30 100644
--- a/tmpfiles.d/etc.conf.m4
+++ b/tmpfiles.d/etc.conf.m4
@@ -15,8 +15,8 @@ t /etc/mtab - - - - security.SMACK64=_
 m4_ifdef(`ENABLE_RESOLVE',
 L! /etc/resolv.conf - - - - ../run/systemd/resolve/stub-resolv.conf
 )m4_dnl
-C! /etc/nsswitch.conf - - - -
 m4_ifdef(`HAVE_PAM',
-C! /etc/pam.d - - - -
+C! /etc/pam.d/other - - - -
+C! /etc/pam.d/system-auth - - - -
 )m4_dnl
 C! /etc/issue - - - -
-- 
2.28.0

