diff -ruN qt-everywhere-src-5.13.0-orig/qtbase/src/corelib/codecs/qiconvcodec.cpp qt-everywhere-src-5.13.0/qtbase/src/corelib/codecs/qiconvcodec.cpp
--- qt-everywhere-src-5.13.0-orig/qtbase/src/corelib/codecs/qiconvcodec.cpp	2019-06-13 13:18:18.000000000 +0900
+++ qt-everywhere-src-5.13.0/qtbase/src/corelib/codecs/qiconvcodec.cpp	2019-08-05 20:11:53.500089077 +0900
@@ -61,7 +61,7 @@
 #elif defined(Q_OS_AIX)
 #  define NO_BOM
 #  define UTF16 "UCS-2"
-#elif defined(Q_OS_FREEBSD)
+#elif defined(Q_OS_FREEBSD) || (defined(Q_OS_LINUX) && !defined(__GLIBC__))
 #  define NO_BOM
 #  if Q_BYTE_ORDER == Q_BIG_ENDIAN
 #    define UTF16 "UTF-16BE"
diff -ruN qt-everywhere-src-5.13.0-orig/qtbase/src/network/configure.json qt-everywhere-src-5.13.0/qtbase/src/network/configure.json
--- qt-everywhere-src-5.13.0-orig/qtbase/src/network/configure.json	2019-06-13 13:18:18.000000000 +0900
+++ qt-everywhere-src-5.13.0/qtbase/src/network/configure.json	2019-08-05 20:14:05.004085440 +0900
@@ -167,7 +167,7 @@
             "test": {
                 "include": "openssl/opensslv.h",
                 "tail": [
-                    "#if !defined(OPENSSL_VERSION_NUMBER) || OPENSSL_VERSION_NUMBER-0 < 0x10100000L",
+                    "#if !defined(OPENSSL_VERSION_NUMBER) || OPENSSL_VERSION_NUMBER-0 < 0x10100000L || defined(LIBRESSL_VERSION_NUMBER)",
                     "#  error OpenSSL >= 1.1 is required",
                     "#endif"
                 ]
diff -ruN qt-everywhere-src-5.13.0-orig/qtbase/src/network/ssl/qsslcontext_openssl.cpp qt-everywhere-src-5.13.0/qtbase/src/network/ssl/qsslcontext_openssl.cpp
--- qt-everywhere-src-5.13.0-orig/qtbase/src/network/ssl/qsslcontext_openssl.cpp	2019-06-13 13:18:18.000000000 +0900
+++ qt-everywhere-src-5.13.0/qtbase/src/network/ssl/qsslcontext_openssl.cpp	2019-08-05 20:14:05.001085440 +0900
@@ -265,7 +265,7 @@
     }
 #endif // ocsp
 
-#if OPENSSL_VERSION_NUMBER >= 0x10002000L
+#if OPENSSL_VERSION_NUMBER >= 0x10002000L && !defined(LIBRESSL_VERSION_NUMBER)
     if (QSslSocket::sslLibraryVersionNumber() >= 0x10002000L) {
         QSharedPointer<SSL_CONF_CTX> cctx(q_SSL_CONF_CTX_new(), &q_SSL_CONF_CTX_free);
         if (cctx) {
diff -ruN qt-everywhere-src-5.13.0-orig/qtbase/src/network/ssl/qsslsocket_openssl.cpp qt-everywhere-src-5.13.0/qtbase/src/network/ssl/qsslsocket_openssl.cpp
--- qt-everywhere-src-5.13.0-orig/qtbase/src/network/ssl/qsslsocket_openssl.cpp	2019-06-13 13:18:18.000000000 +0900
+++ qt-everywhere-src-5.13.0/qtbase/src/network/ssl/qsslsocket_openssl.cpp	2019-08-05 20:14:09.834085306 +0900
@@ -604,7 +604,7 @@
             q_SSL_set_psk_server_callback(ssl, &q_ssl_psk_server_callback);
     }
 #endif
-#if OPENSSL_VERSION_NUMBER >= 0x10101006L
+#if OPENSSL_VERSION_NUMBER >= 0x10101006L && !defined(LIBRESSL_VERSION_NUMBER)
     // Set the client callback for TLSv1.3 PSK
     if (mode == QSslSocket::SslClientMode
         && QSslSocket::sslLibraryBuildVersionNumber() >= 0x10101006L) {
diff -ruN qt-everywhere-src-5.13.0-orig/qtbase/src/network/ssl/qsslsocket_opensslpre11_symbols_p.h qt-everywhere-src-5.13.0/qtbase/src/network/ssl/qsslsocket_opensslpre11_symbols_p.h
--- qt-everywhere-src-5.13.0-orig/qtbase/src/network/ssl/qsslsocket_opensslpre11_symbols_p.h	2019-06-13 13:18:18.000000000 +0900
+++ qt-everywhere-src-5.13.0/qtbase/src/network/ssl/qsslsocket_opensslpre11_symbols_p.h	2019-08-05 20:14:14.643085173 +0900
@@ -78,8 +78,8 @@
 unsigned char * q_ASN1_STRING_data(ASN1_STRING *a);
 BIO *q_BIO_new_file(const char *filename, const char *mode);
 void q_ERR_clear_error();
-Q_AUTOTEST_EXPORT BIO *q_BIO_new(BIO_METHOD *a);
-Q_AUTOTEST_EXPORT BIO_METHOD *q_BIO_s_mem();
+Q_AUTOTEST_EXPORT BIO *q_BIO_new(const BIO_METHOD *a);
+Q_AUTOTEST_EXPORT const BIO_METHOD *q_BIO_s_mem();
 int q_CRYPTO_num_locks();
 void q_CRYPTO_set_locking_callback(void (*a)(int, int, const char *, int));
 void q_CRYPTO_set_id_callback(unsigned long (*a)());
diff -ruN qt-everywhere-src-5.13.0-orig/qtbase/src/network/ssl/qsslsocket_openssl_symbols.cpp qt-everywhere-src-5.13.0/qtbase/src/network/ssl/qsslsocket_openssl_symbols.cpp
--- qt-everywhere-src-5.13.0-orig/qtbase/src/network/ssl/qsslsocket_openssl_symbols.cpp	2019-06-13 13:18:18.000000000 +0900
+++ qt-everywhere-src-5.13.0/qtbase/src/network/ssl/qsslsocket_openssl_symbols.cpp	2019-08-05 20:14:14.642085173 +0900
@@ -247,8 +247,8 @@
 #endif
 DEFINEFUNC2(BIO *, BIO_new_file, const char *filename, filename, const char *mode, mode, return nullptr, return)
 DEFINEFUNC(void, ERR_clear_error, DUMMYARG, DUMMYARG, return, DUMMYARG)
-DEFINEFUNC(BIO *, BIO_new, BIO_METHOD *a, a, return nullptr, return)
-DEFINEFUNC(BIO_METHOD *, BIO_s_mem, void, DUMMYARG, return nullptr, return)
+DEFINEFUNC(BIO *, BIO_new, const BIO_METHOD *a, a, return nullptr, return)
+DEFINEFUNC(const BIO_METHOD *, BIO_s_mem, void, DUMMYARG, return nullptr, return)
 DEFINEFUNC(int, CRYPTO_num_locks, DUMMYARG, DUMMYARG, return 0, return)
 DEFINEFUNC(void, CRYPTO_set_locking_callback, void (*a)(int, int, const char *, int), a, return, DUMMYARG)
 DEFINEFUNC(void, CRYPTO_set_id_callback, unsigned long (*a)(), a, return, DUMMYARG)
@@ -439,7 +439,7 @@
 DEFINEFUNC2(int, SSL_CTX_use_RSAPrivateKey, SSL_CTX *a, a, RSA *b, b, return -1, return)
 DEFINEFUNC3(int, SSL_CTX_use_PrivateKey_file, SSL_CTX *a, a, const char *b, b, int c, c, return -1, return)
 DEFINEFUNC(X509_STORE *, SSL_CTX_get_cert_store, const SSL_CTX *a, a, return nullptr, return)
-#if OPENSSL_VERSION_NUMBER >= 0x10002000L
+#if OPENSSL_VERSION_NUMBER >= 0x10002000L && !defined(LIBRESSL_VERSION_NUMBER)
 DEFINEFUNC(SSL_CONF_CTX *, SSL_CONF_CTX_new, DUMMYARG, DUMMYARG, return nullptr, return);
 DEFINEFUNC(void, SSL_CONF_CTX_free, SSL_CONF_CTX *a, a, return ,return);
 DEFINEFUNC2(void, SSL_CONF_CTX_set_ssl_ctx, SSL_CONF_CTX *a, a, SSL_CTX *b, b, return, return);
diff -ruN qt-everywhere-src-5.13.0-orig/qtbase/src/network/ssl/qsslsocket_openssl_symbols_p.h qt-everywhere-src-5.13.0/qtbase/src/network/ssl/qsslsocket_openssl_symbols_p.h
--- qt-everywhere-src-5.13.0-orig/qtbase/src/network/ssl/qsslsocket_openssl_symbols_p.h	2019-06-13 13:18:18.000000000 +0900
+++ qt-everywhere-src-5.13.0/qtbase/src/network/ssl/qsslsocket_openssl_symbols_p.h	2019-08-05 20:14:05.003085440 +0900
@@ -372,7 +372,7 @@
 int q_SSL_CTX_use_RSAPrivateKey(SSL_CTX *a, RSA *b);
 int q_SSL_CTX_use_PrivateKey_file(SSL_CTX *a, const char *b, int c);
 X509_STORE *q_SSL_CTX_get_cert_store(const SSL_CTX *a);
-#if OPENSSL_VERSION_NUMBER >= 0x10002000L
+#if OPENSSL_VERSION_NUMBER >= 0x10002000L && !defined(LIBRESSL_VERSION_NUMBER)
 SSL_CONF_CTX *q_SSL_CONF_CTX_new();
 void q_SSL_CONF_CTX_free(SSL_CONF_CTX *a);
 void q_SSL_CONF_CTX_set_ssl_ctx(SSL_CONF_CTX *a, SSL_CTX *b);
