From 5fdaa83819f07c89ecf043a3dd31f7d8d79b153d Mon Sep 17 00:00:00 2001
From: Ishimoto Shinobu <nagakamira@gmail.com>
Date: Sat, 17 Apr 2021 09:56:43 -0700
Subject: [PATCH 1/2] [ELF] use SHA1 by default

Signed-off-by: Ishimoto Shinobu <nagakamira@gmail.com>
---
 ELF/Driver.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/ELF/Driver.cpp b/ELF/Driver.cpp
index 3b2a8ba..151536e 100644
--- a/ELF/Driver.cpp
+++ b/ELF/Driver.cpp
@@ -770,7 +770,7 @@ getBuildId(opt::InputArgList &args) {
     return {BuildIdKind::None, {}};
 
   if (arg->getOption().getID() == OPT_build_id)
-    return {BuildIdKind::Fast, {}};
+    return {BuildIdKind::Sha1, {}};
 
   StringRef s = arg->getValue();
   if (s == "fast")
-- 
2.31.0

