From f2dcfbe87f6548e8cf1f9e42b94b7ea9f83bfc2d Mon Sep 17 00:00:00 2001
From: Ishimoto Shinobu <nagakamira@gmail.com>
Date: Wed, 10 Feb 2021 21:10:53 +0900
Subject: [PATCH 1/2] force utmp.h

Signed-off-by: Ishimoto Shinobu <nagakamira@gmail.com>
---
 w.c | 6 +-----
 1 file changed, 1 insertion(+), 5 deletions(-)

diff --git a/w.c b/w.c
index 9d07ac9..c1fda14 100644
--- a/w.c
+++ b/w.c
@@ -55,11 +55,7 @@
 #include <termios.h>
 #include <time.h>
 #include <unistd.h>
-#ifdef HAVE_UTMPX_H
-#	include <utmpx.h>
-#else
-#	include <utmp.h>
-#endif
+#include <utmp.h>
 #include <arpa/inet.h>
 
 static int ignoreuser = 0;	/* for '-u' */
-- 
2.29.2

