Patch by Ishimoto Shinobu

let's fix sys/cdefs.h bug!

diff -ruN quota-4.05/quota.h quota-4.05_b/quota.h
--- quota-4.05/quota.h	2019-03-29 10:39:33.000000000 +0000
+++ quota-4.05_b/quota.h	2019-05-01 16:11:18.697015843 +0000
@@ -1,7 +1,10 @@
 #ifndef GUARD_QUOTA_H
 #define GUARD_QUOTA_H
 
-#include <sys/cdefs.h>
+#ifdef __cplusplus
+extern "C" {
+#endif
+
 #include <sys/types.h>
 #include <stdint.h>
 
