From efcddf354e9060e8c0af8d342fbabd2fa994e9e5 Mon Sep 17 00:00:00 2001
From: Ishimoto Shinobu <nagakamira@gmail.com>
Date: Sat, 21 Nov 2020 15:29:17 +0300
Subject: [PATCH] no locale

Signed-off-by: Ishimoto Shinobu <nagakamira@gmail.com>
---
 Makefile.am | 6 +-----
 1 file changed, 1 insertion(+), 5 deletions(-)

diff --git a/Makefile.am b/Makefile.am
index 6c14a4a..13bd4db 100644
--- a/Makefile.am
+++ b/Makefile.am
@@ -6,11 +6,7 @@ SUBDIRS =					\
 	a11y					\
 	gtkhtml					\
 	components				\
-	art					\
-	po
-
-CLEANFILES =					\
-	po/.intltool-merge-cache
+	art
 
 MAINTAINERCLEANFILES =				\
 	$(srcdir)/INSTALL			\
-- 
2.28.0

