From 9aa3ab62687d710030bfa4344bdbaa8e7973cfd0 Mon Sep 17 00:00:00 2001
From: Ishimoto Shinobu <nagakamira@gmail.com>
Date: Thu, 30 Apr 2020 22:49:25 +0300
Subject: [PATCH] :^)

---
 meson.build | 8 --------
 1 file changed, 8 deletions(-)

diff --git a/meson.build b/meson.build
index 314e3a3f1..5dbe069ee 100644
--- a/meson.build
+++ b/meson.build
@@ -163,14 +163,6 @@ assert(zlib_dep.found() and cc.has_function('inflate', dependencies: zlib_dep) a
 
 ev_platform = get_option('platform')
 if ev_platform == 'gnome'
-  # Evince has a rather soft run-time dependency on hicolor-icon-theme.
-  # If the hicolor theme is not available, Evince fails to display some
-  # icons. Because we cannot check for it at run-time, we instead
-  # would like to require the icon theme at compile-time. But, because
-  # the hicolor-icon-theme does not have a pkgconfig file, on gnome we
-  # require the gnome icon theme instead.
-  adwaita_icon_theme_dep = dependency('adwaita-icon-theme', version: '>= 2.17.1')
-
   # *** Nautilus property page build ***
   enable_nautilus = get_option('nautilus')
   if enable_nautilus
