From 645aa45cedaca6748cbac117014792bec0fbc858 Mon Sep 17 00:00:00 2001
From: root <root@localhost.localdomain>
Date: Sun, 18 Oct 2020 21:48:41 +0000
Subject: [PATCH] like pro

Signed-off-by: root <root@localhost.localdomain>
---
 Makefile | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Makefile b/Makefile
index aed2da0..693d762 100644
--- a/Makefile
+++ b/Makefile
@@ -31,7 +31,7 @@ release: $(TARGET)
 
 install:
 	@echo "Installing... Make sure your binary has DATA_LOCATION set to a loopcube folder"
-	mkdir -p $(DATA_LOCATION)
+	install -dm755 $(DATA_LOCATION)
 	cp -r data $(DATA_LOCATION)/data
 	cp $(TARGET) $(PREFIX)
 
-- 
2.28.0

