From 9f1ee7abc7449ffa6e3ec388e71f57b9c8a333fa Mon Sep 17 00:00:00 2001
From: root <root@localhost.localdomain>
Date: Sun, 18 Oct 2020 21:50:35 +0000
Subject: [PATCH] omfg

Signed-off-by: root <root@localhost.localdomain>
---
 Makefile | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Makefile b/Makefile
index 693d762..a6ce0f0 100644
--- a/Makefile
+++ b/Makefile
@@ -33,7 +33,7 @@ install:
 	@echo "Installing... Make sure your binary has DATA_LOCATION set to a loopcube folder"
 	install -dm755 $(DATA_LOCATION)
 	cp -r data $(DATA_LOCATION)/data
-	cp $(TARGET) $(PREFIX)
+	install -Dm755 $(TARGET) $(PREFIX)/$(TARGET)
 
 uninstall:
 	@echo "Uninstalling... type yes if prompted (be careful)"
-- 
2.28.0

