diff -pruN mozc.orig/gyp/common.gypi mozc/gyp/common.gypi
--- mozc.orig/gyp/common.gypi	2017-02-06 11:22:06.000000000 +0900
+++ mozc/gyp/common.gypi	2017-02-06 15:12:33.972605000 +0900
@@ -128,9 +128,9 @@
         'compiler_host_version_int': 304,  # Clang 3.4 or higher
       }],
       ['target_platform=="Linux"', {
-        'compiler_target': 'clang',
+        'compiler_target': 'gcc',
         'compiler_target_version_int': 304,  # Clang 3.4 or higher
-        'compiler_host': 'clang',
+        'compiler_host': 'gcc',
         'compiler_host_version_int': 304,  # Clang 3.4 or higher
       }],
     ],
@@ -445,8 +445,8 @@
     ['target_platform=="Linux"', {
       'make_global_settings': [
         ['AR', '<!(which ar)'],
-        ['CC', '<!(which clang)'],
-        ['CXX', '<!(which clang++)'],
+        ['CC', '<!(which gcc)'],
+        ['CXX', '<!(which g++)'],
         ['LD', '<!(which ld)'],
         ['NM', '<!(which nm)'],
         ['READELF', '<!(which readelf)'],
diff -pruN mozc.orig/gyp/common.gypi mozc/gyp/common.gypi
--- mozc.orig/gyp/common.gypi	2017-02-06 15:12:33.972605000 +0900
+++ mozc/gyp/common.gypi	2017-02-07 21:23:50.018605000 +0900
@@ -106,6 +106,7 @@
     # Libraries for GNU/Linux environment.
     'linux_ldflags': [
       '-pthread',
+      '-Wl,--build-id',
     ],
 
     'conditions': [
diff -pruN mozc-2.23.2815.102.orig/prediction/zero_query_dict.h mozc-2.23.2815.102/prediction/zero_query_dict.h
--- mozc-2.23.2815.102.orig/prediction/zero_query_dict.h	2018-01-26 18:48:12.000000000 +0900
+++ mozc-2.23.2815.102/prediction/zero_query_dict.h	2018-06-01 16:08:24.548653648 +0900
@@ -147,6 +147,17 @@ class ZeroQueryDict {
       return iter;
     }
 
+    iterator &operator--() {
+      ptr_ -= kTokenByteSize;
+      return *this;
+    }
+
+    iterator operator--(int) {
+      const iterator tmp(ptr_, string_array_);
+      ptr_ -= kTokenByteSize;
+      return tmp;
+    }
+
     iterator &operator-=(ptrdiff_t n) {
       ptr_ -= n * kTokenByteSize;
       return *this;
