diff -Naur gcc-6.2.0.orig/gcc/config/i386/i386.c gcc-6.2.0/gcc/config/i386/i386.c
--- gcc-6.2.0.orig/gcc/config/i386/i386.c	2016-12-24 15:55:23.355343088 -0800
+++ gcc-6.2.0/gcc/config/i386/i386.c	2016-12-24 15:55:58.718681693 -0800
@@ -40325,10 +40325,10 @@
     {
     case IX86_BUILTIN_CPU_INIT:
       {
-	/* Make it call __cpu_indicator_init in libgcc. */
+	/* Make it call __cpu_indicator_init_local in libgcc. */
 	tree call_expr, fndecl, type;
         type = build_function_type_list (integer_type_node, NULL_TREE); 
-	fndecl = build_fn_decl ("__cpu_indicator_init", type);
+	fndecl = build_fn_decl ("__cpu_indicator_init_local", type);
 	call_expr = build_call_expr (fndecl, 0); 
 	return expand_expr (call_expr, target, mode, EXPAND_NORMAL);
       }
diff -Naur gcc-6.2.0.orig/libgcc/config/i386/cpuinfo.c gcc-6.2.0/libgcc/config/i386/cpuinfo.c
--- gcc-6.2.0.orig/libgcc/config/i386/cpuinfo.c	2016-12-24 15:55:15.817270911 -0800
+++ gcc-6.2.0/libgcc/config/i386/cpuinfo.c	2016-12-24 15:57:05.399320163 -0800
@@ -485,7 +485,7 @@
   return 0;
 }
 
-#if defined SHARED && defined USE_ELF_SYMVER
-__asm__ (".symver __cpu_indicator_init, __cpu_indicator_init@GCC_4.8.0");
-__asm__ (".symver __cpu_model, __cpu_model@GCC_4.8.0");
+#ifndef SHARED
+int __cpu_indicator_init_local (void)
+  __attribute__ ((weak, alias ("__cpu_indicator_init")));
 #endif
