From 44d5180478e5ba3213b1f8994b44843c84a82eb6 Mon Sep 17 00:00:00 2001
From: Ishimoto Shinobu <nagakamira@gmail.com>
Date: Thu, 11 Feb 2021 03:22:04 +0900
Subject: [PATCH] Allow building on non-GNU userland

Signed-off-by: Ishimoto Shinobu <nagakamira@gmail.com>
---
 data/meson-add-wants.sh | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/data/meson-add-wants.sh b/data/meson-add-wants.sh
index c33d1b4..c52d8a7 100755
--- a/data/meson-add-wants.sh
+++ b/data/meson-add-wants.sh
@@ -16,7 +16,7 @@ case "$target" in
         ;;
 esac
 
-unitpath="${DESTDIR:-}${unitdir}/${unit}"
+unitpath="${unitdir}/${unit}"
 
 case "$target" in
     */)
@@ -27,4 +27,4 @@ case "$target" in
         ;;
 esac
 
-ln -vfs --relative "$unitpath" "$dir"
+ln -vfs "$unitpath" "$dir"
-- 
2.29.2

