diff -ruN iproute2-4.19.0/misc/Makefile iproute2-4.19.0_b/misc/Makefile
--- iproute2-4.19.0/misc/Makefile	2018-10-24 02:14:57.000000000 +0900
+++ iproute2-4.19.0_b/misc/Makefile	2019-03-03 00:25:42.768111576 +0900
@@ -25,7 +25,7 @@
 	$(QUIET_CC)$(CC) $(CFLAGS) $(LDFLAGS) -o rtacct rtacct.c $(LDLIBS) -lm
 
 arpd: arpd.c
-	$(QUIET_CC)$(CC) $(CFLAGS) -I$(DBM_INCLUDE) $(LDFLAGS) -o arpd arpd.c $(LDLIBS) -ldb
+	echo "Gay"
 
 ssfilter.c: ssfilter.y
 	$(QUIET_YACC)bison ssfilter.y -o ssfilter.c
