From 7f267f4887ce0cee577b9e9205db32ce08d1f508 Mon Sep 17 00:00:00 2001
From: Ishimoto Shinobu <nagakamira@gmail.com>
Date: Sun, 27 Sep 2020 15:04:42 +0300
Subject: [PATCH] systemd paths

Signed-off-by: Ishimoto Shinobu <nagakamira@gmail.com>
---
 meson.build | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/meson.build b/meson.build
index 039269d..88d3228 100644
--- a/meson.build
+++ b/meson.build
@@ -70,9 +70,9 @@ if use_launcher
         dep_systemd = dependency('systemd', version: '>=230')
 
         add_project_arguments('-DSYSTEMUIDMAX=' + dep_systemd.get_pkgconfig_variable('systemuidmax'), language: 'c')
-        conf.set('systemunitdir', dep_systemd.get_pkgconfig_variable('systemdsystemunitdir'))
-        conf.set('userunitdir', dep_systemd.get_pkgconfig_variable('systemduserunitdir'))
-        conf.set('catalogdir', dep_systemd.get_pkgconfig_variable('catalogdir'))
+        conf.set('systemunitdir', '/usr/lib/systemd/system')
+        conf.set('userunitdir', '/usr/lib/systemd/user')
+        conf.set('catalogdir', '/usr/lib/systemd/catalog')
 endif
 
 #
-- 
2.26.2

