diff -ruN parted-3.2-orig/libparted/arch/linux.c parted-3.2/libparted/arch/linux.c
--- parted-3.2-orig/libparted/arch/linux.c	2014-06-15 12:15:54.000000000 -0700
+++ parted-3.2/libparted/arch/linux.c	2019-08-21 04:32:54.512172994 -0700
@@ -39,6 +39,7 @@
 #include <sys/stat.h>
 #include <sys/types.h>
 #include <sys/utsname.h>        /* for uname() */
+#include <sys/sysmacros.h>
 #include <scsi/scsi.h>
 #include <assert.h>
 #ifdef ENABLE_DEVICE_MAPPER
