From 7528d52f1ee6d34afbcba1e23b9863cbd97cf651 Mon Sep 17 00:00:00 2001
From: Thomas Deutschmann <whissi@gentoo.org>
Date: Thu, 1 Apr 2021 00:06:54 +0200
Subject: [PATCH 3/3] Fix docdir for Gentoo

Signed-off-by: Thomas Deutschmann <whissi@gentoo.org>
---
 Makefile.in | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Makefile.in b/Makefile.in
index 5a64799..a4c5962 100644
--- a/Makefile.in
+++ b/Makefile.in
@@ -95,7 +95,7 @@ mandir = @mandir@
 man1ext = 1
 man1dir = $(mandir)/man$(man1ext)
 datadir = @datadir@
-docdir = @docdir@@VERSIONED_PATH@
+docdir = @docdir@
 
 # The following must be substituted using @datadir@ and @libdir@
 # to avoid adding RPM generation paths (CUPS STR #1112)
-- 
2.31.1

