From 606f8c8005bfc0d9414b12b8d9e6d7b93679ab47 Mon Sep 17 00:00:00 2001
From: Ishimoto Shinobu <nagakamira@gmail.com>
Date: Thu, 11 Feb 2021 03:21:56 +0900
Subject: [PATCH] Allow building on non-GNU userland

Signed-off-by: Ishimoto Shinobu <nagakamira@gmail.com>
---
 plugins/meson-add-wants.sh | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/plugins/meson-add-wants.sh b/plugins/meson-add-wants.sh
index c33d1b4..c52d8a7 100755
--- a/plugins/meson-add-wants.sh
+++ b/plugins/meson-add-wants.sh
@@ -16,7 +16,7 @@ case "$target" in
         ;;
 esac
 
-unitpath="${DESTDIR:-}${unitdir}/${unit}"
+unitpath="${unitdir}/${unit}"
 
 case "$target" in
     */)
@@ -27,4 +27,4 @@ case "$target" in
         ;;
 esac
 
-ln -vfs --relative "$unitpath" "$dir"
+ln -vfs "$unitpath" "$dir"
-- 
2.29.2

