From 53ce140fa43857cf577b1aa6cf8c2ee70f545b96 Mon Sep 17 00:00:00 2001
From: Ishimoto Shinobu <nagakamira@gmail.com>
Date: Thu, 3 Sep 2020 02:03:02 +0900
Subject: [PATCH] Allow building on POSIX userland

Signed-off-by: Ishimoto Shinobu <nagakamira@gmail.com>
---
 plugins/meson-add-wants.sh | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/plugins/meson-add-wants.sh b/plugins/meson-add-wants.sh
index c33d1b4..47a2afa 100755
--- a/plugins/meson-add-wants.sh
+++ b/plugins/meson-add-wants.sh
@@ -16,15 +16,15 @@ case "$target" in
         ;;
 esac
 
-unitpath="${DESTDIR:-}${unitdir}/${unit}"
+unitpath="${unitdir}/${unit}"
 
 case "$target" in
     */)
-        mkdir -vp -m 0755 "$dir"
+        mkdir -p -m 0755 "$dir"
         ;;
     *)
-        mkdir -vp -m 0755 "$(dirname "$dir")"
+        mkdir -p -m 0755 "$(dirname "$dir")"
         ;;
 esac
 
-ln -vfs --relative "$unitpath" "$dir"
+ln -fs "$unitpath" "$dir"
-- 
2.18.4

