From ce233cca7100e7164ac8c310910a71601de62001 Mon Sep 17 00:00:00 2001
From: Ishimoto Shinobu <nagakamira@gmail.com>
Date: Sat, 8 Feb 2020 22:27:38 +0900
Subject: [PATCH] ataraxia distro upgrades

Signed-off-by: Ishimoto Shinobu <nagakamira@gmail.com>
---
 .../gs-plugin-fedora-pkgdb-collections.c      | 19 ++++++-------------
 1 file changed, 6 insertions(+), 13 deletions(-)

diff --git a/plugins/fedora-pkgdb-collections/gs-plugin-fedora-pkgdb-collections.c b/plugins/fedora-pkgdb-collections/gs-plugin-fedora-pkgdb-collections.c
index e41bbc8..2717925 100644
--- a/plugins/fedora-pkgdb-collections/gs-plugin-fedora-pkgdb-collections.c
+++ b/plugins/fedora-pkgdb-collections/gs-plugin-fedora-pkgdb-collections.c
@@ -12,7 +12,7 @@
 #include <json-glib/json-glib.h>
 #include <gnome-software.h>
 
-#define FEDORA_PKGDB_COLLECTIONS_API_URI "https://admin.fedoraproject.org/pkgdb/api/collections/"
+#define FEDORA_PKGDB_COLLECTIONS_API_URI "https://raw.githubusercontent.com/ataraxialinux/storage/master/ataraxia.json"
 
 struct GsPluginData {
 	gchar		*cachefn;
@@ -52,13 +52,6 @@ gs_plugin_initialize (GsPlugin *plugin)
 	GsPluginData *priv = gs_plugin_alloc_data (plugin, sizeof(GsPluginData));
 
 	g_mutex_init (&priv->mutex);
-
-	/* check that we are running on Fedora */
-	if (!gs_plugin_check_distro_id (plugin, "fedora")) {
-		gs_plugin_set_enabled (plugin, FALSE);
-		g_debug ("disabling '%s' as we're not Fedora", gs_plugin_get_name (plugin));
-		return;
-	}
 	priv->distros = g_ptr_array_new_with_free_func ((GDestroyNotify) _pkgdb_item_free);
 	priv->settings = g_settings_new ("org.gnome.software");
 
@@ -111,8 +104,8 @@ gs_plugin_setup (GsPlugin *plugin, GCancellable *cancellable, GError **error)
 	g_autoptr(GMutexLocker) locker = g_mutex_locker_new (&priv->mutex);
 
 	/* get the file to cache */
-	priv->cachefn = gs_utils_get_cache_filename ("fedora-pkgdb-collections",
-						     "fedora.json",
+	priv->cachefn = gs_utils_get_cache_filename ("ataraxia-pkgdb-collections",
+						     "ataraxia.json",
 						     GS_UTILS_CACHE_FLAG_WRITEABLE,
 						     error);
 	if (priv->cachefn == NULL)
@@ -264,13 +257,13 @@ _create_upgrade_from_info (GsPlugin *plugin, PkgdbItem *item)
 	if (app != NULL)
 		return app;
 
-	app_id = g_strdup_printf ("org.fedoraproject.Fedora-%u", item->version);
+	app_id = g_strdup_printf ("github.ataraxialinux.Ataraxia Linux-%u", item->version);
 	app_version = g_strdup_printf ("%u", item->version);
 
 	/* icon from disk */
 	ic = as_icon_new ();
 	as_icon_set_kind (ic, AS_ICON_KIND_LOCAL);
-	as_icon_set_filename (ic, "/usr/share/pixmaps/fedora-logo-sprite.png");
+	as_icon_set_filename (ic, "/usr/share/pixmaps/ataraxia-logo.png");
 
 	/* create */
 	app = gs_app_new (app_id);
@@ -291,7 +284,7 @@ _create_upgrade_from_info (GsPlugin *plugin, PkgdbItem *item)
 	gs_app_add_icon (app, ic);
 
 	/* show a Fedora magazine article for the release */
-	url = g_strdup_printf ("https://fedoramagazine.org/whats-new-fedora-%u-workstation",
+	url = g_strdup_printf ("https://github.com/ataraxialinux/ataraxia/releases/tag/%u",
 			       item->version);
 	gs_app_set_url (app, AS_URL_KIND_HOMEPAGE, url);
 
-- 
2.25.0

