ССАНИНА ЕБАНАЯ БЛЯТЬ МАМКУ АВТОРА В РОТ ЕБАЛ
diff -ruN audit-2.8.4/audisp/audispd.c audit-2.8.4_b/audisp/audispd.c
--- audit-2.8.4/audisp/audispd.c	2018-05-21 10:38:08.000000000 -0700
+++ audit-2.8.4_b/audisp/audispd.c	2019-01-11 07:34:55.505720758 -0800
@@ -31,7 +31,7 @@
 #include <pthread.h>
 #include <dirent.h>
 #include <fcntl.h>
-#include <sys/poll.h>
+#include <poll.h>
 #include <netdb.h>
 #include <arpa/inet.h>
 #include <limits.h>
diff -ruN audit-2.8.4/auparse/auparse.c audit-2.8.4_b/auparse/auparse.c
--- audit-2.8.4/auparse/auparse.c	2018-05-21 10:38:08.000000000 -0700
+++ audit-2.8.4_b/auparse/auparse.c	2019-01-11 07:36:36.583314202 -0800
@@ -1104,13 +1104,20 @@
 static int extract_timestamp(const char *b, au_event_t *e)
 {
 	char *ptr, *tmp;
+	char _tmp[341];
 	int rc = 1;
 
         e->host = NULL;
-	if (*b == 'n')
-		tmp = strndupa(b, 340);
-	else
-		tmp = strndupa(b, 80);
+	if (*b == 'n') {
+		strncpy(_tmp, b, 340);
+		_tmp[340] = '\0';
+		tmp = _tmp;
+	}
+	else {
+		strncpy(_tmp, b, 80);
+		_tmp[80] = '\0';
+		tmp = _tmp;
+	}
 	ptr = audit_strsplit(tmp);
 	if (ptr) {
 		// Optionally grab the node - may or may not be included
diff -ruN audit-2.8.4/auparse/interpret.c audit-2.8.4_b/auparse/interpret.c
--- audit-2.8.4/auparse/interpret.c	2018-05-21 10:38:08.000000000 -0700
+++ audit-2.8.4_b/auparse/interpret.c	2019-01-11 07:40:18.861481292 -0800
@@ -863,7 +863,7 @@
 		// Proctitle has arguments separated by NUL bytes
 		// We need to write over the NUL bytes with a space
 		// so that we can see the arguments
-		while ((ptr  = rawmemchr(ptr, '\0'))) {
+		while ((ptr  = memchr(ptr, '\0', SIZE_MAX))) {
 			if (ptr >= end)
 				break;
 			*ptr = ' ';
diff -ruN audit-2.8.4/lib/libaudit.c audit-2.8.4_b/lib/libaudit.c
--- audit-2.8.4/lib/libaudit.c	2018-06-19 11:16:43.000000000 -0700
+++ audit-2.8.4_b/lib/libaudit.c	2019-01-11 07:41:07.733674457 -0800
@@ -33,7 +33,7 @@
 #include <pwd.h>
 #include <grp.h>
 #include <errno.h>
-#include <sys/poll.h>
+#include <poll.h>
 #include <sys/utsname.h>
 #include <sys/stat.h>
 #include <fcntl.h>	/* O_NOFOLLOW needs gnu defined */
diff -ruN audit-2.8.4/lib/netlink.c audit-2.8.4_b/lib/netlink.c
--- audit-2.8.4/lib/netlink.c	2018-05-21 10:38:08.000000000 -0700
+++ audit-2.8.4_b/lib/netlink.c	2019-01-11 07:40:55.943949719 -0800
@@ -27,7 +27,7 @@
 #include <errno.h>
 #include <fcntl.h>
 #include <time.h>
-#include <sys/poll.h>
+#include <poll.h>
 #include "libaudit.h"
 #include "private.h"
 
diff -ruN audit-2.8.4/src/auditctl.c audit-2.8.4_b/src/auditctl.c
--- audit-2.8.4/src/auditctl.c	2018-06-19 11:16:43.000000000 -0700
+++ audit-2.8.4_b/src/auditctl.c	2019-01-11 07:41:20.273381545 -0800
@@ -37,6 +37,7 @@
 #include <sys/time.h>
 #include <fcntl.h>
 #include <errno.h>
+#include <sys/select.h>
 #include <libgen.h>	/* For basename */
 #include <limits.h>	/* PATH_MAX */
 #include "libaudit.h"
diff -ruN audit-2.8.4/src/auditd.c audit-2.8.4_b/src/auditd.c
--- audit-2.8.4/src/auditd.c	2018-05-21 10:38:08.000000000 -0700
+++ audit-2.8.4_b/src/auditd.c	2019-01-11 07:47:35.261236486 -0800
@@ -30,6 +30,7 @@
 #include <errno.h>
 #include <string.h>
 #include <time.h>
+#include <alloca.h>
 #include <sys/resource.h>
 #include <sys/time.h>
 #include <sys/stat.h>
@@ -209,7 +210,7 @@
 
 static int extract_type(const char *str)
 {
-	const char *tptr, *ptr2, *ptr = str;
+	const char *ptr2, *ptr = str;
 	if (*str == 'n') {
 		ptr = strchr(str+1, ' ');
 		if (ptr == NULL)
@@ -218,11 +219,10 @@
 	}
 	// ptr should be at 't'
 	ptr2 = strchr(ptr, ' ');
-	// get type=xxx in a buffer
-	tptr = strndupa(ptr, ptr2 - ptr);
+
 	// find =
-	str = strchr(tptr, '=');
-	if (str == NULL)
+	str = strchr(ptr, '=');
+	if (str == NULL || str >= ptr2)
 		return -1; // Malformed - bomb out
 	// name is 1 past
 	str++;
diff -ruN audit-2.8.4/src/ausearch-lol.c audit-2.8.4_b/src/ausearch-lol.c
--- audit-2.8.4/src/ausearch-lol.c	2018-06-18 07:56:52.000000000 -0700
+++ audit-2.8.4_b/src/ausearch-lol.c	2019-01-11 07:38:39.643788185 -0800
@@ -137,13 +137,19 @@
 static int extract_timestamp(const char *b, event *e)
 {
 	char *ptr, *tmp, *tnode, *ttype;
+	char _tmp[341];
 
 	e->node = NULL;
-	if (*b == 'n')
-		tmp = strndupa(b, 340);
-	else
-		tmp = strndupa(b, 80);
-	ptr = audit_strsplit(tmp);
+	if (*b == 'n') {
+		strncpy(_tmp, b, 340);
+		_tmp[340] = '\0';
+		tmp = _tmp;
+	}
+	else {
+		strncpy(_tmp, b, 80);
+		_tmp[80] = '\0';
+		tmp = _tmp;
+	}
 	if (ptr) {
 		// Check to see if this is the node info
 		if (*ptr == 'n') {
