From 4ec53a33240655a57b4c12d7495473e2cccdde40 Mon Sep 17 00:00:00 2001
From: Ishimoto Shinobu <nagakamira@gmail.com>
Date: Sun, 6 Oct 2019 21:28:23 +0900
Subject: [PATCH] try

Signed-off-by: Ishimoto Shinobu <nagakamira@gmail.com>
---
 libgdm/gdm-user-switching.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/libgdm/gdm-user-switching.c b/libgdm/gdm-user-switching.c
index 41ce285..f951aa6 100644
--- a/libgdm/gdm-user-switching.c
+++ b/libgdm/gdm-user-switching.c
@@ -159,7 +159,7 @@ get_current_seat_id (GDBusConnection *connection)
 
         return seat_id;
 }
-
+/*
 gboolean
 activate_session_id_for_ck (GDBusConnection *connection,
                             GCancellable    *cancellable,
@@ -188,6 +188,7 @@ activate_session_id_for_ck (GDBusConnection *connection,
 
         return TRUE;
 }
+*/
 
 static gboolean
 session_is_login_window (GDBusConnection *connection,
-- 
2.23.0

