From dc957f49c84789e981ce4e1b6b57cf38a411c382 Mon Sep 17 00:00:00 2001
From: Ishimoto Shinobu <nagakamira@gmail.com>
Date: Mon, 2 Dec 2019 21:18:21 +0900
Subject: [PATCH] gnome-session.in tweaks

Signed-off-by: Ishimoto Shinobu <nagakamira@gmail.com>
---
 gnome-session/gnome-session.in | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/gnome-session/gnome-session.in b/gnome-session/gnome-session.in
index 7ded359..d4e99a5 100755
--- a/gnome-session/gnome-session.in
+++ b/gnome-session/gnome-session.in
@@ -25,4 +25,8 @@ if [ -n "$REGION" ]; then
   export LC_PAPER=$REGION
 fi
 
+if [ -f /etc/xdg/menus/gnome-applications.menu ] ; then
+  export XDG_MENU_PREFIX=gnome-
+fi
+
 exec @libexecdir@/gnome-session-binary "$@"
-- 
2.24.0

