From acd94e6e8401fb3a0525936f8957585a8bd5431a Mon Sep 17 00:00:00 2001
From: Ishimoto Shinobu <nagakamira@gmail.com>
Date: Wed, 15 Apr 2020 21:53:35 +0900
Subject: [PATCH] ataraxia locales

Signed-off-by: Ishimoto Shinobu <nagakamira@gmail.com>
---
 libgnome-desktop/gnome-languages.c | 26 +++++++++++++++++++++++++-
 1 file changed, 25 insertions(+), 1 deletion(-)

diff --git a/libgnome-desktop/gnome-languages.c b/libgnome-desktop/gnome-languages.c
index e27ccbd2..54e03811 100644
--- a/libgnome-desktop/gnome-languages.c
+++ b/libgnome-desktop/gnome-languages.c
@@ -485,6 +485,23 @@ select_dirs (const struct dirent *dirent)
         return result;
 }
 
+#if !defined(__GLIBC__)
+static gboolean
+collect_locales_builtin (void)
+{
+       char locale_arr[][13] = { "en_US.UTF-8", "et_EE.UTF-8", "ru_RU.UTF-8", };
+
+	gboolean found_locales = TRUE;
+
+	int i;
+	for (i = 0; i < sizeof(locale_arr)/sizeof(locale_arr[0]); i++) {
+		add_locale(locale_arr[i], TRUE);
+	}
+
+	return found_locales;
+}
+#endif
+
 static gboolean
 collect_locales_from_directory (void)
 {
@@ -582,7 +599,14 @@ collect_locales (void)
 
         found_dir_locales = collect_locales_from_directory ();
 
-        if (!(found_localebin_locales || found_dir_locales)) {
+#if defined(__GLIBC)
+         if (!(found_localebin_locales || found_dir_locales)) {
+#else
+        gboolean found_locales_builtin = FALSE;
+        found_locales_builtin = collect_locales_builtin ();
+
+        if (!(found_locales_builtin || found_localebin_locales || found_dir_locales)) {
+#endif
                 g_warning ("Could not read list of available locales from libc, "
                            "guessing possible locales from available translations, "
                            "but list may be incomplete!");
-- 
2.25.1

