From c9ebfd9c7160672229b91ce591990998243e579f Mon Sep 17 00:00:00 2001
From: Ishimoto Shinobu <nagakamira@gmail.com>
Date: Sat, 21 Nov 2020 19:34:21 +0300
Subject: [PATCH] no locales

Signed-off-by: Ishimoto Shinobu <nagakamira@gmail.com>
---
 Makefile.in | 1 -
 1 file changed, 1 deletion(-)

diff --git a/Makefile.in b/Makefile.in
index a02b318..fa1046a 100644
--- a/Makefile.in
+++ b/Makefile.in
@@ -64,7 +64,6 @@ FILE_PERM=0644
 
 CC=@CC@
 PKG_CONFIG=@PKG_CONFIG@
-MSGFMT_PATH=@MSGFMT_PATH@
 
 # Check if -fstack-protector-strong is supported before enabling it
 STACK_PROTECTOR_UNSUPPORTED = $(shell $(CC) -fstack-protector-strong 2>&1 | grep -c 'stack-protector-strong')
-- 
2.28.0

