diff -ruN perp-2.07-orig/conf.mk perp-2.07/conf.mk
--- perp-2.07-orig/conf.mk	2013-01-12 00:10:32.000000000 +0900
+++ perp-2.07/conf.mk	2019-11-11 01:05:07.904664246 +0900
@@ -5,15 +5,15 @@
 
 ## build configuration, standard gcc + libc:
 CC = gcc
-CFLAGS = -Wall -Wextra -Wshadow -DNDEBUG -O2
-#CFLAGS = -Wall -Wextra -Wshadow -O2
+#CFLAGS = -Wall -Wextra -Wshadow -DNDEBUG -O2
+CFLAGS = -Wall -Wextra -Wshadow -O2
 
 ## build configuration, dietlibc:
 #CC = diet -Os gcc
 #CFLAGS = -Wall -Wextra -Wshadow -DNDEBUG
 
 ## strip configuration
-STRIP = strip
+STRIP ?= strip
 #STRIP = /some/other/stripper
 
 ## install configuration:
diff -ruN perp-2.07-orig/lasagna/Makefile perp-2.07/lasagna/Makefile
--- perp-2.07-orig/lasagna/Makefile	2013-01-12 00:10:32.000000000 +0900
+++ perp-2.07/lasagna/Makefile	2019-11-10 03:15:24.899805073 +0900
@@ -873,7 +873,7 @@
 ## special builds for outvec:
 ## build generator for outvec_STDOUT.h:
 outvec/mk_outvec : outvec/mk_outvec.c
-	$(CC) $(CFLAGS) -o $@ outvec/mk_outvec.c
+	cc $(CFLAGS) -o $@ outvec/mk_outvec.c
 
 ## generate outvec_STDOUT.h:
 outvec/outvec_STDOUT.h : outvec/mk_outvec
@@ -1007,7 +1007,7 @@
 ## special builds for rlimit:
 ## mk_rlimit (rlimit_defs generator):
 rlimit/mk_rlimit : rlimit/mk_rlimit.c
-	$(CC) $(CFLAGS) -o $@ rlimit/mk_rlimit.c
+	cc $(CFLAGS) -o $@ rlimit/mk_rlimit.c
 
 ## generate rlimit_defs table:
 rlimit/rlimit_defs.c.in : rlimit/mk_rlimit
@@ -1061,7 +1061,7 @@
 ## special builds for sysstr:
 ## mysysstr sysno table generator:
 sysstr/mksysstr : sysstr/mksysstr.c
-	$(CC) $(CFLAGS) -o $@ sysstr/mksysstr.c
+	cc $(CFLAGS) -o $@ sysstr/mksysstr.c
 
 ## generate errno table:
 sysstr/sysstr_errno.c.in : sysstr/mksysstr sysstr.h
