From ea7e3a2dd036ba7218677b6a88d4e4251790657b Mon Sep 17 00:00:00 2001
From: Ishimoto Shinobu <nagakamira@gmail.com>
Date: Sat, 27 Feb 2021 02:46:02 +0900
Subject: [PATCH] force systemd user unit dir

Signed-off-by: Ishimoto Shinobu <nagakamira@gmail.com>
---
 meson.build | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/meson.build b/meson.build
index 0645a17..e80ed8e 100644
--- a/meson.build
+++ b/meson.build
@@ -374,7 +374,7 @@ libsystemd_deps = []
 libsystemd = dependency('libsystemd', required: get_option('systemd'))
 systemd = dependency('systemd', required: get_option('systemd'))
 if libsystemd.found() and systemd.found()
-  systemduserunitdir = systemd.get_pkgconfig_variable('systemduserunitdir')
+  systemduserunitdir = '/usr/lib/systemd/user'
   conf.set('WITH_SYSTEMD', 1)
   libsystemd_deps += libsystemd
   with_systemd = true
-- 
2.29.2

