From c03bfe371751ba894aa052b84fd5e4257cd57ed9 Mon Sep 17 00:00:00 2001
From: Ishimoto Shinobu <nagakamira@gmail.com>
Date: Sat, 3 Apr 2021 13:38:44 -0700
Subject: [PATCH] fix rpl_realloc

Signed-off-by: Ishimoto Shinobu <nagakamira@gmail.com>
---
 configure.ac | 2 --
 1 file changed, 2 deletions(-)

diff --git a/configure.ac b/configure.ac
index daab982..2914709 100644
--- a/configure.ac
+++ b/configure.ac
@@ -89,8 +89,6 @@ AC_TYPE_UINT32_T
 AC_TYPE_UINT8_T
 
 # Checks for library functions.
-AC_FUNC_MALLOC
-AC_FUNC_REALLOC
 AC_FUNC_STAT
 AC_CHECK_FUNCS([memset strcasecmp strdup strerror strrchr strtoul strtoull timegm])
 
-- 
2.31.0

