From 8e1062316bbb8b127af66fae7d35b067a3ab4c9a Mon Sep 17 00:00:00 2001
From: Ishimoto Shinobu <nagakamira@gmail.com>
Date: Mon, 7 Sep 2020 00:46:52 +0900
Subject: [PATCH] Use musl targets

Signed-off-by: Ishimoto Shinobu <nagakamira@gmail.com>
---
 build/config/compiler/BUILD.gn | 12 ++++++------
 1 file changed, 6 insertions(+), 6 deletions(-)

diff --git a/build/config/compiler/BUILD.gn b/build/config/compiler/BUILD.gn
index 8166e38ce..d98a911a6 100644
--- a/build/config/compiler/BUILD.gn
+++ b/build/config/compiler/BUILD.gn
@@ -761,8 +761,8 @@ config("compiler_cpu_abi") {
       }
     } else if (current_cpu == "arm") {
       if (is_clang && !is_android && !is_nacl) {
-        cflags += [ "--target=arm-linux-gnueabihf" ]
-        ldflags += [ "--target=arm-linux-gnueabihf" ]
+        cflags += [ "--target=arm-linux-musleabihf" ]
+        ldflags += [ "--target=arm-linux-musleabihf" ]
       }
       if (!is_nacl) {
         cflags += [
@@ -775,8 +775,8 @@ config("compiler_cpu_abi") {
       }
     } else if (current_cpu == "arm64") {
       if (is_clang && !is_android && !is_nacl && !is_fuchsia) {
-        cflags += [ "--target=aarch64-linux-gnu" ]
-        ldflags += [ "--target=aarch64-linux-gnu" ]
+        cflags += [ "--target=aarch64-linux-musl" ]
+        ldflags += [ "--target=aarch64-linux-musl" ]
       }
     } else if (current_cpu == "mipsel" && !is_nacl) {
       ldflags += [ "-Wl,--hash-style=sysv" ]
@@ -1029,12 +1029,12 @@ config("compiler_cpu_abi") {
           "arm-nonsfi",
           "-mfloat-abi=hard",
           "--pnacl-bias=arm-nonsfi",
-          "--target=armv7-unknown-nacl-gnueabihf",
+          "--target=armv7-unknown-nacl-musleabihf",
         ]
         ldflags += [
           "-arch",
           "arm-nonsfi",
-          "--target=armv7-unknown-nacl-gnueabihf",
+          "--target=armv7-unknown-nacl-musleabihf",
         ]
       }
     } else if (current_cpu == "ppc64") {
-- 
2.18.4

