From 0fdfcfad2bce9de7061b21797bfca2b5ea942399 Mon Sep 17 00:00:00 2001
From: Ishimoto Shinobu <nagakamira@gmail.com>
Date: Wed, 6 May 2020 19:43:17 +0300
Subject: [PATCH] fix slc

Signed-off-by: Ishimoto Shinobu <nagakamira@gmail.com>
---
 lib/hx509/Makefile.am | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/lib/hx509/Makefile.am b/lib/hx509/Makefile.am
index decf971..e16342f 100644
--- a/lib/hx509/Makefile.am
+++ b/lib/hx509/Makefile.am
@@ -95,7 +95,7 @@ $(srcdir)/hx509-private.h: $(dist_libhx509_la_SOURCES)
 
 bin_PROGRAMS = hxtool
 
-hxtool-commands.c hxtool-commands.h: hxtool-commands.in $(SLC)
+hxtool-commands.c hxtool-commands.h: hxtool-commands.in
 	$(heim_verbose)$(SLC) $(srcdir)/hxtool-commands.in
 
 dist_hxtool_SOURCES = hxtool.c
-- 
2.26.2

