diff -ruN gsettings-desktop-schemas-3.34.0-orig/schemas/org.gnome.desktop.background.gschema.xml.in gsettings-desktop-schemas-3.34.0/schemas/org.gnome.desktop.background.gschema.xml.in
--- gsettings-desktop-schemas-3.34.0-orig/schemas/org.gnome.desktop.background.gschema.xml.in	2019-09-10 06:16:27.628955100 +0900
+++ gsettings-desktop-schemas-3.34.0/schemas/org.gnome.desktop.background.gschema.xml.in	2019-09-15 00:00:47.691133687 +0900
@@ -11,7 +11,7 @@
       </description>
     </key>
     <key name="picture-uri" type="s">
-      <default>'file://@datadir@/backgrounds/gnome/adwaita-timed.xml'</default>
+      <default>'file://@datadir@/backgrounds/gnome/default.jpg'</default>
       <summary>Picture URI</summary>
       <description>
         URI to use for the background image. Note that the backend only supports
diff -ruN gsettings-desktop-schemas-3.34.0-orig/schemas/org.gnome.desktop.screensaver.gschema.xml.in gsettings-desktop-schemas-3.34.0/schemas/org.gnome.desktop.screensaver.gschema.xml.in
--- gsettings-desktop-schemas-3.34.0-orig/schemas/org.gnome.desktop.screensaver.gschema.xml.in	2019-09-10 06:16:27.629955000 +0900
+++ gsettings-desktop-schemas-3.34.0/schemas/org.gnome.desktop.screensaver.gschema.xml.in	2019-09-15 00:00:52.107035536 +0900
@@ -95,7 +95,7 @@
       </description>
     </key>
     <key name="picture-uri" type="s">
-      <default>'file://@datadir@/backgrounds/gnome/adwaita-timed.xml'</default>
+      <default>'file://@datadir@/backgrounds/gnome/default.jpg'</default>
       <summary>Picture URI</summary>
       <description>
         URI to use for the background image. Note that the backend only supports
