diff -ruN weston-6.0.1-orig/libweston/meson.build weston-6.0.1/libweston/meson.build
--- weston-6.0.1-orig/libweston/meson.build	2019-06-25 00:46:26.000000000 +0900
+++ weston-6.0.1/libweston/meson.build	2019-07-31 21:09:33.179734345 +0900
@@ -478,16 +478,10 @@
 endif
 
 if get_option('weston-launch')
-	dep_pam = cc.find_library('pam')
-
-	if not cc.has_function('pam_open_session', dependencies: dep_pam)
-		error('pam_open_session not found for weston-launch')
-	endif
-
 	executable(
 		'weston-launch',
 		'weston-launch.c',
-		dependencies: [dep_pam, systemd_dep, dep_libdrm],
+		dependencies: [systemd_dep, dep_libdrm],
 		include_directories: include_directories('..'),
 		install: true
 	)
