From c7e44f67aa317c83fa461060dc6dbd41eb8b5ffc Mon Sep 17 00:00:00 2001
From: Ishimoto Shinobu <nagakamira@gmail.com>
Date: Fri, 17 Jan 2020 11:45:07 -0600
Subject: [PATCH] revert "lib: don't set loaded state until seat is fetched"

Signed-off-by: Ishimoto Shinobu <nagakamira@gmail.com>
---
 src/libaccountsservice/act-user-manager.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/libaccountsservice/act-user-manager.c b/src/libaccountsservice/act-user-manager.c
index 93d4423..495cd20 100644
--- a/src/libaccountsservice/act-user-manager.c
+++ b/src/libaccountsservice/act-user-manager.c
@@ -2547,7 +2547,7 @@ maybe_set_is_loaded (ActUserManager *manager)
         /* Don't set is_loaded yet unless the seat is already loaded enough
          * or failed to load.
          */
-        if (priv->seat.state > ACT_USER_MANAGER_SEAT_STATE_GET_ID) {
+        if (priv->seat.state >= ACT_USER_MANAGER_SEAT_STATE_GET_ID) {
                 g_debug ("ActUserManager: Seat loaded, so now setting loaded property");
         } else if (priv->seat.state == ACT_USER_MANAGER_SEAT_STATE_UNLOADED) {
                 g_debug ("ActUserManager: Seat wouldn't load, so giving up on it and setting loaded property");
-- 
2.18.2

