diff -ruN accountsservice-0.6.55-orig/data/meson.build accountsservice-0.6.55/data/meson.build
--- accountsservice-0.6.55-orig/data/meson.build	2019-04-24 00:16:09.000000000 +0900
+++ accountsservice-0.6.55/data/meson.build	2019-12-21 20:40:01.631065243 +0900
@@ -26,17 +26,6 @@
   install_dir: dbus_sys_dir,
 )
 
-policy = act_namespace.to_lower() + '.policy'
-
-i18n.merge_file(
-  policy,
-  input: policy + '.in',
-  output: policy,
-  po_dir: po_dir,
-  install: true,
-  install_dir: policy_dir,
-)
-
 if install_systemd_unit_dir
   service = 'accounts-daemon.service'
 
