--- accountsservice-0.6.54/output/build.ninja.old	2019-01-20 21:23:36.570000000 +0000
+++ accountsservice-0.6.54/output/build.ninja	2019-01-20 21:25:01.140000000 +0000
@@ -60,9 +60,9 @@
 
 # Build rules for targets
 
-build data/org.freedesktop.accounts.policy: CUSTOM_COMMAND ../data/org.freedesktop.accounts.policy.in
- COMMAND = /usr/bin/meson --internal msgfmthelper ../data/org.freedesktop.accounts.policy.in data/org.freedesktop.accounts.policy xml /usr/src/packages/user/accountsservice/src/accountsservice-0.6.55/po
- description = Generating$ org.freedesktop.accounts.policy_data_merge$ with$ a$ custom$ command.
+#build data/org.freedesktop.accounts.policy: CUSTOM_COMMAND ../data/org.freedesktop.accounts.policy.in
+# COMMAND = /usr/bin/meson --internal msgfmthelper ../data/org.freedesktop.accounts.policy.in data/org.freedesktop.accounts.policy xml /usr/src/packages/user/accountsservice/src/accountsservice-0.6.55/po
+# description = Generating$ org.freedesktop.accounts.policy_data_merge$ with$ a$ custom$ command.
 
 build src/accounts-generated.c: CUSTOM_COMMAND /usr/src/packages/user/accountsservice/src/accountsservice-0.6.55/data/org.freedesktop.Accounts.xml | /usr/bin/gdbus-codegen
  COMMAND = /usr/bin/gdbus-codegen --c-generate-autocleanup all --interface-prefix org.freedesktop. --c-namespace Accounts --body --output src/accounts-generated.c /usr/src/packages/user/accountsservice/src/accountsservice-0.6.55/data/org.freedesktop.Accounts.xml

