From d773988a962f4b84661729caaa225ea178e6a10b Mon Sep 17 00:00:00 2001
From: Ishimoto Shinobu <nagakamira@gmail.com>
Date: Tue, 24 Mar 2020 02:13:34 +0900
Subject: [PATCH] sysctl

Signed-off-by: Ishimoto Shinobu <nagakamira@gmail.com>
---
 init.d/sysctl.in | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/init.d/sysctl.in b/init.d/sysctl.in
index e49f4db2..895cc1f2 100644
--- a/init.d/sysctl.in
+++ b/init.d/sysctl.in
@@ -42,7 +42,7 @@ Linux_sysctl()
 	local quiet
 	yesno $rc_verbose || quiet=-q
 
-	sysctl ${quiet} --system
+	sysctl -p ${quiet} /etc/sysctl.d/*.conf
 }
 
 start()
-- 
2.24.1

