From c3f789aa8b7bf7b804dde56cb70a023f66d8dd6a Mon Sep 17 00:00:00 2001
From: Ishimoto Shinobu <nagakamira@gmail.com>
Date: Fri, 8 May 2020 15:00:25 +0000
Subject: [PATCH] gcc bruh

Signed-off-by: Ishimoto Shinobu <nagakamira@gmail.com>
---
 libiberty/configure  | 5 +----
 lto-plugin/configure | 5 +----
 2 files changed, 2 insertions(+), 8 deletions(-)

diff --git a/libiberty/configure b/libiberty/configure
index bb76cf1b8..a2a6e6cf1 100755
--- a/libiberty/configure
+++ b/libiberty/configure
@@ -5376,10 +5376,7 @@ fi
 
 if test x$may_have_cet = xyes; then
   if test "$cross_compiling" = yes; then :
-  { { $as_echo "$as_me:${as_lineno-$LINENO}: error: in \`$ac_pwd':" >&5
-$as_echo "$as_me: error: in \`$ac_pwd':" >&2;}
-as_fn_error $? "cannot run test program while cross compiling
-See \`config.log' for more details" "$LINENO" 5; }
+  echo "bruh"
 else
   cat confdefs.h - <<_ACEOF >conftest.$ac_ext
 /* end confdefs.h.  */
diff --git a/lto-plugin/configure b/lto-plugin/configure
index 1baf6cc70..5d36c751e 100755
--- a/lto-plugin/configure
+++ b/lto-plugin/configure
@@ -5769,10 +5769,7 @@ fi
 
 if test x$may_have_cet = xyes; then
   if test "$cross_compiling" = yes; then :
-  { { $as_echo "$as_me:${as_lineno-$LINENO}: error: in \`$ac_pwd':" >&5
-$as_echo "$as_me: error: in \`$ac_pwd':" >&2;}
-as_fn_error $? "cannot run test program while cross compiling
-See \`config.log' for more details" "$LINENO" 5; }
+  echo "bruh"
 else
   cat confdefs.h - <<_ACEOF >conftest.$ac_ext
 /* end confdefs.h.  */
-- 
2.26.2

