From d4cfe5e51f58d0076d326251e6acec5ea66162ea Mon Sep 17 00:00:00 2001
From: Ishimoto Shinobu <nagakamira@gmail.com>
Date: Thu, 13 Jun 2019 00:31:25 +0900
Subject: [PATCH 03/11] Define musl ldso for s390

Signed-off-by: Ishimoto Shinobu <nagakamira@gmail.com>
---
 gcc/config/s390/linux.h | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/gcc/config/s390/linux.h b/gcc/config/s390/linux.h
index f76986acd..5f6b7041f 100644
--- a/gcc/config/s390/linux.h
+++ b/gcc/config/s390/linux.h
@@ -75,6 +75,9 @@ along with GCC; see the file COPYING3.  If not see
 #define GLIBC_DYNAMIC_LINKER32 "/lib/ld.so.1"
 #define GLIBC_DYNAMIC_LINKER64 "/lib/ld64.so.1"
 
+#define MUSL_DYNAMIC_LINKER32 "/lib/ld-musl-s390.so.1"
+#define MUSL_DYNAMIC_LINKER64 "/lib/ld-musl-s390x.so.1"
+
 #undef  LINK_SPEC
 #define LINK_SPEC \
   "%{m31:-m elf_s390}%{m64:-m elf64_s390} \
-- 
2.20.1

