From d8ef3bc2df4899f0e88fc585b3059b0eb71ccd7c Mon Sep 17 00:00:00 2001
From: Ishimoto Shinobu <nagakamira@gmail.com>
Date: Thu, 7 May 2020 17:38:26 +0300
Subject: [PATCH 06/11] support for fortify-headers on musl

Signed-off-by: Ishimoto Shinobu <nagakamira@gmail.com>
---
 gcc/config/linux.h | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/gcc/config/linux.h b/gcc/config/linux.h
index 95654bcdb..d88df8f15 100644
--- a/gcc/config/linux.h
+++ b/gcc/config/linux.h
@@ -167,6 +167,8 @@ see the files COPYING3 and COPYING.RUNTIME respectively.  If not, see
 
 #ifdef NATIVE_SYSTEM_HEADER_DIR
 #define INCLUDE_DEFAULTS_MUSL_NATIVE			\
+    { NATIVE_SYSTEM_HEADER_DIR "/fortify", 0, 0, 0, 1, 2 },	\
+    { NATIVE_SYSTEM_HEADER_DIR "/fortify", 0, 0, 0, 1, 0 },	\
     { NATIVE_SYSTEM_HEADER_DIR, 0, 0, 0, 1, 2 },	\
     { NATIVE_SYSTEM_HEADER_DIR, 0, 0, 0, 1, 0 },
 #else
-- 
2.26.2

