diff -ruN gcc-9.1.0-orig/gcc/c-family/c-cppbuiltin.c gcc-9.1.0/gcc/c-family/c-cppbuiltin.c
--- gcc-9.1.0-orig/gcc/c-family/c-cppbuiltin.c	2019-01-15 05:12:11.000000000 +0900
+++ gcc-9.1.0/gcc/c-family/c-cppbuiltin.c	2019-05-04 01:09:48.037366817 +0900
@@ -1357,6 +1357,10 @@
   builtin_define_with_value ("__REGISTER_PREFIX__", REGISTER_PREFIX, 0);
   builtin_define_with_value ("__USER_LABEL_PREFIX__", user_label_prefix, 0);
 
+  /* Fortify Source enabled by default for optimization levels > 0 */
+  if (optimize)
+    builtin_define_with_int_value ("_FORTIFY_SOURCE", 2);
+
   /* Misc.  */
   if (flag_gnu89_inline)
     cpp_define (pfile, "__GNUC_GNU_INLINE__");
diff -ruN gcc-9.1.0-orig/gcc/c-family/c.opt gcc-9.1.0/gcc/c-family/c.opt
--- gcc-9.1.0-orig/gcc/c-family/c.opt	2019-04-13 01:12:49.000000000 +0900
+++ gcc-9.1.0/gcc/c-family/c.opt	2019-05-04 01:24:57.237421032 +0900
@@ -576,7 +576,7 @@
 of the destination region.  Same as -Wformat-overflow=1.
 
 Wformat-security
-C ObjC C++ ObjC++ Var(warn_format_security) Warning LangEnabledBy(C ObjC C++ ObjC++,Wformat=, warn_format >= 2, 0)
+C ObjC C++ ObjC++ Var(warn_format_security) Init(1) Warning LangEnabledBy(C ObjC C++ ObjC++,Wformat=, warn_format >= 2, 0)
 Warn about possible security problems with format functions.
 
 Wformat-signedness
@@ -597,7 +597,7 @@
 Warn about zero-length formats.
 
 Wformat=
-C ObjC C++ ObjC++ Joined RejectNegative UInteger Var(warn_format) Warning LangEnabledBy(C ObjC C++ ObjC++,Wall, 1, 0) IntegerRange(0, 2)
+C ObjC C++ ObjC++ Joined RejectNegative UInteger Var(warn_format) Init(1) Warning LangEnabledBy(C ObjC C++ ObjC++,Wall, 1, 0) IntegerRange(0, 2)
 Warn about printf/scanf/strftime/strfmon format string anomalies.
 
 Wformat-overflow=
diff -ruN gcc-9.1.0-orig/gcc/config/aarch64/aarch64-linux.h gcc-9.1.0/gcc/config/aarch64/aarch64-linux.h
--- gcc-9.1.0-orig/gcc/config/aarch64/aarch64-linux.h	2019-04-24 00:51:25.000000000 +0900
+++ gcc-9.1.0/gcc/config/aarch64/aarch64-linux.h	2019-05-04 01:16:30.791467659 +0900
@@ -35,6 +35,7 @@
 #define CPP_SPEC "%{pthread:-D_REENTRANT}"
 
 #define LINUX_TARGET_LINK_SPEC  "%{h*}		\
+   --as-needed					\
    %{static:-Bstatic}				\
    %{shared:-shared}				\
    %{symbolic:-Bsymbolic}			\
diff -ruN gcc-9.1.0-orig/gcc/config/arm/linux-elf.h gcc-9.1.0/gcc/config/arm/linux-elf.h
--- gcc-9.1.0-orig/gcc/config/arm/linux-elf.h	2019-01-01 21:31:55.000000000 +0900
+++ gcc-9.1.0/gcc/config/arm/linux-elf.h	2019-05-04 01:21:25.223112262 +0900
@@ -70,6 +70,7 @@
      %{rdynamic:-export-dynamic} \
      %{!shared:-dynamic-linker " GNU_USER_DYNAMIC_LINKER "}} \
    -X \
+   --as-needed \
    %{mbig-endian:-EB} %{mlittle-endian:-EL}" \
    SUBTARGET_EXTRA_LINK_SPEC
 
diff -ruN gcc-9.1.0-orig/gcc/config/gnu-user.h gcc-9.1.0/gcc/config/gnu-user.h
--- gcc-9.1.0-orig/gcc/config/gnu-user.h	2019-01-16 18:38:21.000000000 +0900
+++ gcc-9.1.0/gcc/config/gnu-user.h	2019-05-04 01:13:31.484067424 +0900
@@ -136,17 +136,17 @@
 #define LIBASAN_EARLY_SPEC "%{!shared:libasan_preinit%O%s} " \
   "%{static-libasan:%{!shared:" \
   LD_STATIC_OPTION " --whole-archive -lasan --no-whole-archive " \
-  LD_DYNAMIC_OPTION "}}%{!static-libasan:-lasan}"
+  LD_DYNAMIC_OPTION "}}%{!static-libasan:%{!fuse-ld=gold:--push-state} --no-as-needed -lasan %{fuse-ld=gold:--as-needed;:--pop-state}}"
 #undef LIBTSAN_EARLY_SPEC
 #define LIBTSAN_EARLY_SPEC "%{!shared:libtsan_preinit%O%s} " \
   "%{static-libtsan:%{!shared:" \
   LD_STATIC_OPTION " --whole-archive -ltsan --no-whole-archive " \
-  LD_DYNAMIC_OPTION "}}%{!static-libtsan:-ltsan}"
+  LD_DYNAMIC_OPTION "}}%{!static-libtsan:%{!fuse-ld=gold:--push-state} --no-as-needed -ltsan %{fuse-ld=gold:--as-needed;:--pop-state}}"
 #undef LIBLSAN_EARLY_SPEC
 #define LIBLSAN_EARLY_SPEC "%{!shared:liblsan_preinit%O%s} " \
   "%{static-liblsan:%{!shared:" \
   LD_STATIC_OPTION " --whole-archive -llsan --no-whole-archive " \
-  LD_DYNAMIC_OPTION "}}%{!static-liblsan:-llsan}"
+  LD_DYNAMIC_OPTION "}}%{!static-liblsan:%{!fuse-ld=gold:--push-state} --no-as-needed -llsan %{fuse-ld=gold:--as-needed;:--pop-state}}"
 #endif
 
 #undef TARGET_F951_OPTIONS
diff -ruN gcc-9.1.0-orig/gcc/config/i386/gnu-user64.h gcc-9.1.0/gcc/config/i386/gnu-user64.h
--- gcc-9.1.0-orig/gcc/config/i386/gnu-user64.h	2019-01-01 21:31:55.000000000 +0900
+++ gcc-9.1.0/gcc/config/i386/gnu-user64.h	2019-05-04 01:19:41.341987517 +0900
@@ -56,6 +56,7 @@
                   "%{" SPEC_64 ":-m " GNU_USER_LINK_EMULATION64 "} \
                    %{" SPEC_32 ":-m " GNU_USER_LINK_EMULATION32 "} \
                    %{" SPEC_X32 ":-m " GNU_USER_LINK_EMULATIONX32 "} \
+  --as-needed \
   %{shared:-shared} \
   %{!shared: \
     %{!static: \
diff -ruN gcc-9.1.0-orig/gcc/config/i386/gnu-user.h gcc-9.1.0/gcc/config/i386/gnu-user.h
--- gcc-9.1.0-orig/gcc/config/i386/gnu-user.h	2019-01-01 21:31:55.000000000 +0900
+++ gcc-9.1.0/gcc/config/i386/gnu-user.h	2019-05-04 01:20:12.729111590 +0900
@@ -74,7 +74,7 @@
   { "link_emulation", GNU_USER_LINK_EMULATION },\
   { "dynamic_linker", GNU_USER_DYNAMIC_LINKER }
 
-#define GNU_USER_TARGET_LINK_SPEC "-m %(link_emulation) %{shared:-shared} \
+#define GNU_USER_TARGET_LINK_SPEC "-m %(link_emulation) --as-needed %{shared:-shared} \
   %{!shared: \
     %{!static: \
       %{!static-pie: \
diff -ruN gcc-9.1.0-orig/gcc/config/mips/gnu-user.h gcc-9.1.0/gcc/config/mips/gnu-user.h
--- gcc-9.1.0-orig/gcc/config/mips/gnu-user.h	2019-01-01 21:31:55.000000000 +0900
+++ gcc-9.1.0/gcc/config/mips/gnu-user.h	2019-05-04 01:20:35.796471936 +0900
@@ -55,6 +55,7 @@
 #undef GNU_USER_TARGET_LINK_SPEC
 #define GNU_USER_TARGET_LINK_SPEC "\
   %{G*} %{EB} %{EL} %{mips*} %{shared} \
+  -as-needed \
   %{!shared: \
     %{!static: \
       %{rdynamic:-export-dynamic} \
diff -ruN gcc-9.1.0-orig/gcc/config/riscv/linux.h gcc-9.1.0/gcc/config/riscv/linux.h
--- gcc-9.1.0-orig/gcc/config/riscv/linux.h	2019-01-01 21:31:55.000000000 +0900
+++ gcc-9.1.0/gcc/config/riscv/linux.h	2019-05-04 01:20:57.327877854 +0900
@@ -58,6 +58,7 @@
   "%{mabi=ilp32:_ilp32}"
 
 #define LINK_SPEC "\
+-as-needed \
 -melf" XLEN_SPEC "lriscv" LD_EMUL_SUFFIX " \
 %{mno-relax:--no-relax} \
 %{shared} \
diff -ruN gcc-9.1.0-orig/gcc/config/rs6000/linux64.h gcc-9.1.0/gcc/config/rs6000/linux64.h
--- gcc-9.1.0-orig/gcc/config/rs6000/linux64.h	2019-03-11 22:58:44.000000000 +0900
+++ gcc-9.1.0/gcc/config/rs6000/linux64.h	2019-05-04 01:17:53.833041958 +0900
@@ -473,13 +473,13 @@
 					   " -m elf64ppc")
 #endif
 
-#define LINK_OS_LINUX_SPEC32 LINK_OS_LINUX_EMUL32 " %{!shared: %{!static: \
+#define LINK_OS_LINUX_SPEC32 LINK_OS_LINUX_EMUL32 " --as-needed %{!shared: %{!static: \
   %{!static-pie: \
     %{rdynamic:-export-dynamic} \
     -dynamic-linker " GNU_USER_DYNAMIC_LINKER32 "}}} \
   %(link_os_extra_spec32)"
 
-#define LINK_OS_LINUX_SPEC64 LINK_OS_LINUX_EMUL64 " %{!shared: %{!static: \
+#define LINK_OS_LINUX_SPEC64 LINK_OS_LINUX_EMUL64 " --as-needed %{!shared: %{!static: \
   %{!static-pie: \
     %{rdynamic:-export-dynamic} \
     -dynamic-linker " GNU_USER_DYNAMIC_LINKER64 "}}} \
diff -ruN gcc-9.1.0-orig/gcc/config/rs6000/sysv4.h gcc-9.1.0/gcc/config/rs6000/sysv4.h
--- gcc-9.1.0-orig/gcc/config/rs6000/sysv4.h	2019-02-12 00:19:59.000000000 +0900
+++ gcc-9.1.0/gcc/config/rs6000/sysv4.h	2019-05-04 01:18:44.403594142 +0900
@@ -790,7 +790,7 @@
 #define GNU_USER_DYNAMIC_LINKER GLIBC_DYNAMIC_LINKER
 #endif
 
-#define LINK_OS_LINUX_SPEC "-m elf32ppclinux %{!shared: %{!static: \
+#define LINK_OS_LINUX_SPEC "-m elf32ppclinux --as-needed %{!shared: %{!static: \
   %{rdynamic:-export-dynamic} \
   -dynamic-linker " GNU_USER_DYNAMIC_LINKER "}}"
 
diff -ruN gcc-9.1.0-orig/gcc/config/s390/linux.h gcc-9.1.0/gcc/config/s390/linux.h
--- gcc-9.1.0-orig/gcc/config/s390/linux.h	2019-01-01 21:31:55.000000000 +0900
+++ gcc-9.1.0/gcc/config/s390/linux.h	2019-05-04 01:16:56.430711859 +0900
@@ -77,7 +77,7 @@
 
 #undef  LINK_SPEC
 #define LINK_SPEC \
-  "%{m31:-m elf_s390}%{m64:-m elf64_s390} \
+  "%{m31:-m elf_s390}%{m64:-m elf64_s390} --as-needed \
    %{shared:-shared} \
    %{!shared: \
       %{static:-static} \
diff -ruN gcc-9.1.0-orig/gcc/doc/invoke.texi gcc-9.1.0/gcc/doc/invoke.texi
--- gcc-9.1.0-orig/gcc/doc/invoke.texi	2019-04-23 19:03:41.000000000 +0900
+++ gcc-9.1.0/gcc/doc/invoke.texi	2019-05-04 01:09:26.689872777 +0900
@@ -8308,6 +8308,12 @@
 Please note the warning under @option{-fgcse} about
 invoking @option{-O2} on programs that use computed gotos.
 
+NOTE: In Ataraxia Linux, @option{-D_FORTIFY_SOURCE=2} is
+set by default, and is activated when @option{-O} is set to 2 or higher.
+This enables additional compile-time and run-time checks for several libc
+functions.  To disable, specify either @option{-U_FORTIFY_SOURCE} or
+@option{-D_FORTIFY_SOURCE=0}.
+
 @item -O3
 @opindex O3
 Optimize yet more.  @option{-O3} turns on all optimizations specified
@@ -13491,6 +13497,10 @@
 linker.  When using the GNU linker, you can also get the same effect with
 @option{-Wl,-Map=output.map}.
 
+
+NOTE: In Ataraxia Linux, for LDFLAGS, the option
+@option{-Wl,-z,relro,now} is used.  To disable, use @option{-Wl,-z,norelro}.
+
 @item -u @var{symbol}
 @opindex u
 Pretend the symbol @var{symbol} is undefined, to force linking of
diff -ruN gcc-9.1.0-orig/gcc/gcc.c gcc-9.1.0/gcc/gcc.c
--- gcc-9.1.0-orig/gcc/gcc.c	2019-03-11 22:58:44.000000000 +0900
+++ gcc-9.1.0/gcc/gcc.c	2019-05-04 01:26:45.898574024 +0900
@@ -696,8 +696,11 @@
 #ifdef LIBASAN_EARLY_SPEC
 #define LIBASAN_SPEC STATIC_LIBASAN_LIBS
 #elif defined(HAVE_LD_STATIC_DYNAMIC)
-#define LIBASAN_SPEC "%{static-libasan:" LD_STATIC_OPTION \
-		     "} -lasan %{static-libasan:" LD_DYNAMIC_OPTION "}" \
+#define LIBASAN_SPEC "%{static-libasan:" LD_STATIC_OPTION "}" \
+		     " %{!static-libasan:%{!fuse-ld=gold:--push-state }--no-as-needed}" \
+		     " -lasan " \
+		     " %{static-libasan:" LD_DYNAMIC_OPTION "}" \
+		     " %{!static-libasan:%{fuse-ld=gold:--as-needed;:--pop-state}}" \
 		     STATIC_LIBASAN_LIBS
 #else
 #define LIBASAN_SPEC "-lasan" STATIC_LIBASAN_LIBS
@@ -732,8 +735,11 @@
 #ifdef LIBLSAN_EARLY_SPEC
 #define LIBLSAN_SPEC STATIC_LIBLSAN_LIBS
 #elif defined(HAVE_LD_STATIC_DYNAMIC)
-#define LIBLSAN_SPEC "%{static-liblsan:" LD_STATIC_OPTION \
-		     "} -llsan %{static-liblsan:" LD_DYNAMIC_OPTION "}" \
+#define LIBLSAN_SPEC "%{static-liblsan:" LD_STATIC_OPTION "}" \
+		     " %{!static-liblsan:%{!fuse-ld=gold:--push-state }--no-as-needed}" \
+		     " -llsan " \
+		     " %{static-liblsan:" LD_DYNAMIC_OPTION "}" \
+		     " %{!static-liblsan:%{fuse-ld=gold:--as-needed;:--pop-state}}" \
 		     STATIC_LIBLSAN_LIBS
 #else
 #define LIBLSAN_SPEC "-llsan" STATIC_LIBLSAN_LIBS
@@ -748,8 +754,11 @@
 #define STATIC_LIBUBSAN_LIBS \
   " %{static-libubsan|static:%:include(libsanitizer.spec)%(link_libubsan)}"
 #ifdef HAVE_LD_STATIC_DYNAMIC
-#define LIBUBSAN_SPEC "%{static-libubsan:" LD_STATIC_OPTION \
-		     "} -lubsan %{static-libubsan:" LD_DYNAMIC_OPTION "}" \
+#define LIBUBSAN_SPEC "%{static-libubsan:" LD_STATIC_OPTION "}" \
+		     " %{!static-libubsan:%{!fuse-ld=gold:--push-state }--no-as-needed}" \
+		     " -lubsan " \
+		     " %{static-libubsan:" LD_DYNAMIC_OPTION "}" \
+		     " %{!static-libubsan:%{fuse-ld=gold:--as-needed;:--pop-state}}" \
 		     STATIC_LIBUBSAN_LIBS
 #else
 #define LIBUBSAN_SPEC "-lubsan" STATIC_LIBUBSAN_LIBS
@@ -875,6 +884,12 @@
 #define LINK_GCC_C_SEQUENCE_SPEC "%G %{!nolibc:%L %G}"
 #endif
 
+#ifdef ENABLE_DEFAULT_SSP
+#define NO_SSP_SPEC "%{nostdlib|nodefaultlibs|ffreestanding:-fno-stack-protector} "
+#else
+#define NO_SSP_SPEC ""
+#endif
+
 #ifndef LINK_SSP_SPEC
 #ifdef TARGET_LIBC_PROVIDES_SSP
 #define LINK_SSP_SPEC "%{fstack-protector|fstack-protector-all" \
@@ -1040,6 +1055,7 @@
    "%{flto|flto=*:%<fcompare-debug*} \
     %{flto} %{fno-lto} %{flto=*} %l " LINK_PIE_SPEC \
    "%{fuse-ld=*:-fuse-ld=%*} " LINK_COMPRESS_DEBUG_SPEC \
+   "-z relro -z now " \
    "%X %{o*} %{e*} %{N} %{n} %{r}\
     %{s} %{t} %{u*} %{z} %{Z} %{!nostdlib:%{!r:%{!nostartfiles:%S}}} \
     %{static|no-pie|static-pie:} %@{L*} %(mfwrap) %(link_libgcc) " \
@@ -1156,7 +1172,7 @@
  %{-version:--version}\
  %{-help=*:--help=%*}\
  %{!fsyntax-only:%{S:%W{o*}%{!o*:-o %b.s}}}\
- %{fsyntax-only:-o %j} %{-param*}\
+ %{fsyntax-only:-o %j} %{-param*} " NO_SSP_SPEC "\
  %{coverage:-fprofile-arcs -ftest-coverage}\
  %{fprofile-arcs|fprofile-generate*|coverage:\
    %{!fprofile-update=single:\
diff -ruN gcc-9.1.0-orig/gcc/params.def gcc-9.1.0/gcc/params.def
--- gcc-9.1.0-orig/gcc/params.def	2019-04-12 23:35:48.000000000 +0900
+++ gcc-9.1.0/gcc/params.def	2019-05-04 01:27:14.409833210 +0900
@@ -752,7 +752,7 @@
 DEFPARAM (PARAM_SSP_BUFFER_SIZE,
 	  "ssp-buffer-size",
 	  "The lower bound for a buffer to be considered for stack smashing protection.",
-	  8, 1, 0)
+	  4, 1, 0)
 
 DEFPARAM (PARAM_MIN_SIZE_FOR_STACK_SHARING,
 	  "min-size-for-stack-sharing",
diff -ruN gcc-9.1.0-orig/include/libiberty.h gcc-9.1.0/include/libiberty.h
--- gcc-9.1.0-orig/include/libiberty.h	2019-01-10 06:39:49.000000000 +0900
+++ gcc-9.1.0/include/libiberty.h	2019-05-04 01:29:45.965931181 +0900
@@ -641,8 +641,11 @@
 /* Like sprintf but provides a pointer to malloc'd storage, which must
    be freed by the caller.  */
 
+/* asprintf may be declared as a macro by glibc with __USE_FORTIFY_LEVEL.  */
+#ifndef asprintf
 extern int asprintf (char **, const char *, ...) ATTRIBUTE_PRINTF_2;
 #endif
+#endif
 
 /* Like asprintf but allocates memory without fail. This works like
    xmalloc.  */
diff -ruN gcc-9.1.0-orig/libatomic/Makefile.in gcc-9.1.0/libatomic/Makefile.in
--- gcc-9.1.0-orig/libatomic/Makefile.in	2019-05-03 17:01:59.000000000 +0900
+++ gcc-9.1.0/libatomic/Makefile.in	2019-05-04 01:32:15.606126510 +0900
@@ -414,7 +414,8 @@
 PAT_S = $(word 3,$(PAT_SPLIT))
 IFUNC_DEF = -DIFUNC_ALT=$(PAT_S)
 IFUNC_OPT = $(word $(PAT_S),$(IFUNC_OPTIONS))
-M_DEPS = -MT $@ -MD -MP -MF $(DEPDIR)/$(@F).Ppo
+@AMDEP_TRUE@M_DEPS = -MT $@ -MD -MP -MF $(DEPDIR)/$(@F).Ppo
+@AMDEP_FALSE@M_DEPS =
 M_SIZE = -DN=$(PAT_N)
 M_IFUNC = $(if $(PAT_S),$(IFUNC_DEF) $(IFUNC_OPT))
 M_FILE = $(PAT_BASE)_n.c
diff -ruN gcc-9.1.0-orig/libiberty/Makefile.in gcc-9.1.0/libiberty/Makefile.in
--- gcc-9.1.0-orig/libiberty/Makefile.in	2019-01-01 21:31:55.000000000 +0900
+++ gcc-9.1.0/libiberty/Makefile.in	2019-05-04 01:31:02.099990325 +0900
@@ -254,6 +254,7 @@
 	  $(AR) $(AR_FLAGS) $(TARGETLIB) \
 	    $(REQUIRED_OFILES) $(EXTRA_OFILES) $(LIBOBJS); \
 	  $(RANLIB) $(TARGETLIB); \
+          cp $(TARGETLIB) ../ ; \
 	  cd ..; \
 	else true; fi; \
 	if [ x"$(NOASANFLAG)" != x ]; then \
diff -ruN gcc-9.1.0-orig/libitm/configure.tgt gcc-9.1.0/libitm/configure.tgt
--- gcc-9.1.0-orig/libitm/configure.tgt	2019-01-01 21:31:55.000000000 +0900
+++ gcc-9.1.0/libitm/configure.tgt	2019-05-04 01:31:41.866980860 +0900
@@ -43,6 +43,16 @@
   esac
 fi
 
+# FIXME: error: inlining failed in call to always_inline 
+# ‘int vfprintf(FILE*, const char*, __va_list_tag*)’
+# : function body can be overwritten at link time
+# Disable Fortify in libitm for now. #508852
+case "${target}" in
+    *-*-linux*)
+	XCFLAGS="${XCFLAGS} -U_FORTIFY_SOURCE"
+	;;
+esac
+
 # Map the target cpu to an ARCH sub-directory.  At the same time,
 # work out any special compilation flags as necessary.
 case "${target_cpu}" in
diff -ruN gcc-9.1.0-orig/libstdc++-v3/po/Makefile.am gcc-9.1.0/libstdc++-v3/po/Makefile.am
--- gcc-9.1.0-orig/libstdc++-v3/po/Makefile.am	2019-01-01 21:31:55.000000000 +0900
+++ gcc-9.1.0/libstdc++-v3/po/Makefile.am	2019-05-04 01:27:52.480847581 +0900
@@ -38,6 +38,7 @@
 EXTRA_DIST = string_literals.cc POTFILES.in $(PACKAGE).pot $(LOCALE_IN)
 
 .po.mo:
+	env --unset=LD_LIBRARY_PATH \
 	$(MSGFMT) -o $@ $<
 
 all-local: all-local-$(USE_NLS)
diff -ruN gcc-9.1.0-orig/libstdc++-v3/po/Makefile.in gcc-9.1.0/libstdc++-v3/po/Makefile.in
--- gcc-9.1.0-orig/libstdc++-v3/po/Makefile.in	2019-02-27 20:25:44.000000000 +0900
+++ gcc-9.1.0/libstdc++-v3/po/Makefile.in	2019-05-04 01:28:13.624301863 +0900
@@ -547,6 +547,7 @@
 
 
 .po.mo:
+	env --unset=LD_LIBRARY_PATH \
 	$(MSGFMT) -o $@ $<
 
 all-local: all-local-$(USE_NLS)
