From 5a53acfb46658b2114b134584dd60c5e1d9bce81 Mon Sep 17 00:00:00 2001
From: Ishimoto Shinobu <nagakamira@gmail.com>
Date: Mon, 8 Jun 2020 00:50:21 +0900
Subject: [PATCH] musl

---
 tools/inetcat.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/tools/inetcat.c b/tools/inetcat.c
index af03dc0..67eb14c 100644
--- a/tools/inetcat.c
+++ b/tools/inetcat.c
@@ -24,6 +24,9 @@
 #ifdef HAVE_CONFIG_H
 #include <config.h>
 #endif
+#ifndef fd_set
+#include <sys/select.h>
+#endif
 #include <stdio.h>
 #include <stdlib.h>
 #include <stdint.h>

