From 400fabe42d34d05db15663ee0c9f6c054f6d0094 Mon Sep 17 00:00:00 2001
From: Ishimoto Shinobu <nagakamira@gmail.com>
Date: Wed, 29 Apr 2020 19:13:17 +0300
Subject: [PATCH] gtk: don't use check-version.py

Signed-off-by: Ishimoto Shinobu <nagakamira@gmail.com>
---
 meson.build | 10 ----------
 1 file changed, 10 deletions(-)

diff --git a/meson.build b/meson.build
index 68d5685..a663234 100644
--- a/meson.build
+++ b/meson.build
@@ -987,16 +987,6 @@ if host_machine.system() != 'windows'
     install_dir : join_paths(gtk_datadir, 'gtk-3.0', 'valgrind'))
 endif
 
-test(
-  'version-check',
-  find_program('check-version.py'),
-  args: [
-    join_paths(meson.current_source_dir(), 'configure.ac'),
-    join_paths(meson.current_source_dir(), 'meson.build'),
-  ],
-  suite: 'gtk',
-)
-
 summary = [
   '',
   '------',
-- 
2.26.2

