diff --git a/src/util/rand_xor.c b/src/util/rand_xor.c
index 81b64f1ea71..14e6c9d27c3 100644
--- a/src/util/rand_xor.c
+++ b/src/util/rand_xor.c
@@ -27,6 +27,8 @@
 #if !DETECT_OS_WINDOWS
 #if defined(HAVE_GETRANDOM)
 #include <sys/random.h>
+#include <sys/types.h>
+#include <sys/stat.h>
 #endif
 #include <unistd.h>
 #include <fcntl.h>

diff --git a/src/gallium/frontends/nine/nine_debug.c b/src/gallium/frontends/nine/nine_debug.c
index 784d535..f4a6c41 100644
--- a/src/gallium/frontends/nine/nine_debug.c
+++ b/src/gallium/frontends/nine/nine_debug.c
@@ -65,7 +65,7 @@ _nine_debug_printf( unsigned long flag,
 {
     static boolean first = TRUE;
     static unsigned long dbg_flags = DBG_ERROR | DBG_WARN;
-    unsigned long tid = 0;
+    pthread_t tid = 0;
 
     if (first) {
         first = FALSE;
