diff -ruN mesa-19.2.0-orig/meson.build mesa-19.2.0/meson.build
--- mesa-19.2.0-orig/meson.build	2019-09-26 01:59:31.000000000 +0900
+++ mesa-19.2.0/meson.build	2019-10-03 20:00:26.800550472 +0900
@@ -379,7 +379,10 @@
 
 # Android uses emutls for versions <= P/28. For USE_ELF_TLS we need ELF TLS.
 if not with_platform_android or get_option('platform-sdk-version') >= 29
-  pre_args += '-DUSE_ELF_TLS'
+  with_glx_use_tls = get_option('glx-use-tls')
+  if with_glx_use_tls
+    pre_args += '-DUSE_ELF_TLS'
+  endif
 endif
 
 if with_glx != 'disabled'
diff -ruN mesa-19.2.0-orig/meson_options.txt mesa-19.2.0/meson_options.txt
--- mesa-19.2.0-orig/meson_options.txt	2019-09-26 01:59:31.000000000 +0900
+++ mesa-19.2.0/meson_options.txt	2019-10-03 20:00:52.815353511 +0900
@@ -215,6 +215,12 @@
   description : 'Build support for GLX platform'
 )
 option(
+  'glx-use-tls',
+  type : 'boolean',
+  value : true,
+  description : 'Build support for initial-exec TLS model on GLX platform'
+)
+option(
   'egl',
   type : 'combo',
   value : 'auto',
