diff -ruN mesa-19.2.0-orig/meson.build mesa-19.2.0/meson.build
--- mesa-19.2.0-orig/meson.build	2019-09-26 01:59:31.000000000 +0900
+++ mesa-19.2.0/meson.build	2019-10-05 22:51:56.770929132 +0900
@@ -502,10 +502,12 @@
   endif
 endif
 dep_xvmc = null_dep
+dep_xv = null_dep
 with_gallium_xvmc = false
 if _xvmc != 'false'
   dep_xvmc = dependency('xvmc', version : '>= 1.0.6', required : _xvmc == 'true')
-  with_gallium_xvmc = dep_xvmc.found()
+  dep_xv = dependency('xv', required : _xvmc == 'true')
+  with_gallium_xvmc = dep_xvmc.found() and dep_xv.found()
 endif
 
 xvmc_drivers_path = get_option('xvmc-libs-path')
diff -ruN mesa-19.2.0-orig/src/gallium/state_trackers/xvmc/meson.build mesa-19.2.0/src/gallium/state_trackers/xvmc/meson.build
--- mesa-19.2.0-orig/src/gallium/state_trackers/xvmc/meson.build	2019-09-26 01:59:31.000000000 +0900
+++ mesa-19.2.0/src/gallium/state_trackers/xvmc/meson.build	2019-10-05 22:51:32.129875163 +0900
@@ -23,7 +23,9 @@
   files('attributes.c', 'block.c', 'context.c', 'surface.c', 'subpicture.c'),
   c_args : [c_vis_args],
   include_directories : [inc_common],
-  dependencies : [dep_xvmc, dep_x11_xcb, dep_xcb, dep_xcb_dri2, dep_xcb_dri3],
+  dependencies : [
+    dep_xvmc, dep_x11_xcb, dep_xcb, dep_xcb_dri2, dep_xcb_dri3, dep_xv,
+  ],
 )
 
 # These tests will not work without a working xvmc configuration. 
@@ -36,7 +38,8 @@
       _name,
       files('tests/test_@0@.c'.format(x), 'tests/testlib.c'),
       dependencies : [
-        dep_real_xvmc, dep_x11_xcb, dep_xcb, dep_xcb_dri2, dep_xcb_dri3, dep_xvmcw,
+        dep_real_xvmc, dep_x11_xcb, dep_xcb, dep_xcb_dri2, dep_xcb_dri3,
+        dep_xvmcw, dep_xv,
       ],
       install: true,
     )
@@ -46,7 +49,8 @@
     'xvmc_bench',
     files('tests/xvmc_bench.c', 'tests/testlib.c'),
     dependencies : [
-      dep_real_xvmc, dep_x11_xcb, dep_xcb, dep_xcb_dri2, dep_xcb_dri3, dep_xvmcw,
+      dep_real_xvmc, dep_x11_xcb, dep_xcb, dep_xcb_dri2, dep_xcb_dri3,
+      dep_xvmcw, dep_xv,
     ],
     install: true,
   )
