--- v4l-utils-1.18.0-orig/utils/ir-ctl/keymap.c
+++ v4l-utils-1.18.0/utils/ir-ctl/keymap.c
@@ -9,6 +9,7 @@
 #include <string.h>
 #include <limits.h>
 #include <stdbool.h>
+#include <sys/types.h>
 
 #include "keymap.h"
 #include "toml.h"
--- v4l-utils-1.18.0-orig/utils/ir-ctl/keymap.h
+++ v4l-utils-1.18.0/utils/ir-ctl/keymap.h
@@ -1,4 +1,6 @@
 /* SPDX-License-Identifier: GPL-2.0 */
+#include <sys/types.h>
+
 #ifndef __KEYMAP_H
 #define __KEYMAP_H
 
@@ -31,6 +33,8 @@
 	char *keycode;
 	u_int32_t raw[1];
 };
+
+typedef int error_t;
 
 void free_keymap(struct keymap *map);
 error_t parse_keymap(char *fname, struct keymap **keymap, bool verbose);
--- v4l-utils-1.18.0-orig/utils/keytable/keymap.h
+++ v4l-utils-1.18.0/utils/keytable/keymap.h
@@ -1,4 +1,6 @@
 /* SPDX-License-Identifier: GPL-2.0 */
+#include <sys/types.h>
+
 #ifndef __KEYMAP_H
 #define __KEYMAP_H
 
@@ -31,6 +33,8 @@
 	char *keycode;
 	u_int32_t raw[1];
 };
+
+typedef int error_t;
 
 void free_keymap(struct keymap *map);
 error_t parse_keymap(char *fname, struct keymap **keymap, bool verbose);
