From d44979b3f0d09102d60c35f808b180123c0799f6 Mon Sep 17 00:00:00 2001
From: Ishimoto Shinobu <nagakamira@gmail.com>
Date: Wed, 24 Jun 2020 07:43:46 +0900
Subject: [PATCH] no gtk-doc

Signed-off-by: Ishimoto Shinobu <nagakamira@gmail.com>
---
 Makefile.am  |  2 +-
 configure.ac | 11 -----------
 2 files changed, 1 insertion(+), 12 deletions(-)

diff --git a/Makefile.am b/Makefile.am
index 056d414..727f4ea 100644
--- a/Makefile.am
+++ b/Makefile.am
@@ -1,4 +1,4 @@
-SUBDIRS = gfbgraph docs tests
+SUBDIRS = gfbgraph tests
 ACLOCAL_AMFLAGS = -I m4
 
 libgfbgraphdocdir = ${prefix}/doc/libgfbgraph
diff --git a/configure.ac b/configure.ac
index fd1231f..e8241a1 100644
--- a/configure.ac
+++ b/configure.ac
@@ -31,13 +31,6 @@ AC_PROG_CC
 
 LT_INIT
 
-# check for gtk-doc
-m4_ifdef([GTK_DOC_CHECK], [
-GTK_DOC_CHECK([1.14],[--flavour no-tmpl])
-],[
-AM_CONDITIONAL([ENABLE_GTK_DOC], false)
-])
-
 GOBJECT_INTROSPECTION_CHECK([1.30.0])
 
 PKG_CHECK_MODULES(LIBGFBGRAPH, [glib-2.0 gio-2.0 gobject-2.0 rest-0.7 json-glib-1.0])
@@ -53,9 +46,5 @@ AC_SUBST(GOA_API_CHANGE_CPPFLAGS)
 AC_OUTPUT([
 Makefile
 libgfbgraph.pc
-docs/Makefile
-docs/reference/Makefile
-docs/reference/apiversion.xml
-docs/reference/version.xml
 gfbgraph/Makefile
 tests/Makefile])
-- 
2.27.0

