diff -Naur a/src/plugins/lvm-dbus.c b/src/plugins/lvm-dbus.c
--- a/src/plugins/lvm-dbus.c	2021-01-11 10:48:39.000000000 +0000
+++ b/src/plugins/lvm-dbus.c	2021-02-04 09:59:08.823000000 +0000
@@ -241,7 +241,6 @@
     return TRUE;
 }
 
-static volatile guint avail_deps = 0;
 static volatile guint avail_dbus_deps = 0;
 static volatile guint avail_features = 0;
 static volatile guint avail_module_deps = 0;
