diff -ruN libvpx-1.8.1-orig/configure libvpx-1.8.1/configure
--- libvpx-1.8.1-orig/configure	2019-07-15 14:55:33.000000000 -0700
+++ libvpx-1.8.1/configure	2019-08-21 02:20:14.345484621 -0700
@@ -165,10 +165,6 @@
     [ -f "${source_path}/${t}.mk" ] && enable_feature ${t}
 done
 
-if ! diff --version >/dev/null; then
-  die "diff missing: Try installing diffutils via your package manager."
-fi
-
 if ! perl --version >/dev/null; then
     die "Perl is required to build"
 fi
