--- mutter-3.34.0-orig/src/wayland/meta-wayland-buffer.c
+++ mutter-3.34.0/src/wayland/meta-wayland-buffer.c
@@ -26,7 +26,7 @@
 
 #include "wayland/meta-wayland-buffer.h"
 
-#include <drm_fourcc.h>
+#include <drm/drm_fourcc.h>
 
 #include "backends/meta-backend-private.h"
 #include "clutter/clutter.h"
--- mutter-3.34.0-orig/src/wayland/meta-wayland-dma-buf.c
+++ mutter-3.34.0/src/wayland/meta-wayland-dma-buf.c
@@ -29,7 +29,7 @@
 
 #include "wayland/meta-wayland-dma-buf.h"
 
-#include <drm_fourcc.h>
+#include <drm/drm_fourcc.h>
 
 #include "backends/meta-backend-private.h"
 #include "backends/meta-egl-ext.h"
