--- gnome-control-center-3.36.0-orig/panels/meson.build
+++ gnome-control-center-3.36.0/panels/meson.build
@@ -18,7 +18,6 @@
   'notifications',
   'online-accounts',
   'power',
-  'printers',
   'region',
   'removable-media',
   'search',
--- gnome-control-center-3.36.0-orig/shell/cc-panel-loader.c
+++ gnome-control-center-3.36.0/shell/cc-panel-loader.c
@@ -50,7 +50,7 @@
 extern GType cc_notifications_panel_get_type (void);
 extern GType cc_goa_panel_get_type (void);
 extern GType cc_power_panel_get_type (void);
-extern GType cc_printers_panel_get_type (void);
+//extern GType cc_printers_panel_get_type (void);
 extern GType cc_region_panel_get_type (void);
 extern GType cc_removable_media_panel_get_type (void);
 extern GType cc_search_panel_get_type (void);
@@ -114,7 +114,7 @@
   PANEL_TYPE("notifications",    cc_notifications_panel_get_type,        NULL),
   PANEL_TYPE("online-accounts",  cc_goa_panel_get_type,                  NULL),
   PANEL_TYPE("power",            cc_power_panel_get_type,                NULL),
-  PANEL_TYPE("printers",         cc_printers_panel_get_type,             NULL),
+//  PANEL_TYPE("printers",         cc_printers_panel_get_type,             NULL),
   PANEL_TYPE("region",           cc_region_panel_get_type,               NULL),
   PANEL_TYPE("removable-media",  cc_removable_media_panel_get_type,      NULL),
   PANEL_TYPE("search",           cc_search_panel_get_type,               NULL),
--- gnome-control-center-3.36.0-orig/tests/meson.build
+++ gnome-control-center-3.36.0/tests/meson.build
@@ -6,5 +6,4 @@
 
 subdir('interactive-panels')
 
-subdir('printers')
 subdir('info')
