From 2ee1e1bbaae8e5eff357e182086b0dc753ce4463 Mon Sep 17 00:00:00 2001
From: Ishimoto Shinobu <nagakamira@gmail.com>
Date: Mon, 2 Dec 2019 22:07:16 +0900
Subject: [PATCH] restore consolekit support

Signed-off-by: Ishimoto Shinobu <nagakamira@gmail.com>
---
 panels/power/cc-power-panel.c | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/panels/power/cc-power-panel.c b/panels/power/cc-power-panel.c
index c31249c..15448b7 100644
--- a/panels/power/cc-power-panel.c
+++ b/panels/power/cc-power-panel.c
@@ -2212,9 +2212,9 @@ can_suspend_or_hibernate (CcPowerPanel *self,
     }
 
   variant = g_dbus_connection_call_sync (connection,
-                                         "org.freedesktop.login1",
-                                         "/org/freedesktop/login1",
-                                         "org.freedesktop.login1.Manager",
+                                         "org.freedesktop.ConsoleKit",
+                                         "/org/freedesktop/ConsoleKit/Manager",
+                                         "org.freedesktop.ConsoleKit.Manager",
                                          method_name,
                                          NULL,
                                          NULL,
-- 
2.24.0

