From b877da2862c8a3bdd9a071dcd55e6ac74f3b3570 Mon Sep 17 00:00:00 2001
From: Ishimoto Shinobu <nagakamira@gmail.com>
Date: Thu, 5 Dec 2019 19:39:29 +0900
Subject: [PATCH] oh

Signed-off-by: Ishimoto Shinobu <nagakamira@gmail.com>
---
 panels/datetime/cc-datetime-panel.c | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/panels/datetime/cc-datetime-panel.c b/panels/datetime/cc-datetime-panel.c
index 5f106353c..bc498fc1c 100644
--- a/panels/datetime/cc-datetime-panel.c
+++ b/panels/datetime/cc-datetime-panel.c
@@ -1280,7 +1280,11 @@ cc_date_time_panel_init (CcDateTimePanel *self)
 
   /* After the initial setup, so we can be sure that
    * the model is filled up */
-  get_initial_timezone (self);
+  //get_initial_timezone (self);
+  date_time_mechanism_call_get_timezone (self->dtm,
+                                         self->cancellable,
+                                         get_timezone_cb,
+                                         self);
 
   g_signal_connect_object (gtk_entry_get_completion (GTK_ENTRY (self->timezone_searchentry)),
                            "match-selected", G_CALLBACK (city_changed_cb), self, G_CONNECT_SWAPPED);
