diff -ruN gnome-control-center-3.32.2-orig/shell/cc-shell-model.c gnome-control-center-3.32.2/shell/cc-shell-model.c
--- gnome-control-center-3.32.2-orig/shell/cc-shell-model.c	2019-05-24 11:51:00.000000000 -0700
+++ gnome-control-center-3.32.2/shell/cc-shell-model.c	2019-08-21 10:40:27.376829427 -0700
@@ -455,7 +455,7 @@
    * panel was found or not. It is a programming error to try to set
    * the visibility of a non-existant panel.
    */
-  g_assert (valid);
+  //g_assert (valid);
 
   gtk_list_store_set (GTK_LIST_STORE (self), &iter, COL_VISIBILITY, visibility, -1);
 }
