diff -ruN gnome-control-center-3.32.2-orig/meson.build gnome-control-center-3.32.2/meson.build
--- gnome-control-center-3.32.2-orig/meson.build	2019-05-24 11:50:58.000000000 -0700
+++ gnome-control-center-3.32.2/meson.build	2019-08-21 08:24:39.856074784 -0700
@@ -144,26 +144,6 @@
   dependency('gtk+-3.0', version: '>= 3.22.20')
 ]
 
-# Check for CUPS 1.4 or newer
-cups_dep = dependency('cups', version : '>= 1.4', required: false)
-assert(cups_dep.found(), 'CUPS 1.4 or newer not found')
-
-# https://bugzilla.gnome.org/show_bug.cgi?id=696766
-cups_cflags = []
-if cups_dep.version().version_compare('>= 1.6')
-  cups_cflags += '-D_PPD_DEPRECATED=""'
-endif
-
-# cups headers
-check_headers = [
-  ['HAVE_CUPS_CUPS_H', 'cups/cups.h'],
-  ['HAVE_CUPS_PPD_H', 'cups/ppd.h']
-]
-
-foreach header: check_headers
-  assert(cc.has_header(header[1], args: cups_cflags), 'CUPS headers not found: ' + header[1])
-endforeach
-
 # Optional dependency for the user accounts panel
 enable_cheese = get_option('cheese')
 if enable_cheese
diff -ruN gnome-control-center-3.32.2-orig/panels/meson.build gnome-control-center-3.32.2/panels/meson.build
--- gnome-control-center-3.32.2-orig/panels/meson.build	2019-05-24 11:50:58.000000000 -0700
+++ gnome-control-center-3.32.2/panels/meson.build	2019-08-21 08:24:32.288064059 -0700
@@ -12,14 +12,12 @@
   'notifications',
   'online-accounts',
   'power',
-  'printers',
   'privacy',
   'region',
   'search',
   'sharing',
   'sound',
-  'universal-access',
-  'user-accounts'
+  'universal-access'
 ]
 
 if host_is_linux
diff -ruN gnome-control-center-3.32.2-orig/shell/cc-panel-loader.c gnome-control-center-3.32.2/shell/cc-panel-loader.c
--- gnome-control-center-3.32.2-orig/shell/cc-panel-loader.c	2019-05-24 11:51:00.000000000 -0700
+++ gnome-control-center-3.32.2/shell/cc-panel-loader.c	2019-08-21 08:28:46.801387035 -0700
@@ -51,7 +51,6 @@
 extern GType cc_notifications_panel_get_type (void);
 extern GType cc_goa_panel_get_type (void);
 extern GType cc_power_panel_get_type (void);
-extern GType cc_printers_panel_get_type (void);
 extern GType cc_privacy_panel_get_type (void);
 extern GType cc_region_panel_get_type (void);
 extern GType cc_search_panel_get_type (void);
@@ -61,7 +60,6 @@
 extern GType cc_bolt_panel_get_type (void);
 #endif /* BUILD_THUNDERBOLT */
 extern GType cc_ua_panel_get_type (void);
-extern GType cc_user_panel_get_type (void);
 #ifdef BUILD_WACOM
 extern GType cc_wacom_panel_get_type (void);
 #endif /* BUILD_WACOM */
@@ -104,7 +102,6 @@
   PANEL_TYPE("notifications",    cc_notifications_panel_get_type,        NULL),
   PANEL_TYPE("online-accounts",  cc_goa_panel_get_type,                  NULL),
   PANEL_TYPE("power",            cc_power_panel_get_type,                NULL),
-  PANEL_TYPE("printers",         cc_printers_panel_get_type,             NULL),
   PANEL_TYPE("privacy",          cc_privacy_panel_get_type,              NULL),
   PANEL_TYPE("region",           cc_region_panel_get_type,               NULL),
   PANEL_TYPE("search",           cc_search_panel_get_type,               NULL),
@@ -114,7 +111,6 @@
   PANEL_TYPE("thunderbolt",      cc_bolt_panel_get_type,                 NULL),
 #endif
   PANEL_TYPE("universal-access", cc_ua_panel_get_type,                   NULL),
-  PANEL_TYPE("user-accounts",    cc_user_panel_get_type,                 NULL),
 #ifdef BUILD_WACOM
   PANEL_TYPE("wacom",            cc_wacom_panel_get_type,                cc_wacom_panel_static_init_func),
 #endif
diff -ruN gnome-control-center-3.32.2-orig/tests/meson.build gnome-control-center-3.32.2/tests/meson.build
--- gnome-control-center-3.32.2-orig/tests/meson.build	2019-05-24 11:51:00.000000000 -0700
+++ gnome-control-center-3.32.2/tests/meson.build	2019-08-21 08:25:55.466597048 -0700
@@ -6,5 +6,4 @@
 
 subdir('interactive-panels')
 
-subdir('printers')
 subdir('info')
