diff --git a/doc/maintainers.rdoc b/doc/maintainers.rdoc
index ec6a6b7134..11a1079e57 100644
--- a/doc/maintainers.rdoc
+++ b/doc/maintainers.rdoc
@@ -72,6 +72,8 @@ Zachary Scott (zzak)
   Akinori MUSHA (knu)
 [lib/irb.rb, lib/irb/*]
   Keiju ISHITSUKA (keiju)
+[lib/json]
+  NARUSE, Yui (naruse)
 [lib/logger.rb]
   Naotoshi Seo (sonots)
 [lib/mathn.rb]
diff --git a/lib/json/.document b/lib/json/.document
new file mode 100644
index 0000000000..7ae6b614de
--- /dev/null
+++ b/lib/json/.document
@@ -0,0 +1,3 @@
+*.rb
+add/*.rb
+pure/*.rb
