--- docbook2X-0.8.8-original/configure.ac	2007-03-03 07:44:14.000000000 -0600
+++ docbook2X-0.8.8-fixed/configure.ac	2007-10-10 14:19:31.000000000 -0500
@@ -149,6 +149,11 @@
 dnl Ensure that all static_* are fully expanded.
 
 eval static_datadir="$datadir"
+old_val=""
+until test "$static_datadir" = "$old_val"; do
+	old_val="$static_datadir"
+	eval static_datadir="$static_datadir"
+done
 
 eval static_bindir="$bindir"
 old_val=""
