diff -ruN apt-1.8.2-orig/CMakeLists.txt apt-1.8.2/CMakeLists.txt
--- apt-1.8.2-orig/CMakeLists.txt	2019-05-28 23:40:29.000000000 +0900
+++ apt-1.8.2/CMakeLists.txt	2019-08-04 00:05:41.624776600 +0900
@@ -10,8 +10,8 @@
 
 enable_testing()
 
-option(WITH_DOC "Build documentation." ON)
-option(USE_NLS "Localisation support." ON)
+option(WITH_DOC "Build documentation." OFF)
+option(USE_NLS "Localisation support." OFF)
 
 set(CMAKE_MODULE_PATH "${PROJECT_SOURCE_DIR}/CMake")
 
@@ -232,7 +232,6 @@
 add_subdirectory(apt-inst)
 add_subdirectory(cmdline)
 add_subdirectory(completions)
-add_subdirectory(doc)
 add_subdirectory(dselect)
 add_subdirectory(ftparchive)
 add_subdirectory(methods)
