From f6291dff804d29c9f7de8e4c7e854abdfbfcd628 Mon Sep 17 00:00:00 2001
From: Ishimoto Shinobu <nagakamira@gmail.com>
Date: Sun, 18 Apr 2021 14:33:56 -0700
Subject: [PATCH] better configure

Signed-off-by: Ishimoto Shinobu <nagakamira@gmail.com>
---
 configure.ac | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/configure.ac b/configure.ac
index c0aab53..cc30772 100644
--- a/configure.ac
+++ b/configure.ac
@@ -538,7 +538,7 @@ if test "x$THREADLIB_CPPFLAGS" != x; then
 else
   GPG_ERROR_CONFIG_MT_CFLAGS=""
 fi
-GPG_ERROR_CONFIG_HOST="$host"
+GPG_ERROR_CONFIG_HOST="$XTARGET"
 
 case "$includedir" in
   /usr/include|/include) ;;
-- 
2.31.0

