--- configure.ac.orig   Thu Apr 16 20:17:13 2020
+++ configure.ac        Sat May 23 11:42:11 2020
@@ -227,7 +227,7 @@
 AM_CONDITIONAL([USE_DOCBOOK2X], [test "x$db2xman" != "xdocbook2man"])

 if test "x$db2xman" = "xdocbook2man"; then
-	docdtd="\"-//Davenport//DTD DocBook V3.0//EN\""
+	docdtd="\"-//Davenport//DTD DocBook V3.0//EN\" \"/dev/null\""
 else
	docdtd="\"-//OASIS//DTD DocBook XML\" \"http://www.oasis-open.org/docbook/xml/4.5/docbookx.dtd\""
 fi
