From 66999a37d7b700ed85675cfc9ac8c38ea68a7301 Mon Sep 17 00:00:00 2001
From: Ishimoto Shinobu <nagakamira@gmail.com>
Date: Fri, 27 Mar 2020 23:52:25 +0900
Subject: [PATCH] fix build

Signed-off-by: Ishimoto Shinobu <nagakamira@gmail.com>
---
 configure | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/configure b/configure
index 5507de7..35e7001 100755
--- a/configure
+++ b/configure
@@ -41,7 +41,7 @@ OSENUM=
 OSNAME=
 UTF8_LOCALE=
 
-CC=`printf "all:\\n\\t@echo \\\$(CC)\\n" | env -i make -sf -`
+CC=
 CFLAGS=
 LDADD=
 LDFLAGS=
-- 
2.25.1

