From 89fb13bce5590745709cf023b38ef7ad54dcec8d Mon Sep 17 00:00:00 2001
From: Ishimoto Shinobu <nagakamira@gmail.com>
Date: Sun, 23 Aug 2020 17:28:36 +0500
Subject: [PATCH] fix cross build

Signed-off-by: Ishimoto Shinobu <nagakamira@gmail.com>
---
 configure | 13 ++-----------
 1 file changed, 2 insertions(+), 11 deletions(-)

diff --git a/configure b/configure
index 5507de7..3ba7db1 100755
--- a/configure
+++ b/configure
@@ -41,7 +41,7 @@ OSENUM=
 OSNAME=
 UTF8_LOCALE=
 
-CC=`printf "all:\\n\\t@echo \\\$(CC)\\n" | env -i make -sf -`
+CC=
 CFLAGS=
 LDADD=
 LDFLAGS=
@@ -168,14 +168,6 @@ __HEREDOC__
 	if ${COMP} -o "test-${1}" "${SOURCEDIR}/test-${1}.c" ${3} ${4} 1>&3 2>&3
 	then
 		echo "partial result of ${n}: ${CC} succeeded" 1>&3
-	else
-		echo "result of ${n}: ${CC} failed with exit status $?" 1>&3
-		echo "result of compiling ${n}: no" 1>&3
-		echo 1>&3
-		return 1
-	fi
-
-	if ./test-${1} 1>&3 2>&3; then
 		echo "tested ${n}: yes" 1>&2
 		echo "result of running ${n}: yes" 1>&3
 		echo 1>&3
@@ -185,10 +177,9 @@ __HEREDOC__
 		rm "test-${1}"
 		return 0
 	else
-		echo "result of ${n}: execution failed with exit status $?" 1>&3
+		echo "result of ${n}: ${CC} failed with exit status $?" 1>&3
 		echo "result of running ${n}: no" 1>&3
 		echo 1>&3
-		rm "test-${1}"
 		return 1
 	fi
 }
-- 
2.18.4

