diff -ruN mandoc-1.14.5-orig/main.c mandoc-1.14.5/main.c
--- mandoc-1.14.5-orig/main.c	2019-03-10 18:56:43.000000000 +0900
+++ mandoc-1.14.5/main.c	2019-12-01 19:58:48.732107513 +0900
@@ -1167,11 +1167,7 @@
 	int		 argc, use_ofn;
 	pid_t		 pager_pid;
 
-	pager = getenv("MANPAGER");
-	if (pager == NULL || *pager == '\0')
-		pager = getenv("PAGER");
-	if (pager == NULL || *pager == '\0')
-		pager = "more -s";
+	pager = "less -s";
 	cp = mandoc_strdup(pager);
 
 	/*
