From 8307dd573d76b0e4fbb1fbbcb0ffa1809314047a Mon Sep 17 00:00:00 2001
From: Ishimoto Shinobu <nagakamira@gmail.com>
Date: Fri, 21 Aug 2020 20:06:51 +0500
Subject: [PATCH] add missing locale.h

Signed-off-by: Ishimoto Shinobu <nagakamira@gmail.com>
---
 src/main.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/main.c b/src/main.c
index 1a669fc..8615217 100644
--- a/src/main.c
+++ b/src/main.c
@@ -21,6 +21,7 @@
 
 #include <poll.h>
 #include <stdlib.h>
+#include <locale.h>
 
 #include <dbus/dbus-glib-bindings.h>
 #include <glib.h>
-- 
2.18.4

