From 928fdff8841b3e2041a169af32181357729ed772 Mon Sep 17 00:00:00 2001
From: Ishimoto Shinobu <nagakamira@gmail.com>
Date: Thu, 26 Mar 2020 21:37:40 +0900
Subject: [PATCH] don't link against systemd

Signed-off-by: Ishimoto Shinobu <nagakamira@gmail.com>
---
 pppd/CMakeLists.txt | 5 -----
 1 file changed, 5 deletions(-)

diff --git a/pppd/CMakeLists.txt b/pppd/CMakeLists.txt
index 185e4a6..56e7d2e 100644
--- a/pppd/CMakeLists.txt
+++ b/pppd/CMakeLists.txt
@@ -31,11 +31,6 @@ if (HAVE_PAMAPPL_H)
 	target_link_libraries(pppd -lpam)
 endif()
 
-if (HAVE_SD_DAEMON_H)
-	add_definitions(-DSYSTEMD)
-	target_link_libraries(pppd -lsystemd)
-endif()
-
 if (OPTION_LPPPD_INET6)
 	target_sources(pppd PUBLIC ipv6cp.c eui64.c)
 endif()
-- 
2.25.1

