From e6ea8b10495d469c318c18c549f53c117305f40b Mon Sep 17 00:00:00 2001
From: Ishimoto Shinobu <nagakamira@gmail.com>
Date: Mon, 22 Jun 2020 05:09:19 +0900
Subject: [PATCH] no gnu gettext

---
 configure.ac | 1 -
 1 file changed, 1 deletion(-)

diff --git a/configure.ac b/configure.ac
index b1bcc707..a8abdf02 100644
--- a/configure.ac
+++ b/configure.ac
@@ -89,7 +89,6 @@ dnl *** i18n support ***
 dnl ********************
 IT_PROG_INTLTOOL([0.40.0], [no-xml])
 xgversion="`xgettext --version | head -1 | grep -o '[[0-9.]]\+$'`"
-AS_IF([test x"$xgversion" \< x"0.16"], AC_MSG_ERROR([GNU gettext tools must be at least version 0.16]))
 GETTEXT_PACKAGE=m4_default([], [AC_PACKAGE_NAME])
 AC_SUBST([GETTEXT_PACKAGE])
 AC_DEFINE_UNQUOTED([GETTEXT_PACKAGE], ["$GETTEXT_PACKAGE"], [Name of default gettext domain])
