From ad2465ad0d54324d07bdbaa2ade1b17b526f1e3b Mon Sep 17 00:00:00 2001
From: Ishimoto Shinobu <nagakamira@gmail.com>
Date: Sun, 4 Apr 2021 08:51:03 -0700
Subject: [PATCH 2/4] don't use ldd

Signed-off-by: Ishimoto Shinobu <nagakamira@gmail.com>
---
 dracut-functions.sh |  3 ++-
 dracut.sh           | 14 +++++++-------
 2 files changed, 9 insertions(+), 8 deletions(-)

diff --git a/dracut-functions.sh b/dracut-functions.sh
index 8ce1ef0..965d87c 100755
--- a/dracut-functions.sh
+++ b/dracut-functions.sh
@@ -83,7 +83,8 @@ find_binary() {
 
 ldconfig_paths()
 {
-    $DRACUT_LDCONFIG ${dracutsysrootdir:+-r ${dracutsysrootdir} -f /etc/ld.so.conf} -pN 2>/dev/null | grep -E -v '/(lib|lib64|usr/lib|usr/lib64)/[^/]*$' | sed -n 's,.* => \(.*\)/.*,\1,p' | sort | uniq
+#    $DRACUT_LDCONFIG ${dracutsysrootdir:+-r ${dracutsysrootdir} -f /etc/ld.so.conf} -pN 2>/dev/null | grep -E -v '/(lib|lib64|usr/lib|usr/lib64)/[^/]*$' | sed -n 's,.* => \(.*\)/.*,\1,p' | sort | uniq
+     :
 }
 
 # Version comparision function.  Assumes Linux style version scheme.
diff --git a/dracut.sh b/dracut.sh
index 062ed6c..1dee69f 100755
--- a/dracut.sh
+++ b/dracut.sh
@@ -2091,13 +2091,13 @@ if [[ $kernel_only != yes ]]; then
     for f in "$dracutsysrootdir"/etc/ld.so.conf "$dracutsysrootdir"/etc/ld.so.conf.d/*; do
         [[ -f $f ]] && inst_simple "${f#$dracutsysrootdir}"
     done
-    if ! $DRACUT_LDCONFIG -r "$initdir" -f /etc/ld.so.conf; then
-        if [[ $EUID = 0 ]]; then
-            derror "ldconfig exited ungracefully"
-        else
-            derror "ldconfig might need uid=0 (root) for chroot()"
-        fi
-    fi
+#    if ! $DRACUT_LDCONFIG -r "$initdir" -f /etc/ld.so.conf; then
+#        if [[ $EUID = 0 ]]; then
+#            derror "ldconfig exited ungracefully"
+#        else
+#            derror "ldconfig might need uid=0 (root) for chroot()"
+#        fi
+#    fi
 fi
 
 if [[ $do_strip = yes ]] && ! [[ $DRACUT_FIPS_MODE ]]; then
-- 
2.31.0

