From 5c575b2bc1105974007b80c6b42de5e20f0223d4 Mon Sep 17 00:00:00 2001
From: Ishimoto Shinobu <nagakamira@gmail.com>
Date: Fri, 15 Jan 2021 16:46:31 +0900
Subject: [PATCH 2/4] don't use ldd

Signed-off-by: Ishimoto Shinobu <nagakamira@gmail.com>
---
 dracut-functions.sh |  3 ++-
 dracut.sh           | 14 +++++++-------
 2 files changed, 9 insertions(+), 8 deletions(-)

diff --git a/dracut-functions.sh b/dracut-functions.sh
index e5e6884..3394a92 100755
--- a/dracut-functions.sh
+++ b/dracut-functions.sh
@@ -81,7 +81,8 @@ find_binary() {
 
 ldconfig_paths()
 {
-    $DRACUT_LDCONFIG ${dracutsysrootdir:+-r ${dracutsysrootdir} -f /etc/ld.so.conf} -pN 2>/dev/null | grep -E -v '/(lib|lib64|usr/lib|usr/lib64)/[^/]*$' | sed -n 's,.* => \(.*\)/.*,\1,p' | sort | uniq
+#    $DRACUT_LDCONFIG ${dracutsysrootdir:+-r ${dracutsysrootdir} -f /etc/ld.so.conf} -pN 2>/dev/null | grep -E -v '/(lib|lib64|usr/lib|usr/lib64)/[^/]*$' | sed -n 's,.* => \(.*\)/.*,\1,p' | sort | uniq
+     :
 }
 
 # Version comparision function.  Assumes Linux style version scheme.
diff --git a/dracut.sh b/dracut.sh
index 0f46483..f562fa6 100755
--- a/dracut.sh
+++ b/dracut.sh
@@ -1729,13 +1729,13 @@ if [[ $kernel_only != yes ]]; then
     for f in $dracutsysrootdir/etc/ld.so.conf $dracutsysrootdir/etc/ld.so.conf.d/*; do
         [[ -f $f ]] && inst_simple "${f#$dracutsysrootdir}"
     done
-    if ! $DRACUT_LDCONFIG -r "$initdir" -f /etc/ld.so.conf; then
-        if [[ $EUID = 0 ]]; then
-            derror "ldconfig exited ungracefully"
-        else
-            derror "ldconfig might need uid=0 (root) for chroot()"
-        fi
-    fi
+#    if ! $DRACUT_LDCONFIG -r "$initdir" -f /etc/ld.so.conf; then
+#        if [[ $EUID = 0 ]]; then
+#            derror "ldconfig exited ungracefully"
+#        else
+#            derror "ldconfig might need uid=0 (root) for chroot()"
+#        fi
+#    fi
 fi
 
 if [[ $do_hardlink = yes ]] && command -v hardlink >/dev/null; then
-- 
2.29.2

