From dec390f6484bd032e77db1cafa3610079c7b1fb0 Mon Sep 17 00:00:00 2001
From: Ishimoto Shinobu <nagakamira@gmail.com>
Date: Fri, 15 Jan 2021 16:49:01 +0900
Subject: [PATCH 3/4] add support for systemd-veritysetup

Signed-off-by: Ishimoto Shinobu <nagakamira@gmail.com>
---
 modules.d/90crypt/module-setup.sh | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/modules.d/90crypt/module-setup.sh b/modules.d/90crypt/module-setup.sh
index 58e5f34..57beb3b 100755
--- a/modules.d/90crypt/module-setup.sh
+++ b/modules.d/90crypt/module-setup.sh
@@ -148,7 +148,9 @@ install() {
         # the enablement symlinks
         inst_multiple -o \
                       $systemdutildir/system-generators/systemd-cryptsetup-generator \
+                      $systemdutildir/system-generators/systemd-veritysetup-generator \
                       $systemdutildir/systemd-cryptsetup \
+                      $systemdutildir/systemd-veritysetup \
                       $systemdsystemunitdir/systemd-ask-password-console.path \
                       $systemdsystemunitdir/systemd-ask-password-console.service \
                       $systemdsystemunitdir/cryptsetup.target \
-- 
2.29.2

