From 6b8348b2fc38e0e7005dd57a563f633b1b51be91 Mon Sep 17 00:00:00 2001
From: Ishimoto Shinobu <nagakamira@gmail.com>
Date: Sun, 4 Apr 2021 08:51:25 -0700
Subject: [PATCH 3/4] add support for systemd-veritysetup

Signed-off-by: Ishimoto Shinobu <nagakamira@gmail.com>
---
 modules.d/90crypt/module-setup.sh | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/modules.d/90crypt/module-setup.sh b/modules.d/90crypt/module-setup.sh
index 94ad848..0dfae71 100755
--- a/modules.d/90crypt/module-setup.sh
+++ b/modules.d/90crypt/module-setup.sh
@@ -148,7 +148,9 @@ install() {
         # the enablement symlinks
         inst_multiple -o \
                       $systemdutildir/system-generators/systemd-cryptsetup-generator \
+                      $systemdutildir/system-generators/systemd-veritysetup-generator \
                       $systemdutildir/systemd-cryptsetup \
+                      $systemdutildir/systemd-veritysetup \
                       $systemdsystemunitdir/systemd-ask-password-console.path \
                       $systemdsystemunitdir/systemd-ask-password-console.service \
                       $systemdsystemunitdir/cryptsetup.target \
-- 
2.31.0

