From 7aa923853625add4c9fb6f6a3eb97e686f4d7331 Mon Sep 17 00:00:00 2001
From: Ishimoto Shinobu <nagakamira@gmail.com>
Date: Mon, 15 Jul 2019 20:23:13 +0900
Subject: [PATCH] let's do it easy

Signed-off-by: Ishimoto Shinobu <nagakamira@gmail.com>
---
 kagami.in | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/kagami.in b/kagami.in
index 778ea36..43a49ea 100755
--- a/kagami.in
+++ b/kagami.in
@@ -790,11 +790,11 @@ sysup() {
 			die "Package database not found"
 		fi
 
-		for pkgs in $(find "$ROOTDIR/$PACKAGES/" -mindepth 1 -maxdepth 1 -type d -print0 | xargs -0 | sort | sed -e "s,$ROOTDIR/$PACKAGES/,,g"); do
+		for pkgs in $(ls "$ROOTDIR/$PACKAGES" | sort); do
 			echo "$pkgs" >> /tmp/2update
 		done
 	else
-		for pkgs in $(find "$ROOTDIR/$PACKAGES/" -mindepth 1 -maxdepth 1 -type d -print0 | xargs -0 | sort | sed -e "s,$ROOTDIR/$PACKAGES/,,g"); do
+		for pkgs in $(ls "$ROOTDIR/$PACKAGES" | sort); do
 			myfolder="$(find_portdir $pkgs)"
 
 			if [ -f "$myfolder/KagamiBuild" ]; then
