From c20f9f988f6ce5cc296f482e3b49a31d194df881 Mon Sep 17 00:00:00 2001
From: Ishimoto Shinobu <nagakamira@gmail.com>
Date: Mon, 15 Jul 2019 20:40:58 +0900
Subject: [PATCH] needed for emerge

Signed-off-by: Ishimoto Shinobu <nagakamira@gmail.com>
---
 kagami.in | 13 +++++++++++++
 1 file changed, 13 insertions(+)

diff --git a/kagami.in b/kagami.in
index 43a49ea..f7b949f 100755
--- a/kagami.in
+++ b/kagami.in
@@ -723,6 +723,19 @@ emerge_package() {
 		die "Your package was not specified"
 	fi
 
+	if [ -f "$ROOTDIR/$PACKAGES/$NAME/METADATA" ]; then
+		if [ "$NEEDED" = "yes" ]; then
+			warn "$NAME: This package has been already installed"
+			ALREADYINSTALLED=1
+			break
+		fi
+	fi
+
+	if [ "$ALREADYINSTALLED" = "1" ]; then
+		rm -rf "$TMPDIR"
+		exit 0
+	fi
+
 	pushd "$(find_portdir $pkg)"
 		export TOPDIR="$PWD"
 		source_recipe
