From 077c7ef9467482d0aae2010ff04276a994369bb7 Mon Sep 17 00:00:00 2001
From: Ishimoto Shinobu <nagakamira@gmail.com>
Date: Sun, 26 Jul 2020 15:51:48 +0300
Subject: [PATCH] musl qsort_r

Signed-off-by: Ishimoto Shinobu <nagakamira@gmail.com>
---
 pipewire-jack/src/pipewire-jack.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/pipewire-jack/src/pipewire-jack.c b/pipewire-jack/src/pipewire-jack.c
index dfd7718..68754fc 100644
--- a/pipewire-jack/src/pipewire-jack.c
+++ b/pipewire-jack/src/pipewire-jack.c
@@ -4043,7 +4043,11 @@ const char ** jack_get_ports (jack_client_t *client,
 	pthread_mutex_unlock(&c->context.lock);
 
 	if (count > 0) {
+#ifdef __GLIBC__
 		qsort_r(tmp, count, sizeof(struct object *), port_compare_func, c);
+#else
+		qsort(tmp, count, sizeof(struct object *), port_compare_func);
+#endif
 
 		res = malloc(sizeof(char*) * (count + 1));
 		for (i = 0; i < count; i++)
-- 
2.28.0.rc2

