From 9058d0899f849f466a644af7d2e0a446d6769d01 Mon Sep 17 00:00:00 2001
From: Ishimoto Shinobu <nagakamira@gmail.com>
Date: Sun, 26 Jul 2020 17:17:49 +0300
Subject: [PATCH] suckless

Signed-off-by: Ishimoto Shinobu <nagakamira@gmail.com>
---
 Make.rules | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Make.rules b/Make.rules
index 042585b..725603d 100644
--- a/Make.rules
+++ b/Make.rules
@@ -21,7 +21,7 @@ include $(TOPDIR)/Make.version
 	  -Wl,-soname,$@.1 \
 	  -Wl,--version-script=$(MAP) \
 	  -o $@ $^ $(LDLIBS)
-	ln -vfs $@ $@.1
+	ln -fs $@ $@.1
 
 %.abixml : %.so
 	$(ABIDW) --headers-dir $(TOPDIR)/src/include/efivar/ --out-file $@ $^
-- 
2.28.0.rc2

