From e93998d8d7096300dc404dee16f35d4538fc519c Mon Sep 17 00:00:00 2001
From: Ishimoto Shinobu <nagakamira@gmail.com>
Date: Mon, 24 Aug 2020 18:49:54 +0500
Subject: [PATCH] use better font

Signed-off-by: Ishimoto Shinobu <nagakamira@gmail.com>
---
 terminal/terminal-preferences.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/terminal/terminal-preferences.c b/terminal/terminal-preferences.c
index 3f941e7..2d796fb 100644
--- a/terminal/terminal-preferences.c
+++ b/terminal/terminal-preferences.c
@@ -795,7 +795,7 @@ terminal_preferences_class_init (TerminalPreferencesClass *klass)
       g_param_spec_string ("font-name",
                            NULL,
                            "FontName",
-                           "Monospace 12",
+                           "Source Code Pro 10",
                            G_PARAM_READWRITE | G_PARAM_STATIC_STRINGS);
 
   /**
-- 
2.18.4

