From 9b6356061c0b45d1415b673a905f6058e54a5e40 Mon Sep 17 00:00:00 2001
From: Ishimoto Shinobu <nagakamira@gmail.com>
Date: Thu, 6 Feb 2020 12:06:46 -0600
Subject: [PATCH] fix

Signed-off-by: Ishimoto Shinobu <nagakamira@gmail.com>
---
 src/CMakeLists.txt | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index c80ec42..5907110 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -99,6 +99,7 @@ SET (zypper_HEADERS
   commands/locale/localescmd.h
   commands/locale/addlocalecmd.h
   commands/locale/removelocalecmd.h
+  utils/rpmatch.h
 )
 
 SET( zypper_SRCS
@@ -177,6 +178,7 @@ SET( zypper_SRCS
   commands/locale/localescmd.cc
   commands/locale/addlocalecmd.cc
   commands/locale/removelocalecmd.cc
+  utils/rpmatch.c
   ${zypper_HEADERS}
 )
 
-- 
2.18.2

