From 67c4565e97f952a87156dbbcec47e191e4583e45 Mon Sep 17 00:00:00 2001
From: Ishimoto Shinobu <nagakamira@gmail.com>
Date: Thu, 6 Feb 2020 11:49:55 -0600
Subject: [PATCH] uuh

Signed-off-by: Ishimoto Shinobu <nagakamira@gmail.com>
---
 src/utils/rpmatch.c | 13 -------------
 1 file changed, 13 deletions(-)

diff --git a/src/utils/rpmatch.c b/src/utils/rpmatch.c
index ee9bcdc..ce15cb2 100644
--- a/src/utils/rpmatch.c
+++ b/src/utils/rpmatch.c
@@ -30,27 +30,14 @@
  *
  ***********************************************************************************/
 
-#include "config.h"
 #include "rpmatch.h"
 
-#if HAVE_STDIO_H
 #include <stdio.h>
-#endif
-#if HAVE_STDLIB_H
 #include <stdlib.h>
-#endif
-#if HAVE_STRING_H
 #include <string.h>
-#endif
-#if HAVE_REGEX_H
 #include <regex.h>
-#endif
-#if HAVE_LIBINTL_H
 #include <libintl.h>
-#endif
-#if HAVE_LOCALE_H
 #include <locale.h>
-#endif
 
 /**
  * The compare_t structure describes one entry in a list of english
-- 
2.18.2

