From 9d08735d2ee4ff18129c62ccc1fc6443bf309542 Mon Sep 17 00:00:00 2001
From: Ishimoto Shinobu <nagakamira@gmail.com>
Date: Thu, 6 Feb 2020 11:43:26 -0600
Subject: [PATCH] Update prompt.cc

---
 src/utils/prompt.cc | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/utils/prompt.cc b/src/utils/prompt.cc
index 2b35cc14b..764a7899c 100644
--- a/src/utils/prompt.cc
+++ b/src/utils/prompt.cc
@@ -9,6 +9,7 @@
 #include <iostream>
 #include <sstream>
 #include <fstream>
+#include <stdlib.h>
 #include <poll.h>
 #include <readline/readline.h>
 #include <termios.h>
