From b3b29d14d6da88dc843b498fe2a5ccdd2cde4637 Mon Sep 17 00:00:00 2001
From: Ishimoto Shinobu <nagakamira@gmail.com>
Date: Thu, 6 Feb 2020 11:38:42 -0600
Subject: [PATCH] Update Command.cc

---
 src/Command.cc | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/src/Command.cc b/src/Command.cc
index 2bd6cfee9..c0b71b4b7 100644
--- a/src/Command.cc
+++ b/src/Command.cc
@@ -52,12 +52,14 @@ using namespace zypp;
 ///////////////////////////////////////////////////////////////////
 namespace
 {
+/*
 #define EXPAND_PARAMETER_PACKS_IN_LAMBDAS
 #ifdef __GNUC__
 #  if !__GNUC_PREREQ(4,9)
 #    undef EXPAND_PARAMETER_PACKS_IN_LAMBDAS
 #  endif
 #endif
+*/
 
 #ifdef EXPAND_PARAMETER_PACKS_IN_LAMBDAS
   template < typename T, typename ... Args >
