diff -ruN cogl-1.22.4-orig/configure.ac cogl-1.22.4/configure.ac
--- cogl-1.22.4-orig/configure.ac	2019-04-23 19:18:51.000000000 +0900
+++ cogl-1.22.4/configure.ac	2019-09-19 20:59:46.485356897 +0900
@@ -1506,14 +1506,6 @@
 cogl-gles2/Makefile
 cogl-gles2/cogl-gles2-1.0.pc
 cogl-gles2/cogl-gles2-2.0-experimental.pc
-doc/Makefile
-doc/reference/Makefile
-doc/reference/cogl/Makefile
-doc/reference/cogl/cogl-docs.xml
-doc/reference/cogl-2.0-experimental/Makefile
-doc/reference/cogl-2.0-experimental/cogl-2.0-experimental-docs.xml
-doc/reference/cogl-gst/Makefile
-doc/reference/cogl-gst/cogl-gst-docs.xml
 examples/Makefile
 tests/Makefile
 tests/config.env
diff -ruN cogl-1.22.4-orig/Makefile.am cogl-1.22.4/Makefile.am
--- cogl-1.22.4-orig/Makefile.am	2019-04-23 19:18:51.000000000 +0900
+++ cogl-1.22.4/Makefile.am	2019-09-19 20:59:25.801884340 +0900
@@ -18,7 +18,7 @@
 SUBDIRS += cogl-gst
 endif
 
-SUBDIRS += tests examples doc po build
+SUBDIRS += tests examples po build
 
 ACLOCAL_AMFLAGS = -I build/autotools ${ACLOCAL_FLAGS}
 
@@ -33,7 +33,7 @@
 DISTCLEANFILES=
 
 DISTCHECK_CONFIGURE_FLAGS = \
-	--enable-gtk-doc \
+	--disable-gtk-doc \
 	--enable-maintainer-flags \
 	--enable-gles2 \
 	--enable-gl \
