From 812a143e251b8f6af40f6b6960921fc94467ff53 Mon Sep 17 00:00:00 2001
From: Ishimoto Shinobu <nagakamira@gmail.com>
Date: Thu, 11 Feb 2021 01:45:24 +0900
Subject: [PATCH] doas: remove afterinstall

Signed-off-by: Ishimoto Shinobu <nagakamira@gmail.com>
---
 usr.bin/doas/Makefile | 4 ----
 1 file changed, 4 deletions(-)

diff --git a/usr.bin/doas/Makefile b/usr.bin/doas/Makefile
index 4bd1ff3..111000f 100644
--- a/usr.bin/doas/Makefile
+++ b/usr.bin/doas/Makefile
@@ -15,8 +15,4 @@ CFLAGS+= -I${.CURDIR}
 COPTS+=	-Wall
 YFLAGS=
 
-afterinstall:
-	chown ${BINOWN}:${BINOWN} ${DESTDIR}/${BINDIR}/doas
-	chmod ${BINMODE} ${DESTDIR}/${BINDIR}/doas
-
 include ${.TOPDIR}/mk/bsd.prog.mk
-- 
2.29.2

