From ac5697e989135d3fa0d28453fa9850c0eeeab4ec Mon Sep 17 00:00:00 2001
From: Ishimoto Shinobu <nagakamira@gmail.com>
Date: Sat, 3 Apr 2021 11:01:24 -0700
Subject: [PATCH] debian was a mistake

---
 usr.bin/mg/Makefile | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/usr.bin/mg/Makefile b/usr.bin/mg/Makefile
index eb2d082..1c7b603 100644
--- a/usr.bin/mg/Makefile
+++ b/usr.bin/mg/Makefile
@@ -33,7 +33,7 @@ SRCS+=	cmode.c cscope.c dired.c grep.c interpreter.c tags.c
 #SRCS+=	log.c
 
 afterinstall:
-	${INSTALL} -d -o root -g wheel ${DESTDIR}${DOCDIR}/mg
+	${INSTALL} -d ${DESTDIR}${DOCDIR}/mg
 	${INSTALL} ${INSTALL_COPY} -o ${DOCOWN} -g ${DOCGRP} -m ${DOCMODE} \
 	    $(shell pwd | sed 's|/obj||')/tutorial ${DESTDIR}${DOCDIR}/mg
 
