diff -ruN ocaml-4.09.0-orig/Makefile ocaml-4.09.0/Makefile
--- ocaml-4.09.0-orig/Makefile	2019-09-17 21:29:28.000000000 +0900
+++ ocaml-4.09.0/Makefile	2019-09-18 21:36:58.781376590 +0900
@@ -43,8 +43,8 @@
 
 include stdlib/StdlibModules
 
-CAMLC=$(BOOT_OCAMLC) -g -nostdlib -I boot -use-prims runtime/primitives
-CAMLOPT=$(CAMLRUN) ./ocamlopt -g -nostdlib -I stdlib -I otherlibs/dynlink
+CAMLC=$(BOOT_OCAMLC) -nostdlib -I boot -use-prims runtime/primitives
+CAMLOPT=$(CAMLRUN) ./ocamlopt -nostdlib -I stdlib -I otherlibs/dynlink
 ARCHES=amd64 i386 arm arm64 power s390x
 INCLUDES=-I utils -I parsing -I typing -I bytecomp -I file_formats \
         -I lambda -I middle_end -I middle_end/closure \
