diff -ruN ocaml-4.10.0-orig/tools/check-parser-uptodate-or-warn.sh ocaml-4.10.0/tools/check-parser-uptodate-or-warn.sh
--- ocaml-4.10.0-orig/tools/check-parser-uptodate-or-warn.sh	2020-02-21 22:50:54.000000000 +0900
+++ ocaml-4.10.0/tools/check-parser-uptodate-or-warn.sh	2020-02-29 20:29:34.154203362 +0900
@@ -51,8 +51,8 @@
 }
 
 # The check itself
-SOURCE_MTIME=$(mtime parsing/parser.mly)
-GENERATED_MTIME=$(mtime boot/menhir/parser.ml)
+SOURCE_MTIME=$(mtime "$builddir"/parsing/parser.mly)
+GENERATED_MTIME=$(mtime "$builddir"/boot/menhir/parser.ml)
 if test -z "$SOURCE_MTIME" -o -z "$GENERATED_MTIME"
 then
   echo
