diff -ruN ocaml-4.10.0-orig/asmcomp/emitaux.ml ocaml-4.10.0/asmcomp/emitaux.ml
--- ocaml-4.10.0-orig/asmcomp/emitaux.ml	2020-02-21 22:50:54.000000000 +0900
+++ ocaml-4.10.0/asmcomp/emitaux.ml	2020-02-29 20:28:13.606122284 +0900
@@ -213,6 +213,7 @@
     | None -> a.efa_word 0
     end
   in
+  a.efa_align Arch.size_addr;
   a.efa_word (List.length !frame_descriptors);
   List.iter emit_frame !frame_descriptors;
   Label_table.iter emit_debuginfo debuginfos;
diff -ruN ocaml-4.10.0-orig/runtime/power.S ocaml-4.10.0/runtime/power.S
--- ocaml-4.10.0-orig/runtime/power.S	2020-02-21 22:50:54.000000000 +0900
+++ ocaml-4.10.0/runtime/power.S	2020-02-29 20:28:31.293700907 +0900
@@ -654,6 +654,7 @@
 
         .section ".data"
         .globl  caml_system__frametable
+	.align  wordalign
         .type   caml_system__frametable, @object
 caml_system__frametable:
         datag   1               /* one descriptor */
