From b3b1829a5b016e5663444577f9fc0ca28991b2b0 Mon Sep 17 00:00:00 2001
From: Ishimoto Shinobu <nagakamira@gmail.com>
Date: Thu, 20 Aug 2020 17:49:46 +0500
Subject: [PATCH 2/2] pie

Signed-off-by: Ishimoto Shinobu <nagakamira@gmail.com>
---
 asmcomp/emitaux.ml | 1 +
 runtime/power.S    | 1 +
 2 files changed, 2 insertions(+)

diff --git a/asmcomp/emitaux.ml b/asmcomp/emitaux.ml
index 9a1e621..2a8b4df 100644
--- a/asmcomp/emitaux.ml
+++ b/asmcomp/emitaux.ml
@@ -268,6 +268,7 @@ let emit_frames a =
     match rdbg with
     | [] -> assert false
     | d :: rest -> emit rs d rest in
+  a.efa_align Arch.size_addr;
   a.efa_word (List.length !frame_descriptors);
   List.iter emit_frame !frame_descriptors;
   Label_table.iter emit_debuginfo debuginfos;
diff --git a/runtime/power.S b/runtime/power.S
index 1933a10..921e921 100644
--- a/runtime/power.S
+++ b/runtime/power.S
@@ -651,6 +651,7 @@ caml_system__code_end:
 
         .section ".data"
         .globl  caml_system__frametable
+	.align  wordalign
         .type   caml_system__frametable, @object
 caml_system__frametable:
         datag   1               /* one descriptor */
-- 
2.18.4

