From 69eadf3b978bc078b0934ff46108166d1f819009 Mon Sep 17 00:00:00 2001
From: Ishimoto Shinobu <nagakamira@gmail.com>
Date: Sat, 17 Oct 2020 22:29:05 +0300
Subject: [PATCH] python 3.9

Signed-off-by: Ishimoto Shinobu <nagakamira@gmail.com>
---
 meson.build | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/meson.build b/meson.build
index 59b8ff5..458c5cf 100644
--- a/meson.build
+++ b/meson.build
@@ -152,7 +152,7 @@ pygobject_version = '3.8.0'
 pygobject_dep = dependency('pygobject-3.0', version: '>= ' + pygobject_version, required: get_option('python'))
 have_python = pygobject_dep.found()
 if have_python
-  python_dep = dependency('python-3.8-embed', version: '>= 3.8')
+  python_dep = dependency('python-3.9-embed', version: '>= 3.8')
 
   version_array = pygobject_version.split('.')
   config_h.set('PYGOBJECT_REQUIRED_MAJOR', version_array[0].to_int())
-- 
2.26.2

