From f6fb9089490dcb3902eda9e458678b82e78c6d60 Mon Sep 17 00:00:00 2001
From: Ishimoto Shinobu <nagakamira@gmail.com>
Date: Sun, 1 Nov 2020 16:19:26 +0300
Subject: [PATCH] fix icu build

Signed-off-by: Ishimoto Shinobu <nagakamira@gmail.com>
---
 Source/WTF/wtf/CMakeLists.txt | 1 +
 1 file changed, 1 insertion(+)

diff --git a/Source/WTF/wtf/CMakeLists.txt b/Source/WTF/wtf/CMakeLists.txt
index 564ca91d..4c2c293a 100644
--- a/Source/WTF/wtf/CMakeLists.txt
+++ b/Source/WTF/wtf/CMakeLists.txt
@@ -569,3 +569,4 @@ file(COPY
     DESTINATION
     ${WTF_SCRIPTS_DIR}
 )
+add_compile_definitions("U_DEFINE_FALSE_AND_TRUE=1")
-- 
2.28.0

