From 711caddc7a7ddf42babca8799f5a77d8559917e2 Mon Sep 17 00:00:00 2001
From: Ishimoto Shinobu <47295761+protonesso@users.noreply.github.com>
Date: Fri, 1 Nov 2019 23:09:13 +0900
Subject: [PATCH] fix build with slibtool

---
 Makefile.am | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Makefile.am b/Makefile.am
index 6344b4e2..4644fbd2 100644
--- a/Makefile.am
+++ b/Makefile.am
@@ -106,7 +106,7 @@ src/builtin.o: src/builtin.inc
 
 bin_PROGRAMS = jq
 jq_SOURCES = src/main.c src/version.h
-jq_LDFLAGS = -static-libtool-libs
+jq_LDFLAGS = -all-static
 jq_LDADD = libjq.la -lm
 
 if WIN32
