diff -ruN libcap-2.33-orig/libcap/psx.c libcap-2.33/libcap/psx.c
--- libcap-2.33-orig/libcap/psx.c	2020-03-02 08:06:25.000000000 +0900
+++ libcap-2.33/libcap/psx.c	2020-03-07 01:56:48.548849689 +0900
@@ -13,6 +13,7 @@
 #define _GNU_SOURCE
 
 #include <errno.h>
+#include <sched.h>
 #include <pthread.h>
 #include <signal.h>
 #include <stdarg.h>
@@ -533,7 +534,11 @@
 	if (!waiting) {
 	    break;
 	}
+#ifdef __GLIBC__
 	pthread_yield();
+#else
+	sched_yield();
+#endif
     }
 
     errno = restore_errno;
