diff '--color=auto' -ruN a/neofetch b/neofetch
--- a/neofetch	2021-03-29 11:11:56.862714440 +0300
+++ b/neofetch	2021-03-29 11:13:53.925485042 +0300
@@ -1519,6 +1519,7 @@
             has pkgtool && dir ${br_prefix}/var/log/packages/*
             has scratch && dir ${br_prefix}/var/lib/scratchpkg/index/*/.pkginfo
             has kagami  && dir ${br_prefix}/var/lib/kagami/pkgs/*
+            has tsukuri && dir ${br_prefix}/usr/lib/tsukuri/db/*
             has cave    && dir ${br_prefix}/var/db/paludis/repositories/cross-installed/*/data/*/ \
                                ${br_prefix}/var/db/paludis/repositories/installed/data/*/
             shopt -u nullglob
