From 087f47f5d42e1d48f86b9c1607dce58ef1247f87 Mon Sep 17 00:00:00 2001
From: Ishimoto Shinobu <nagakamira@gmail.com>
Date: Sun, 18 Apr 2021 09:26:07 -0700
Subject: [PATCH 6/9] musl asm ptrace fix for ppc

Signed-off-by: Ishimoto Shinobu <nagakamira@gmail.com>
---
 backends/ppc_initreg.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/backends/ppc_initreg.c b/backends/ppc_initreg.c
index e5cca7e..c1d90c8 100644
--- a/backends/ppc_initreg.c
+++ b/backends/ppc_initreg.c
@@ -32,6 +32,7 @@
 
 #include <stdlib.h>
 #if defined(__powerpc__) && defined(__linux__)
+# include <asm/ptrace.h>
 # include <sys/ptrace.h>
 # include <asm/ptrace.h>
 # include <sys/user.h>
-- 
2.31.0

