From 22985170253efdc5d90b8cc5c27bb63865ec613c Mon Sep 17 00:00:00 2001
From: Ishimoto Shinobu <nagakamira@gmail.com>
Date: Sun, 18 Apr 2021 09:32:52 -0700
Subject: [PATCH 9/9] options

Signed-off-by: Ishimoto Shinobu <nagakamira@gmail.com>
---
 config/eu.am | 9 +--------
 1 file changed, 1 insertion(+), 8 deletions(-)

diff --git a/config/eu.am b/config/eu.am
index 7f2f0e2..93d6038 100644
--- a/config/eu.am
+++ b/config/eu.am
@@ -61,16 +61,10 @@ else
 NULL_DEREFERENCE_WARNING=
 endif
 
-if HAVE_IMPLICIT_FALLTHROUGH_WARNING
-# Use strict fallthrough. Only __attribute__((fallthrough)) will prevent the
-# warning
-IMPLICIT_FALLTHROUGH_WARNING=-Wimplicit-fallthrough=5
-else
 IMPLICIT_FALLTHROUGH_WARNING=
-endif
 
 AM_CFLAGS = -std=gnu99 -Wall -Wshadow -Wformat=2 \
-	    -Wold-style-definition -Wstrict-prototypes -Wtrampolines \
+	    -Wold-style-definition -Wstrict-prototypes \
 	    $(LOGICAL_OP_WARNING) $(DUPLICATED_COND_WARNING) \
 	    $(NULL_DEREFERENCE_WARNING) $(IMPLICIT_FALLTHROUGH_WARNING) \
 	    $(if $($(*F)_no_Werror),,$(WERROR)) \
@@ -80,7 +74,6 @@ AM_CFLAGS = -std=gnu99 -Wall -Wshadow -Wformat=2 \
 	    $($(*F)_CFLAGS)
 
 AM_CXXFLAGS = -std=c++11 -Wall -Wshadow \
-	   -Wtrampolines \
 	   $(LOGICAL_OP_WARNING) $(DUPLICATED_COND_WARNING) \
 	   $(NULL_DEREFERENCE_WARNING) $(IMPLICIT_FALLTHROUGH_WARNING) \
 	   $(if $($(*F)_no_Werror),,-Werror) \
-- 
2.31.0

