From a95b74d23a02b421958a790870814b4bb7815a91 Mon Sep 17 00:00:00 2001
From: Ishimoto Shinobu <nagakamira@gmail.com>
Date: Thu, 6 Feb 2020 11:27:39 -0600
Subject: [PATCH] aaa

Signed-off-by: Ishimoto Shinobu <nagakamira@gmail.com>
---
 zypp/base/LogControl.h | 1 +
 1 file changed, 1 insertion(+)

diff --git a/zypp/base/LogControl.h b/zypp/base/LogControl.h
index b31c453..b95e7ad 100644
--- a/zypp/base/LogControl.h
+++ b/zypp/base/LogControl.h
@@ -13,6 +13,7 @@
 #define ZYPP_BASE_LOGCONTROL_H
 
 #include <iosfwd>
+#include <sys/stat.h>
 
 #include "zypp/base/Logger.h"
 #include "zypp/base/PtrTypes.h"
-- 
2.18.2

