From 28619b8820632a3af8b9e63fddb7e5ac5d6fb75a Mon Sep 17 00:00:00 2001
From: Ishimoto Shinobu <nagakamira@gmail.com>
Date: Thu, 6 Feb 2020 10:32:08 -0600
Subject: [PATCH] define globs

---
 zypp/Glob.h | 17 +++++++++++++++++
 1 file changed, 17 insertions(+)

diff --git a/zypp/Glob.h b/zypp/Glob.h
index a96d72c14..9a13752e4 100644
--- a/zypp/Glob.h
+++ b/zypp/Glob.h
@@ -27,6 +27,23 @@ extern "C"
 
 #include "zypp/Pathname.h"
 
+#define	GLOB_ERR	(1 << 0)/* Return on read errors.  */
+#define	GLOB_MARK	(1 << 1)/* Append a slash to each name.  */
+#define	GLOB_NOSORT	(1 << 2)/* Don't sort the names.  */
+#define	GLOB_DOOFFS	(1 << 3)/* Insert PGLOB->gl_offs NULLs.  */
+#define	GLOB_NOCHECK	(1 << 4)/* If nothing matches, return the pattern.  */
+#define	GLOB_APPEND	(1 << 5)/* Append to results of a previous call.  */
+#define	GLOB_NOESCAPE	(1 << 6)/* Backslashes don't quote metacharacters.  */
+#define	GLOB_PERIOD	(1 << 7)/* Leading `.' can be matched by metachars.  */
+#define GLOB_MAGCHAR	 (1 << 8)/* Set in gl_flags if any metachars seen.  */
+#define GLOB_ALTDIRFUNC (1 << 9)/* Use gl_opendir et al functions.  */
+#define GLOB_BRACE	 (1 << 10)/* Expand "{a,b}" to "a" "b".  */
+#define GLOB_NOMAGIC	 (1 << 11)/* If no magic chars, return the pattern.  */
+#define GLOB_TILDE	 (1 << 12)/* Expand ~user and ~ to home directories. */
+#define GLOB_ONLYDIR	 (1 << 13)/* Match only directories.  */
+#define GLOB_TILDE_CHECK (1 << 14)/* Like GLOB_TILDE but return an error
+				      if the user name is not available.  */
+
 ///////////////////////////////////////////////////////////////////
 namespace zypp
 { /////////////////////////////////////////////////////////////////
