From 2b9dd785d57b6be4ac8523ac12feba68c83ae8b7 Mon Sep 17 00:00:00 2001
From: Ishimoto Shinobu <nagakamira@gmail.com>
Date: Thu, 6 Feb 2020 11:24:14 -0600
Subject: [PATCH] #include <sys/stat.h>

---
 zypp/base/InputStream.cc | 1 +
 1 file changed, 1 insertion(+)

diff --git a/zypp/base/InputStream.cc b/zypp/base/InputStream.cc
index 2aeeb73ec..6778a7eff 100644
--- a/zypp/base/InputStream.cc
+++ b/zypp/base/InputStream.cc
@@ -10,6 +10,7 @@
  *
 */
 #include <iostream>
+#include <sys/stat.h>
 #include "zypp/base/LogTools.h"
 
 #include "zypp/base/InputStream.h"
