From 53a9a4d65e804a2640e7bef10368155f27352162 Mon Sep 17 00:00:00 2001
From: Ishimoto Shinobu <nagakamira@gmail.com>
Date: Thu, 6 Feb 2020 10:22:44 -0600
Subject: [PATCH] #include <sys/stat.h>

---
 zypp/parser/xml/Reader.cc | 1 +
 1 file changed, 1 insertion(+)

diff --git a/zypp/parser/xml/Reader.cc b/zypp/parser/xml/Reader.cc
index d06be45bf..73cf33bdc 100644
--- a/zypp/parser/xml/Reader.cc
+++ b/zypp/parser/xml/Reader.cc
@@ -13,6 +13,7 @@
 #include <libxml/xmlerror.h>
 
 #include <iostream>
+#include <sys/stat.h>
 
 #include "zypp/base/LogControl.h"
 #include "zypp/base/LogTools.h"
