From 5ba212eab7ef72232df201e8387b81d1302ef774 Mon Sep 17 00:00:00 2001
From: Ishimoto Shinobu <nagakamira@gmail.com>
Date: Thu, 6 Feb 2020 10:20:06 -0600
Subject: [PATCH] add #include <sys/stat.h>

---
 zypp/parser/xml/Node.cc | 1 +
 1 file changed, 1 insertion(+)

diff --git a/zypp/parser/xml/Node.cc b/zypp/parser/xml/Node.cc
index c90b8e153..50112df96 100644
--- a/zypp/parser/xml/Node.cc
+++ b/zypp/parser/xml/Node.cc
@@ -13,6 +13,7 @@
 #include <libxml/xmlerror.h>
 
 #include <iostream>
+#include <sys/stat.h>
 
 #include "zypp/base/LogControl.h"
 #include "zypp/base/LogTools.h"
