From 82ec8fbe049c689e2f3466bdfc95953dc3e30f97 Mon Sep 17 00:00:00 2001
From: Ishimoto Shinobu <nagakamira@gmail.com>
Date: Thu, 6 Feb 2020 10:36:43 -0600
Subject: [PATCH] #include <sys/stat.h>

---
 zypp/solver/detail/Testcase.cc | 1 +
 1 file changed, 1 insertion(+)

diff --git a/zypp/solver/detail/Testcase.cc b/zypp/solver/detail/Testcase.cc
index 272735740..2bcc1119b 100644
--- a/zypp/solver/detail/Testcase.cc
+++ b/zypp/solver/detail/Testcase.cc
@@ -13,6 +13,7 @@
 #include <fstream>
 #include <sstream>
 #include <streambuf>
+#include <sys/stat.h>
 
 #define ZYPP_USE_RESOLVER_INTERNALS
 
