From 9b62d4eada8a269df712bdb71dc586739437be25 Mon Sep 17 00:00:00 2001
From: Ishimoto Shinobu <nagakamira@gmail.com>
Date: Thu, 6 Feb 2020 10:39:23 -0600
Subject: [PATCH] #include <iostream> #include <iomanip> #include <ctime>

---
 zypp/Date.cc | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/zypp/Date.cc b/zypp/Date.cc
index 43059dd8b..aae648ffb 100644
--- a/zypp/Date.cc
+++ b/zypp/Date.cc
@@ -10,6 +10,8 @@
  *
 */
 #include <iostream>
+#include <iomanip>
+#include <ctime>
 //#include "zypp/base/Logger.h"
 
 #include "zypp/base/String.h"
