From ea7c28d5a5db84a77c501cb33dab5f69c729a330 Mon Sep 17 00:00:00 2001
From: Ishimoto Shinobu <nagakamira@gmail.com>
Date: Thu, 6 Feb 2020 10:47:11 -0600
Subject: [PATCH] Update Date.cc

---
 zypp/Date.cc | 3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/zypp/Date.cc b/zypp/Date.cc
index aae648ffb..a35cb09bb 100644
--- a/zypp/Date.cc
+++ b/zypp/Date.cc
@@ -10,8 +10,7 @@
  *
 */
 #include <iostream>
-#include <iomanip>
-#include <ctime>
+#include <time.h>
 //#include "zypp/base/Logger.h"
 
 #include "zypp/base/String.h"
