From fb2f4add8f6d0ddf635b988322ab62aa2d701012 Mon Sep 17 00:00:00 2001
From: Ishimoto Shinobu <nagakamira@gmail.com>
Date: Thu, 6 Feb 2020 10:41:51 -0600
Subject: [PATCH] Update Fetcher.cc

---
 zypp/Fetcher.cc | 1 +
 1 file changed, 1 insertion(+)

diff --git a/zypp/Fetcher.cc b/zypp/Fetcher.cc
index c6bf2173c..795f1b286 100644
--- a/zypp/Fetcher.cc
+++ b/zypp/Fetcher.cc
@@ -13,6 +13,7 @@
 #include <fstream>
 #include <list>
 #include <map>
+#include <sys/stat.h>
 
 #include "zypp/base/Easy.h"
 #include "zypp/base/LogControl.h"
