diff -ruN ifupdown/main.c ifupdown_b/main.c
--- ifupdown/main.c	2016-09-22 20:07:16.000000000 +0900
+++ ifupdown_b/main.c	2019-03-06 02:48:29.789147870 +0900
@@ -34,16 +34,7 @@
 }
 
 static void version() {
-	printf("%s version " IFUPDOWN_VERSION "\n"
-		"\n"
-		"Copyright (c) 1999-2009 Anthony Towns\n"
-		"              2010-2015 Andrew Shadura\n"
-		"              2015-2016 Guus Sliepen\n"
-		"\n"
-		"This program is free software; you can redistribute it and/or modify\n"
-		"it under the terms of the GNU General Public License as published by\n"
-		"the Free Software Foundation; either version 2 of the License, or (at\n"
-		"your option) any later version.\n", argv0);
+	printf("ifupdown 1.4.8.8", argv0);
 
 	exit(0);
 }
