diff -ruN llvm-9.0.0.src-orig/test/CMakeLists.txt llvm-9.0.0.src/test/CMakeLists.txt
--- llvm-9.0.0.src-orig/test/CMakeLists.txt	2019-05-29 08:22:12.000000000 +0900
+++ llvm-9.0.0.src/test/CMakeLists.txt	2019-10-13 06:28:01.105692067 +0900
@@ -62,7 +62,6 @@
           llvm-diff
           llvm-dis
           llvm-dlltool
-          dsymutil
           llvm-dwarfdump
           llvm-dwp
           llvm-elfabi
diff -ruN llvm-9.0.0.src-orig/tools/LLVMBuild.txt llvm-9.0.0.src/tools/LLVMBuild.txt
--- llvm-9.0.0.src-orig/tools/LLVMBuild.txt	2019-04-21 02:58:29.000000000 +0900
+++ llvm-9.0.0.src/tools/LLVMBuild.txt	2019-10-13 06:16:52.325534913 +0900
@@ -17,7 +17,6 @@
 [common]
 subdirectories =
  bugpoint
- dsymutil
  llc
  lli
  llvm-ar
