--- xorg-server-1.20.5-orig/glx/meson.build
+++ xorg-server-1.20.5/glx/meson.build
@@ -39,7 +39,7 @@
             common_dep,
             dl_dep,
             dependency('glproto', version: '>= 1.4.17'),
-            dependency('gl', version: '>= 9.2.0'),
+            dependency('gl', version: '>= 1.2'),
         ],
         c_args: [
             glx_align64,
@@ -74,7 +74,7 @@
             common_dep,
             dl_dep,
             dependency('glproto', version: '>= 1.4.17'),
-            dependency('gl', version: '>= 9.2.0'),
+            dependency('gl', version: '>= 1.2'),
         ],
     )
 
