diff -ruN musl-1.1.22-orig/configure musl-1.1.22/configure
--- musl-1.1.22-orig/configure	2019-04-10 09:39:21.000000000 +0900
+++ musl-1.1.22/configure	2019-05-23 04:09:48.820825318 +0900
@@ -311,7 +311,7 @@
 arm*) ARCH=arm ;;
 aarch64*) ARCH=aarch64 ;;
 i?86-nt32*) ARCH=nt32 ;;
-i?86*) ARCH=i386 ;;
+i?86*|pentium*|p6|6x86|athlon|athlon_*) ARCH=i386 ;;
 x86_64-x32*|x32*|x86_64*x32) ARCH=x32 ;;
 x86_64-nt64*) ARCH=nt64 ;;
 x86_64*) ARCH=x86_64 ;;
