From 146a2953dc47a6479dd055a8cdcb00c14745e3f4 Mon Sep 17 00:00:00 2001
From: Ishimoto Shinobu <nagakamira@gmail.com>
Date: Wed, 20 Jan 2021 04:55:57 +0900
Subject: [PATCH 4/5] fix GCC/Clang PIE specs

Signed-off-by: Ishimoto Shinobu <nagakamira@gmail.com>
---
 configure.ac | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/configure.ac b/configure.ac
index 7656f24..f54de35 100644
--- a/configure.ac
+++ b/configure.ac
@@ -1143,7 +1143,7 @@ if test x"$efiemu_excuse" = x ; then
   if test x"$grub_cv_target_cc_efiemu64_link_format" = xunknown; then
     efiemu_excuse="no suitable link format for efiemu64 found"
   else
-    EFIEMU64_LINK_FORMAT="-Wl,$grub_cv_target_cc_efiemu64_link_format"
+    EFIEMU64_LINK_FORMAT="-Wl,$grub_cv_target_cc_efiemu64_link_format -no-pie"
   fi
 fi
 if test x"$enable_efiemu" = xyes && test x"$efiemu_excuse" != x ; then
-- 
2.29.2

