From 238fbe493096ab16ecd7d8125e60452580ff9cf7 Mon Sep 17 00:00:00 2001
From: Ishimoto Shinobu <nagakamira@gmail.com>
Date: Sat, 27 Feb 2021 00:13:44 +0900
Subject: [PATCH] don't check for header

Signed-off-by: Ishimoto Shinobu <nagakamira@gmail.com>
---
 util/grub-mkimagexx.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/util/grub-mkimagexx.c b/util/grub-mkimagexx.c
index 00f49cc..e6799d4 100644
--- a/util/grub-mkimagexx.c
+++ b/util/grub-mkimagexx.c
@@ -2275,9 +2275,11 @@ SUFFIX (grub_mkimage_load_image) (const char *kernel_path,
   kernel_img = xmalloc (kernel_size);
   grub_util_load_image (kernel_path, kernel_img);
 
+#if 0
   e = (Elf_Ehdr *) kernel_img;
   if (! SUFFIX (check_elf_header) (e, kernel_size, image_target))
     grub_util_error ("invalid ELF header");
+#endif
 
   section_offset = grub_target_to_host (e->e_shoff);
   smd.section_entsize = grub_target_to_host16 (e->e_shentsize);
-- 
2.29.2

