diff -ruN grub-2.04-orig/util/grub-install-common.c grub-2.04/util/grub-install-common.c
--- grub-2.04-orig/util/grub-install-common.c	2019-04-23 03:54:47.000000000 -0500
+++ grub-2.04/util/grub-install-common.c	2020-01-18 12:04:53.479570766 -0600
@@ -618,8 +618,8 @@
   d = grub_util_fd_opendir (locale_dir);
   if (!d)
     {
-      grub_util_warn (_("cannot open directory `%s': %s"),
-		      locale_dir, grub_util_fd_strerror ());
+      //grub_util_warn (_("cannot open directory `%s': %s"),
+	//	      locale_dir, grub_util_fd_strerror ());
       return;
     }
 
