From 208ef980393f3e541af466ce8dc18701c235360d Mon Sep 17 00:00:00 2001
From: Ishimoto Shinobu <nagakamira@gmail.com>
Date: Wed, 10 Feb 2021 05:53:14 +0900
Subject: [PATCH 2/4] fix build on musl libc

Signed-off-by: Ishimoto Shinobu <nagakamira@gmail.com>
---
 common/elfdefinitions.h | 1 +
 1 file changed, 1 insertion(+)

diff --git a/common/elfdefinitions.h b/common/elfdefinitions.h
index 19b2943..ae6ad55 100644
--- a/common/elfdefinitions.h
+++ b/common/elfdefinitions.h
@@ -39,6 +39,7 @@
 
 #ifndef _ELFDEFINITIONS_H_
 #define _ELFDEFINITIONS_H_
+#define _ELF_H
 
 #include <stdint.h>
 
-- 
2.29.2

