From 494fdd415992bc1882d4adefd0c29d36634bc2e9 Mon Sep 17 00:00:00 2001
From: Ishimoto Shinobu <nagakamira@gmail.com>
Date: Sun, 21 Feb 2021 02:33:02 +0900
Subject: [PATCH 4/4] add support for STN_UNDEF

Signed-off-by: Ishimoto Shinobu <nagakamira@gmail.com>
---
 common/elfdefinitions.h | 1 +
 1 file changed, 1 insertion(+)

diff --git a/common/elfdefinitions.h b/common/elfdefinitions.h
index ae6ad55..7a34b87 100644
--- a/common/elfdefinitions.h
+++ b/common/elfdefinitions.h
@@ -1111,6 +1111,7 @@ enum {
  */
 #define _ELF_DEFINE_SECTION_INDICES()					\
 _ELF_DEFINE_SHN(SHN_UNDEF, 	0, 	 "undefined section")		\
+_ELF_DEFINE_SHN(STN_UNDEF, 	0, 	 "undefined section")		\
 _ELF_DEFINE_SHN(SHN_LORESERVE, 	0xFF00U, "start of reserved area")	\
 _ELF_DEFINE_SHN(SHN_LOPROC, 	0xFF00U,				\
 	"start of processor-specific range")				\
-- 
2.29.2

