From e9a7630a538978523a52066712ddd233e0fef48b Mon Sep 17 00:00:00 2001
From: Ishimoto Shinobu <nagakamira@gmail.com>
Date: Sun, 18 Apr 2021 09:00:20 -0700
Subject: [PATCH 4/6] add support for STN_UNDEF

Signed-off-by: Ishimoto Shinobu <nagakamira@gmail.com>
---
 common/elfdefinitions.h | 1 +
 1 file changed, 1 insertion(+)

diff --git a/common/elfdefinitions.h b/common/elfdefinitions.h
index 19b2943..ec83c95 100644
--- a/common/elfdefinitions.h
+++ b/common/elfdefinitions.h
@@ -1110,6 +1110,7 @@ enum {
  */
 #define _ELF_DEFINE_SECTION_INDICES()					\
 _ELF_DEFINE_SHN(SHN_UNDEF, 	0, 	 "undefined section")		\
+_ELF_DEFINE_SHN(STN_UNDEF, 	0, 	 "undefined section")		\
 _ELF_DEFINE_SHN(SHN_LORESERVE, 	0xFF00U, "start of reserved area")	\
 _ELF_DEFINE_SHN(SHN_LOPROC, 	0xFF00U,				\
 	"start of processor-specific range")				\
-- 
2.31.0

