From 36864e68a94ca8cb6744ca076adf04aff024fad1 Mon Sep 17 00:00:00 2001
From: Ishimoto Shinobu <nagakamira@gmail.com>
Date: Sun, 18 Apr 2021 09:00:49 -0700
Subject: [PATCH 5/6] fix build on musl libc

Signed-off-by: Ishimoto Shinobu <nagakamira@gmail.com>
---
 common/elfdefinitions.h | 1 +
 1 file changed, 1 insertion(+)

diff --git a/common/elfdefinitions.h b/common/elfdefinitions.h
index ec83c95..7a34b87 100644
--- a/common/elfdefinitions.h
+++ b/common/elfdefinitions.h
@@ -39,6 +39,7 @@
 
 #ifndef _ELFDEFINITIONS_H_
 #define _ELFDEFINITIONS_H_
+#define _ELF_H
 
 #include <stdint.h>
 
-- 
2.31.0

