From c56e0a6d697eb05fba5dcd8d29147df1db58c7e0 Mon Sep 17 00:00:00 2001
From: Ishimoto Shinobu <nagakamira@gmail.com>
Date: Sun, 18 Apr 2021 09:06:24 -0700
Subject: [PATCH 6/6] remove ar, ranlib and nm

Signed-off-by: Ishimoto Shinobu <nagakamira@gmail.com>
---
 Makefile | 2 --
 1 file changed, 2 deletions(-)

diff --git a/Makefile b/Makefile
index 3efdf81..1c54cbc 100644
--- a/Makefile
+++ b/Makefile
@@ -25,13 +25,11 @@ SUBDIR += isa  # ('isa' does not build on all platforms yet).
 
 # Build tools after the libraries.
 SUBDIR += addr2line
-SUBDIR += ar
 SUBDIR += brandelf
 SUBDIR += cxxfilt
 SUBDIR += elfcopy
 SUBDIR += elfdump
 SUBDIR += findtextrel
-SUBDIR += nm
 SUBDIR += readelf
 SUBDIR += size
 SUBDIR += strings
-- 
2.31.0

