--- krb5-1.18-orig/src/plugins/preauth/pkinit/pkinit_crypto_openssl.c
+++ krb5-1.18/src/plugins/preauth/pkinit/pkinit_crypto_openssl.c
@@ -185,7 +185,8 @@
     (*_x509_pp) = PKCS7_cert_from_signer_info(_p7,_si)
 #endif
 
-#if OPENSSL_VERSION_NUMBER < 0x10100000L
+#if (defined(OPENSSL_VERSION_NUMBER) && OPENSSL_VERSION_NUMBER < 0x10100000L) || \
+     defined(LIBRESSL_VERSION_NUMBER)
 
 /* 1.1 standardizes constructor and destructor names, renaming
  * EVP_MD_CTX_{create,destroy} and deprecating ASN1_STRING_data. */
@@ -245,6 +246,10 @@
 
 #endif
 
+#if defined(LIBRESSL_VERSION_NUMBER) && !defined(static_ASN1_SEQUENCE_END_name)
+#define static_ASN1_SEQUENCE_END_name	ASN1_SEQUENCE_END_name
+#endif
+
 static struct pkcs11_errstrings {
     short code;
     char *text;
@@ -2924,7 +2929,9 @@
     return retval;
 }
 
-#if OPENSSL_VERSION_NUMBER >= 0x10100000L
+#if ((defined(OPENSSL_VERSION_NUMBER) && OPENSSL_VERSION_NUMBER >= 0x10100000L) && \
+     !defined(LIBRESSL_VERSION_NUMBER)) || \
+     (defined(LIBRESSL_VERSION_NUMBER) && LIBRESSL_VERSION_NUMBER >= 0x20900000L)
 
 /*
  * We need to decode DomainParameters from RFC 3279 section 2.3.3.  We would
--- krb5-1.18-orig/src/plugins/preauth/pkinit/pkinit_crypto_openssl.h
+++ krb5-1.18/src/plugins/preauth/pkinit/pkinit_crypto_openssl.h
@@ -46,7 +46,9 @@
 #include <openssl/asn1.h>
 #include <openssl/pem.h>
 
-#if OPENSSL_VERSION_NUMBER >= 0x10100000L
+#if ((defined(OPENSSL_VERSION_NUMBER) && OPENSSL_VERSION_NUMBER >= 0x10100000L) && \
+     !defined(LIBRESSL_VERSION_NUMBER)) || \
+     (defined(LIBRESSL_VERSION_NUMBER) && LIBRESSL_VERSION_NUMBER >= 0x20900000L)
 #include <openssl/asn1t.h>
 #else
 #include <openssl/asn1_mac.h>
--- krb5-1.18-orig/src/tests/softpkcs11/main.c
+++ krb5-1.18/src/tests/softpkcs11/main.c
@@ -45,7 +45,7 @@
 
 #include <pkcs11.h>
 
-#if OPENSSL_VERSION_NUMBER < 0x10100000L
+#if OPENSSL_VERSION_NUMBER < 0x10100000L || defined(LIBRESSL_VERSION_NUMBER)
 #define EVP_PKEY_get0_RSA(key) ((key)->pkey.rsa)
 #define RSA_PKCS1_OpenSSL RSA_PKCS1_SSLeay
 #define RSA_get0_key compat_rsa_get0_key
